{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "be124db9",
   "metadata": {},
   "source": [
    "# Aggregated .maf files from CBTN"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "400aae99",
   "metadata": {},
   "source": [
    "`scp /Users/stearb/Downloads/CBTN_somatic_variants_VEP_HIGH_July29.csv stearb-hpc9.research.chop.edu:/home/stearb/U24/data/somatic_variants/`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "68aef358",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import polars as pl\n",
    "import numpy as np\n",
    "\n",
    "def fill_missing_cols(df):\n",
    "    if 'node_id' not in df.columns:\n",
    "        raise ValueError('Must have at least a \"node_id\" column.')\n",
    "    \n",
    "    all_cols = set([ 'node_label', 'node_synonyms', 'node_dbxrefs',\n",
    "            'node_definition','node_namespace','value','lowerbound','upperbound','unit'])\n",
    "    missing_cols = list(all_cols - set(df.columns))\n",
    "    nan_cols_df = pd.DataFrame(np.full([len(df), len(missing_cols)], np.nan),columns=missing_cols)\n",
    "\n",
    "    if isinstance(df, pd.DataFrame):\n",
    "        nan_cols_df.index = df.index\n",
    "        return pd.concat([df,nan_cols_df],axis=1)\n",
    "    elif isinstance(df, pl.DataFrame):\n",
    "        return pl.concat([df,pl.from_pandas(nan_cols_df)],how='horizontal') # no index for polars\n",
    "    else:\n",
    "        raise ValueError(f'Must Pass either a pandas DataFrame or a polars DataFrame but recieved \"{type(df)}\".')\n",
    "    \n",
    "\n",
    "def bin_column(df, col2bin=''):\n",
    "    '''works with polars df'''\n",
    "    \n",
    "    '''tpm_bins = list([0.0000000,7e-4,8e-4,9e-4]) + list(np.linspace(1e-3,9e-3,9)) + \\\n",
    "               list(np.round(np.linspace(1e-2,9e-2,9),2)) + list(np.round(np.linspace(.1,1,10),2)) + \\\n",
    "               list(np.linspace(2,100,99)) + list(np.arange(100,1100,100)[1:]) +  \\\n",
    "                list(np.arange(2000,11000,1000)) + list(np.arange(20_000,110_000,10_000)) + [300_000] '''\n",
    "    \n",
    "    bins = np.unique(hscloChrom['lowerbound'].values) #hscloChrom['lowerbound'].values\n",
    "    # check that none of the disease expression values are out of range, ie larger than the max bin value\n",
    "    # use for pandas\n",
    "    #if np.sum(df['Disease_MeanTpm'] > np.max(tpm_bins)) == 0:\n",
    "        \n",
    "    # use for polars\n",
    "    if len(df.filter(df[col2bin] > np.max(bins))) == 0:\n",
    "        df_binned = df.with_columns(  df[col2bin].cut(breaks=bins).alias('bins')  )\n",
    "        # check that order has been maintained #assert pl.all(df[col2bin] == df_binned[col2bin])\n",
    "    else:\n",
    "        print('OUT OF RANGE ERROR!')\n",
    "        assert False \n",
    "    return df_binned"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "8579f31c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HGVSg</th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>dbSNP_RS</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>SYMBOL_SOURCE</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>gnomad_3_1_1_AF</th>\n",
       "      <th>gnomad_3_1_1_AF_popmax</th>\n",
       "      <th>gnomad_3_1_1_AF_non_cancer_popmax</th>\n",
       "      <th>gnomad_3_1_1_AF_non_cancer_all_popmax</th>\n",
       "      <th>Matched_Norm_Sample_Barcode</th>\n",
       "      <th>Tumor_Sample_Barcode</th>\n",
       "      <th>VARIANT_CLASS</th>\n",
       "      <th>path</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr7:g.138981269T&gt;G</td>\n",
       "      <td>KIAA1549</td>\n",
       "      <td>p.Met1?</td>\n",
       "      <td>novel</td>\n",
       "      <td>start_lost</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:22219</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>2.75e-05</td>\n",
       "      <td>3.05e-05</td>\n",
       "      <td>2.45e-05</td>\n",
       "      <td>0.000123</td>\n",
       "      <td>BS_65FNRFMM</td>\n",
       "      <td>BS_QWWYFRGZ</td>\n",
       "      <td>SNV</td>\n",
       "      <td>/sbgenomics/project-files/CBTN_somatic/002daa4...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chrX:g.45069977_45069978del</td>\n",
       "      <td>KDM6A</td>\n",
       "      <td>p.Ser775ArgfsTer15</td>\n",
       "      <td>novel</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:12637</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>BS_65FNRFMM</td>\n",
       "      <td>BS_QWWYFRGZ</td>\n",
       "      <td>deletion</td>\n",
       "      <td>/sbgenomics/project-files/CBTN_somatic/002daa4...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1:g.153767777_153767786dup</td>\n",
       "      <td>INTS3</td>\n",
       "      <td>p.Val735GlyfsTer21</td>\n",
       "      <td>novel</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:26153</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>BS_KX3258Y6</td>\n",
       "      <td>BS_MVQYCQMP</td>\n",
       "      <td>insertion</td>\n",
       "      <td>/sbgenomics/project-files/CBTN_somatic/0046a77...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr2:g.11216160_11216161delinsAA</td>\n",
       "      <td>ROCK2</td>\n",
       "      <td>p.Glu486_Glu487delinsAspTer</td>\n",
       "      <td>novel</td>\n",
       "      <td>stop_gained,splice_region_variant</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:10252</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>BS_KX3258Y6</td>\n",
       "      <td>BS_MVQYCQMP</td>\n",
       "      <td>substitution</td>\n",
       "      <td>/sbgenomics/project-files/CBTN_somatic/0046a77...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr2:g.95927379A&gt;G</td>\n",
       "      <td>ANKRD36C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1230259790</td>\n",
       "      <td>splice_donor_variant</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:32946</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>BS_KX3258Y6</td>\n",
       "      <td>BS_MVQYCQMP</td>\n",
       "      <td>SNV</td>\n",
       "      <td>/sbgenomics/project-files/CBTN_somatic/0046a77...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              HGVSg Hugo_Symbol                        HGVSp  \\\n",
       "0               chr7:g.138981269T>G    KIAA1549                      p.Met1?   \n",
       "1       chrX:g.45069977_45069978del       KDM6A           p.Ser775ArgfsTer15   \n",
       "2     chr1:g.153767777_153767786dup       INTS3           p.Val735GlyfsTer21   \n",
       "3  chr2:g.11216160_11216161delinsAA       ROCK2  p.Glu486_Glu487delinsAspTer   \n",
       "4                chr2:g.95927379A>G    ANKRD36C                          NaN   \n",
       "\n",
       "       dbSNP_RS                        Consequence SYMBOL_SOURCE     HGNC_ID  \\\n",
       "0         novel                         start_lost          HGNC  HGNC:22219   \n",
       "1         novel                 frameshift_variant          HGNC  HGNC:12637   \n",
       "2         novel                 frameshift_variant          HGNC  HGNC:26153   \n",
       "3         novel  stop_gained,splice_region_variant          HGNC  HGNC:10252   \n",
       "4  rs1230259790               splice_donor_variant          HGNC  HGNC:32946   \n",
       "\n",
       "          BIOTYPE IMPACT gnomad_3_1_1_AF gnomad_3_1_1_AF_popmax  \\\n",
       "0  protein_coding   HIGH        2.75e-05               3.05e-05   \n",
       "1  protein_coding   HIGH               .                      .   \n",
       "2  protein_coding   HIGH               .                      .   \n",
       "3  protein_coding   HIGH               .                      .   \n",
       "4  protein_coding   HIGH               .                      .   \n",
       "\n",
       "  gnomad_3_1_1_AF_non_cancer_popmax gnomad_3_1_1_AF_non_cancer_all_popmax  \\\n",
       "0                          2.45e-05                              0.000123   \n",
       "1                                 .                                     .   \n",
       "2                                 .                                     .   \n",
       "3                                 .                                     .   \n",
       "4                                 .                                     .   \n",
       "\n",
       "  Matched_Norm_Sample_Barcode Tumor_Sample_Barcode VARIANT_CLASS  \\\n",
       "0                 BS_65FNRFMM          BS_QWWYFRGZ           SNV   \n",
       "1                 BS_65FNRFMM          BS_QWWYFRGZ      deletion   \n",
       "2                 BS_KX3258Y6          BS_MVQYCQMP     insertion   \n",
       "3                 BS_KX3258Y6          BS_MVQYCQMP  substitution   \n",
       "4                 BS_KX3258Y6          BS_MVQYCQMP           SNV   \n",
       "\n",
       "                                                path  \n",
       "0  /sbgenomics/project-files/CBTN_somatic/002daa4...  \n",
       "1  /sbgenomics/project-files/CBTN_somatic/002daa4...  \n",
       "2  /sbgenomics/project-files/CBTN_somatic/0046a77...  \n",
       "3  /sbgenomics/project-files/CBTN_somatic/0046a77...  \n",
       "4  /sbgenomics/project-files/CBTN_somatic/0046a77...  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv('/home/stearb/U24/data/somatic_variants/CBTN_somatic_variants_VEP_HIGH_Aug2.csv')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "fe0137ff",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2814154/2915465253.py:1: DtypeWarning: Columns (32,34,38,40,50) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  hgnc_master = pd.read_csv('/home/stearb/U24/data/helper_files/hgnc_master.txt',sep='\\t')\n"
     ]
    }
   ],
   "source": [
    "hgnc_master = pd.read_csv('/home/stearb/U24/data/helper_files/hgnc_master.txt',sep='\\t')\n",
    "\n",
    "# merge in ensembl ids\n",
    "df = pd.merge(df,hgnc_master[['hgnc_id','ensembl_gene_id']].rename({'hgnc_id':'HGNC_ID'},axis=1))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "6aec2a2a",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2814154/3217460768.py:7: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  dfmeta['disease_code'] = [i.split(' ')[-1].replace('(','').replace(')','') for\n",
      "/tmp/ipykernel_2814154/3217460768.py:10: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  dfmeta.drop('Histological Diagnosis (MONDO)',axis=1,inplace=True)\n"
     ]
    }
   ],
   "source": [
    "# need to get patiant specific mondo codes and also biospecimen to patient id mappings\n",
    "meta = pd.read_csv('/home/stearb/U24/data/somatic_variants/kidsfirst-biospecimen-table-2025-07-29.tsv',sep='\\t')\n",
    "\n",
    "dfmeta = meta[['Participant ID','Sample ID','Histological Diagnosis (MONDO)']]\n",
    "\n",
    "\n",
    "dfmeta['disease_code'] = [i.split(' ')[-1].replace('(','').replace(')','') for \n",
    "                                           i in dfmeta['Histological Diagnosis (MONDO)']]\n",
    "\n",
    "dfmeta.drop('Histological Diagnosis (MONDO)',axis=1,inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "529cda13",
   "metadata": {},
   "outputs": [],
   "source": [
    "# merge in MONDO code\n",
    "df = pd.merge(df.rename({'Tumor_Sample_Barcode':'Sample ID'},axis=1)\n",
    "        , dfmeta, how='left', on= 'Sample ID') "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0303284d",
   "metadata": {},
   "source": [
    "# need variant type"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "142d410d-92c4-4ac3-9425-56368cfc69ef",
   "metadata": {},
   "source": [
    "# ****No ENSEMBL protein IDs for the CBTN dataset****"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "4735c81e",
   "metadata": {},
   "outputs": [],
   "source": [
    "#df = pd.merge(df,hgnc_master[['hgnc_id','ensembl_gene_id']].rename({'hgnc_id':'HGNC_ID'},axis=1))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "7dca8d4f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# variant\n",
    "df['HGVSg'] = df['HGVSg'].replace({':': '.'},regex=True) \n",
    "df['hgvsg_variant_code'] = 'HGVSG:' + df['HGVSg'] \n",
    "\n",
    "# chr1:g.26697414G>A\n",
    "#df['HGVSg'] = 'HGVSG:' df['Chromosome'] + ':g.' + df['Start_Position'] + df['Reference_Allele'] + '>' \\\n",
    "#                                    df['Tumor_Seq_Allele2']\n",
    "\n",
    "\n",
    "# ensembl gene\n",
    "df['ens_gene_code'] = 'ENSEMBL:' + df['ensembl_gene_id']\n",
    "\n",
    "# disease/MONDO code column taken care of when we merged them in above\n",
    "\n",
    "\n",
    "# cohort\n",
    "df['cohort_code'] = 'KFCOHORT:SD-BHJXBDQK'\n",
    "\n",
    "# study\n",
    "df['study_code'] = 'KFSTUDY:KF-CBTN-SOMATIC'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "fb7ac74a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subject</th>\n",
       "      <th>predicate</th>\n",
       "      <th>object</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5729</th>\n",
       "      <td>HGVSG:chr11.g.17501945G&gt;T</td>\n",
       "      <td>related_to_gene</td>\n",
       "      <td>ENSEMBL:ENSG00000006611</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4540</th>\n",
       "      <td>HGVSG:chr3.g.58831780del</td>\n",
       "      <td>related_to_gene</td>\n",
       "      <td>ENSEMBL:ENSG00000163689</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>245</th>\n",
       "      <td>HGVSG:chr1.g.205727749G&gt;A</td>\n",
       "      <td>related_to_gene</td>\n",
       "      <td>ENSEMBL:ENSG00000069275</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6891</th>\n",
       "      <td>HGVSG:chr8.g.142544240del</td>\n",
       "      <td>related_to_gene</td>\n",
       "      <td>ENSEMBL:ENSG00000181790</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4328</th>\n",
       "      <td>HGVSG:chr6.g.13486843_13486844del</td>\n",
       "      <td>related_to_gene</td>\n",
       "      <td>ENSEMBL:ENSG00000145990</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                subject        predicate  \\\n",
       "5729          HGVSG:chr11.g.17501945G>T  related_to_gene   \n",
       "4540           HGVSG:chr3.g.58831780del  related_to_gene   \n",
       "245           HGVSG:chr1.g.205727749G>A  related_to_gene   \n",
       "6891          HGVSG:chr8.g.142544240del  related_to_gene   \n",
       "4328  HGVSG:chr6.g.13486843_13486844del  related_to_gene   \n",
       "\n",
       "                       object  \n",
       "5729  ENSEMBL:ENSG00000006611  \n",
       "4540  ENSEMBL:ENSG00000163689  \n",
       "245   ENSEMBL:ENSG00000069275  \n",
       "6891  ENSEMBL:ENSG00000181790  \n",
       "4328  ENSEMBL:ENSG00000145990  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "e_var_gene = df[['hgvsg_variant_code','ens_gene_code']].dropna().drop_duplicates().reset_index(drop=True)\n",
    "e_var_gene.columns = ['subject','object']\n",
    "\n",
    "e_var_gene['predicate'] = 'related_to_gene'\n",
    "e_var_gene = e_var_gene[['subject','predicate','object']]\n",
    "\n",
    "e_var_gene.sample(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "4d1ac6ff",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subject</th>\n",
       "      <th>predicate</th>\n",
       "      <th>object</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1528</th>\n",
       "      <td>HGVSG:chr3.g.89450218del</td>\n",
       "      <td>belongs_to_cohort</td>\n",
       "      <td>KFCOHORT:SD-BHJXBDQK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6828</th>\n",
       "      <td>HGVSG:chr21.g.42732068A&gt;G</td>\n",
       "      <td>belongs_to_cohort</td>\n",
       "      <td>KFCOHORT:SD-BHJXBDQK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6782</th>\n",
       "      <td>HGVSG:chr1.g.180178825_180178835del</td>\n",
       "      <td>belongs_to_cohort</td>\n",
       "      <td>KFCOHORT:SD-BHJXBDQK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5476</th>\n",
       "      <td>HGVSG:chr6.g.22290254C&gt;A</td>\n",
       "      <td>belongs_to_cohort</td>\n",
       "      <td>KFCOHORT:SD-BHJXBDQK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2111</th>\n",
       "      <td>HGVSG:chr22.g.32498233_32498234del</td>\n",
       "      <td>belongs_to_cohort</td>\n",
       "      <td>KFCOHORT:SD-BHJXBDQK</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  subject          predicate  \\\n",
       "1528             HGVSG:chr3.g.89450218del  belongs_to_cohort   \n",
       "6828            HGVSG:chr21.g.42732068A>G  belongs_to_cohort   \n",
       "6782  HGVSG:chr1.g.180178825_180178835del  belongs_to_cohort   \n",
       "5476             HGVSG:chr6.g.22290254C>A  belongs_to_cohort   \n",
       "2111   HGVSG:chr22.g.32498233_32498234del  belongs_to_cohort   \n",
       "\n",
       "                    object  \n",
       "1528  KFCOHORT:SD-BHJXBDQK  \n",
       "6828  KFCOHORT:SD-BHJXBDQK  \n",
       "6782  KFCOHORT:SD-BHJXBDQK  \n",
       "5476  KFCOHORT:SD-BHJXBDQK  \n",
       "2111  KFCOHORT:SD-BHJXBDQK  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "e_var_cohort = df[['hgvsg_variant_code','cohort_code']].dropna().drop_duplicates().reset_index(drop=True)\n",
    "e_var_cohort.columns = ['subject','object']\n",
    "e_var_cohort['predicate'] = 'belongs_to_cohort'\n",
    "e_var_cohort = e_var_cohort[['subject','predicate','object']]\n",
    "e_var_cohort.sample(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "4d4c3021",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subject</th>\n",
       "      <th>predicate</th>\n",
       "      <th>object</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KFSTUDY:KF-CBTN-SOMATIC</td>\n",
       "      <td>study_has_cohort</td>\n",
       "      <td>KFCOHORT:SD-BHJXBDQK</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   subject         predicate                object\n",
       "0  KFSTUDY:KF-CBTN-SOMATIC  study_has_cohort  KFCOHORT:SD-BHJXBDQK"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "e_study_cohort = df[['study_code','cohort_code']].dropna().drop_duplicates().reset_index(drop=True)\n",
    "e_study_cohort.columns = ['subject','object']\n",
    "\n",
    "e_study_cohort['predicate'] = 'study_has_cohort'\n",
    "e_study_cohort = e_study_cohort[['subject','predicate','object']]\n",
    "e_study_cohort"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "9464b1e0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['chr1',\n",
       " 'chr10',\n",
       " 'chr11',\n",
       " 'chr12',\n",
       " 'chr13',\n",
       " 'chr14',\n",
       " 'chr15',\n",
       " 'chr16',\n",
       " 'chr17',\n",
       " 'chr18',\n",
       " 'chr19',\n",
       " 'chr2',\n",
       " 'chr20',\n",
       " 'chr21',\n",
       " 'chr22',\n",
       " 'chr3',\n",
       " 'chr4',\n",
       " 'chr5',\n",
       " 'chr6',\n",
       " 'chr7',\n",
       " 'chr8',\n",
       " 'chr9',\n",
       " 'chrX',\n",
       " 'chrY']"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "q = pl.scan_csv('/home/stearb/U24/data/HSCLO/OWLNETS_edgelist_HSCLO.tsv',separator='\\t')\n",
    "hsclo = q.select(['node_id','lowerbound','upperbound']).collect()\n",
    "hsclo = hsclo.to_pandas()\n",
    "\n",
    "chroms = np.unique([i.split(' ')[-1].split('.')[0] for i in hsclo.node_id.astype(str)])\n",
    "chroms = [i.tolist() for i in chroms]  \n",
    "chroms.remove('Human_Genome_hg38')\n",
    "chroms.remove('MtDNA')\n",
    "chroms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "eacd83be",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\n",
      "905 904\n",
      "chr10\n",
      "398 398\n",
      "chr11\n",
      "512 512\n",
      "chr12\n",
      "537 535\n",
      "chr13\n",
      "150 149\n",
      "chr14\n",
      "301 301\n",
      "chr15\n",
      "331 331\n",
      "chr16\n",
      "338 335\n",
      "chr17\n",
      "648 648\n",
      "chr18\n",
      "130 129\n",
      "chr19\n",
      "602 599\n",
      "chr2\n",
      "3244 3243\n",
      "chr20\n",
      "225 225\n",
      "chr21\n",
      "89 89\n",
      "chr22\n",
      "232 232\n",
      "chr3\n",
      "588 588\n",
      "chr4\n",
      "434 434\n",
      "chr5\n",
      "413 412\n",
      "chr6\n",
      "503 500\n",
      "chr7\n",
      "496 495\n",
      "chr8\n",
      "331 329\n",
      "chr9\n",
      "378 377\n",
      "chrX\n",
      "358 358\n",
      "chrY\n",
      "6 6\n"
     ]
    }
   ],
   "source": [
    "e_var_hsclo_MASTER = list()\n",
    "\n",
    "# Go by chromosome and merge each variant to the correct HSCLO term/code\n",
    "# most of this is extracting the variant position from the hgvsg_variant_code col\n",
    "# and saving it to the variant_position col. Then bin on this col.\n",
    "\n",
    "for CHROM in chroms:\n",
    "    \n",
    "    print(CHROM)\n",
    "    \n",
    "    # search for 'chr1.' or 'chr2.' if CHROM is either 1 or 2. This prevents matching on 11,12,13, when matching on 'chr1'\n",
    "    if CHROM == 'chr1': CHROM = 'chr1\\.'\n",
    "    elif CHROM == 'chr2': CHROM == 'chr2\\.'\n",
    "    \n",
    "    # Select variant col which has the variant chromosomal position for just current CHROM\n",
    "    dfChrom = df[df['hgvsg_variant_code'].str.contains(CHROM)][['hgvsg_variant_code']].dropna()\n",
    "    \n",
    "    # format to extract just numerical position \n",
    "    dfChrom['variant_position'] = \\\n",
    "          [i.split('.')[-1].replace('del','').replace('ins','')\\\n",
    "            .replace('A','').replace('C','').replace('T','').replace('G','')\\\n",
    "            .replace('>','') if 'inv' not in i or 'dup' not in i else i for i\\\n",
    "            in dfChrom['hgvsg_variant_code']]\n",
    "\n",
    "    dfChrom['variant_position'] = dfChrom['variant_position'].replace('dup','',regex=True).replace('inv','',regex=True)\n",
    "    dfChrom['variant_position'] = [i.split('_')[0] for i in dfChrom['variant_position']]\n",
    "    dfChrom['variant_position'] = dfChrom['variant_position'].astype(int)\n",
    "    dfChrom = dfChrom.drop_duplicates().reset_index(drop=True)\n",
    "    \n",
    "    # get hsclo codes for current CHROM\n",
    "    hscloChrom = hsclo[hsclo['node_id'].str.contains(CHROM)].dropna()\n",
    "\n",
    "    ###### do the binning\n",
    "    dfChromBins = bin_column(pl.DataFrame(dfChrom),col2bin='variant_position').to_pandas()\n",
    "\n",
    "    # make lowerbound col to merge in hsclo codes\n",
    "    dfChromBins['lowerbound'] = [float(i.split(',')[0][1:]) for i in dfChromBins['bins']]\n",
    "\n",
    "    # merge in hsclo codes \n",
    "    dfMerged = pd.merge(dfChromBins,hscloChrom,on='lowerbound',how='left')\n",
    "    \n",
    "    # this may introduce an error, if im matching on lowerbound only, as there are multiple resolution\n",
    "    # levels, so upperbound must also be specified!\n",
    "    # ...check that the variant_position is b/t lower and upperbound\n",
    "    print(len(dfMerged) ,len(dfMerged[(dfMerged['variant_position'] > dfMerged['lowerbound']) &\\\n",
    "        (dfMerged['variant_position'] < dfMerged['upperbound'])]))\n",
    "    \n",
    "    #assert dfMerged.shape == dfMerged[(dfMerged['variant_position'] > dfMerged['lowerbound']) &\\\n",
    "    #    (dfMerged['variant_position'] < dfMerged['upperbound'])].shape\n",
    "    \n",
    "    e_var_hsclo = dfMerged[['hgvsg_variant_code','node_id']].drop_duplicates().dropna().reset_index(drop=True)\n",
    "    e_var_hsclo.columns = ['subject','object']\n",
    "    e_var_hsclo['predicate'] = 'has_location'\n",
    "    e_var_hsclo = e_var_hsclo[['subject','predicate','object']]\n",
    "    \n",
    "    # e_var_hsclo is longer than dfChrom bc each variant is mapped to multiple HSCLO resolution levels\n",
    "    #print(len(e_var_hsclo),len(dfChrom)) \n",
    "    \n",
    "    if len(e_var_hsclo_MASTER) == 0:\n",
    "        e_var_hsclo_MASTER = e_var_hsclo\n",
    "    else:\n",
    "        e_var_hsclo_MASTER = pd.concat([e_var_hsclo_MASTER,e_var_hsclo])\n",
    "       \n",
    "e_var_hsclo_MASTER = e_var_hsclo_MASTER.drop_duplicates().dropna().reset_index(drop=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "7a1fe3b1",
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir /mnt/isilon/opentargets/U24KG/data/owlnets_nodes_and_edges/CBTN_somatic"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "7f04d796",
   "metadata": {},
   "outputs": [],
   "source": [
    "edges_all = pd.concat([e_var_gene,e_var_cohort,e_study_cohort,\n",
    "                       e_var_hsclo_MASTER])   # e_var_prot,e_var_trans, e_trans_prot\n",
    "\n",
    "edges_all = edges_all.drop_duplicates().dropna().reset_index(drop=True)\n",
    "\n",
    "edges_all.to_csv('/mnt/isilon/opentargets/U24KG/data/owlnets_nodes_and_edges/CBTN_somatic/OWLNETS_edgelist.txt',\n",
    "             sep= \"\\t\",index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "7fd3a567",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>node_id</th>\n",
       "      <th>value</th>\n",
       "      <th>lowerbound</th>\n",
       "      <th>node_label</th>\n",
       "      <th>node_dbxrefs</th>\n",
       "      <th>node_synonyms</th>\n",
       "      <th>unit</th>\n",
       "      <th>node_definition</th>\n",
       "      <th>upperbound</th>\n",
       "      <th>node_namespace</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HGVSG:chr7.g.138981269T&gt;G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HGVSG:chrX.g.45069977_45069978del</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HGVSG:chr1.g.153767777_153767786dup</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HGVSG:chr2.g.11216160_11216161delinsAA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HGVSG:chr2.g.95927379A&gt;G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14678</th>\n",
       "      <td>ENSEMBL:ENSG00000061987</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14679</th>\n",
       "      <td>ENSEMBL:ENSG00000165192</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14680</th>\n",
       "      <td>ENSEMBL:ENSG00000280725</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14681</th>\n",
       "      <td>ENSEMBL:ENSG00000132394</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14682</th>\n",
       "      <td>KFCOHORT:SD-BHJXBDQK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>14683 rows Ã— 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      node_id  value  lowerbound  node_label  \\\n",
       "0                   HGVSG:chr7.g.138981269T>G    NaN         NaN         NaN   \n",
       "1           HGVSG:chrX.g.45069977_45069978del    NaN         NaN         NaN   \n",
       "2         HGVSG:chr1.g.153767777_153767786dup    NaN         NaN         NaN   \n",
       "3      HGVSG:chr2.g.11216160_11216161delinsAA    NaN         NaN         NaN   \n",
       "4                    HGVSG:chr2.g.95927379A>G    NaN         NaN         NaN   \n",
       "...                                       ...    ...         ...         ...   \n",
       "14678                 ENSEMBL:ENSG00000061987    NaN         NaN         NaN   \n",
       "14679                 ENSEMBL:ENSG00000165192    NaN         NaN         NaN   \n",
       "14680                 ENSEMBL:ENSG00000280725    NaN         NaN         NaN   \n",
       "14681                 ENSEMBL:ENSG00000132394    NaN         NaN         NaN   \n",
       "14682                    KFCOHORT:SD-BHJXBDQK    NaN         NaN         NaN   \n",
       "\n",
       "       node_dbxrefs  node_synonyms  unit  node_definition  upperbound  \\\n",
       "0               NaN            NaN   NaN              NaN         NaN   \n",
       "1               NaN            NaN   NaN              NaN         NaN   \n",
       "2               NaN            NaN   NaN              NaN         NaN   \n",
       "3               NaN            NaN   NaN              NaN         NaN   \n",
       "4               NaN            NaN   NaN              NaN         NaN   \n",
       "...             ...            ...   ...              ...         ...   \n",
       "14678           NaN            NaN   NaN              NaN         NaN   \n",
       "14679           NaN            NaN   NaN              NaN         NaN   \n",
       "14680           NaN            NaN   NaN              NaN         NaN   \n",
       "14681           NaN            NaN   NaN              NaN         NaN   \n",
       "14682           NaN            NaN   NaN              NaN         NaN   \n",
       "\n",
       "       node_namespace  \n",
       "0                 NaN  \n",
       "1                 NaN  \n",
       "2                 NaN  \n",
       "3                 NaN  \n",
       "4                 NaN  \n",
       "...               ...  \n",
       "14678             NaN  \n",
       "14679             NaN  \n",
       "14680             NaN  \n",
       "14681             NaN  \n",
       "14682             NaN  \n",
       "\n",
       "[14683 rows x 10 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nodes_all = pd.concat([edges_all['subject'],edges_all['object']])\\\n",
    "                          .drop_duplicates().dropna().reset_index(drop=True)\n",
    "\n",
    "nodes_all = pd.DataFrame(nodes_all,columns=['node_id'])\n",
    "nodes_all = fill_missing_cols(nodes_all)\n",
    "\n",
    "# NO Need to save HSCLO node ids\n",
    "nodes_all = nodes_all[~nodes_all['node_id'].str.startswith('HSCLO')]\\\n",
    "    .drop_duplicates().dropna(subset=['node_id']).reset_index(drop=True)\n",
    "\n",
    "nodes_all"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "18cff689-60d7-4932-8082-42b7775ffe6e",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "nodes_all.to_csv(\n",
    "    '/mnt/isilon/opentargets/U24KG/data/owlnets_nodes_and_edges/CBTN_somatic/OWLNETS_node_metadata.txt',\n",
    "             sep= \"\\t\",index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0c8f8137-d83f-4085-a9c3-fcde228487d1",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
