{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "35cb6ca8",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import polars as pl\n",
    "import numpy as np\n",
    "import re"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e8521a1e",
   "metadata": {},
   "source": [
    "# Proces the neuroblastoma somatic variants. \n",
    "They were pulled from VWB on June 24th 2025. These are just the VEP=HIGH impact variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "080aa506",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "def fill_missing_cols(df):\n",
    "    if 'node_id' not in df.columns:\n",
    "        raise ValueError('Must have at least a \"node_id\" column.')\n",
    "    \n",
    "    all_cols = set([ 'node_label', 'node_synonyms', 'node_dbxrefs',\n",
    "            'node_definition','node_namespace','value','lowerbound','upperbound','unit'])\n",
    "    missing_cols = list(all_cols - set(df.columns))\n",
    "    nan_cols_df = pd.DataFrame(np.full([len(df), len(missing_cols)], np.nan),columns=missing_cols)\n",
    "\n",
    "    if isinstance(df, pd.DataFrame):\n",
    "        nan_cols_df.index = df.index\n",
    "        return pd.concat([df,nan_cols_df],axis=1)\n",
    "    elif isinstance(df, pl.DataFrame):\n",
    "        return pl.concat([df,pl.from_pandas(nan_cols_df)],how='horizontal') # no index for polars\n",
    "    else:\n",
    "        raise ValueError(f'Must Pass either a pandas DataFrame or a polars DataFrame but recieved \"{type(df)}\".')\n",
    "    \n",
    "\n",
    "def bin_column(df, col2bin=''):\n",
    "    '''works with polars df'''\n",
    "    \n",
    "    '''tpm_bins = list([0.0000000,7e-4,8e-4,9e-4]) + list(np.linspace(1e-3,9e-3,9)) + \\\n",
    "               list(np.round(np.linspace(1e-2,9e-2,9),2)) + list(np.round(np.linspace(.1,1,10),2)) + \\\n",
    "               list(np.linspace(2,100,99)) + list(np.arange(100,1100,100)[1:]) +  \\\n",
    "                list(np.arange(2000,11000,1000)) + list(np.arange(20_000,110_000,10_000)) + [300_000] '''\n",
    "    \n",
    "    bins = np.unique(hscloChrom['lowerbound'].values) #hscloChrom['lowerbound'].values\n",
    "    # check that none of the disease expression values are out of range, ie larger than the max bin value\n",
    "    # use for pandas\n",
    "    #if np.sum(df['Disease_MeanTpm'] > np.max(tpm_bins)) == 0:\n",
    "        \n",
    "    # use for polars\n",
    "    if len(df.filter(df[col2bin] > np.max(bins))) == 0:\n",
    "        df_binned = df.with_columns(  df[col2bin].cut(breaks=bins).alias('bins')  )\n",
    "        # check that order has been maintained #assert pl.all(df[col2bin] == df_binned[col2bin])\n",
    "    else:\n",
    "        print('OUT OF RANGE ERROR!')\n",
    "        assert False \n",
    "    return df_binned"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "38c66bb0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>participant_id</th>\n",
       "      <th>variant_class</th>\n",
       "      <th>zygosity</th>\n",
       "      <th>study_code</th>\n",
       "      <th>ensembl_transcript_id</th>\n",
       "      <th>ensembl_gene_id</th>\n",
       "      <th>symbol</th>\n",
       "      <th>feature_type</th>\n",
       "      <th>hgvsp</th>\n",
       "      <th>aa_change</th>\n",
       "      <th>consequence</th>\n",
       "      <th>symbol.1</th>\n",
       "      <th>vep_impact</th>\n",
       "      <th>rsnumber</th>\n",
       "      <th>hgvsg</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BS_SY6CCX4G</td>\n",
       "      <td>PT_3YW2V4JK</td>\n",
       "      <td>deletion</td>\n",
       "      <td>HET</td>\n",
       "      <td>KF-NBL</td>\n",
       "      <td>ENST00000428771</td>\n",
       "      <td>ENSG00000188290</td>\n",
       "      <td>HES4</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENSP00000393198.2:p.Arg50GlyfsTer60</td>\n",
       "      <td>p.Gly49/247Xaa</td>\n",
       "      <td>['frameshift']</td>\n",
       "      <td>HES4</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1:g.999832del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BS_SY6CCX4G</td>\n",
       "      <td>PT_3YW2V4JK</td>\n",
       "      <td>deletion</td>\n",
       "      <td>HET</td>\n",
       "      <td>KF-NBL</td>\n",
       "      <td>NM_001142467.2</td>\n",
       "      <td>57801</td>\n",
       "      <td>HES4</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>NP_001135939.1:p.Arg50GlyfsTer60</td>\n",
       "      <td>p.Gly49/247Xaa</td>\n",
       "      <td>['frameshift']</td>\n",
       "      <td>HES4</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1:g.999832del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BS_8313ZZ6E</td>\n",
       "      <td>PT_3YW2V4JK</td>\n",
       "      <td>deletion</td>\n",
       "      <td>WT</td>\n",
       "      <td>KF-NBL</td>\n",
       "      <td>ENST00000428771</td>\n",
       "      <td>ENSG00000188290</td>\n",
       "      <td>HES4</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENSP00000393198.2:p.Arg50GlyfsTer60</td>\n",
       "      <td>p.Gly49/247Xaa</td>\n",
       "      <td>['frameshift']</td>\n",
       "      <td>HES4</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1:g.999832del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BS_8313ZZ6E</td>\n",
       "      <td>PT_3YW2V4JK</td>\n",
       "      <td>deletion</td>\n",
       "      <td>WT</td>\n",
       "      <td>KF-NBL</td>\n",
       "      <td>NM_001142467.2</td>\n",
       "      <td>57801</td>\n",
       "      <td>HES4</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>NP_001135939.1:p.Arg50GlyfsTer60</td>\n",
       "      <td>p.Gly49/247Xaa</td>\n",
       "      <td>['frameshift']</td>\n",
       "      <td>HES4</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1:g.999832del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BS_Z40644EA</td>\n",
       "      <td>PT_W50HSNE2</td>\n",
       "      <td>SNV</td>\n",
       "      <td>WT</td>\n",
       "      <td>KF-NBL</td>\n",
       "      <td>XM_017001945.1</td>\n",
       "      <td>5792</td>\n",
       "      <td>PTPRF</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>XP_016857434.1:p.Ser531Ter</td>\n",
       "      <td>p.Ser531/1807Ter</td>\n",
       "      <td>['stop gained']</td>\n",
       "      <td>PTPRF</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1:g.43591872C&gt;A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     sample_id participant_id variant_class zygosity study_code  \\\n",
       "0  BS_SY6CCX4G    PT_3YW2V4JK      deletion      HET     KF-NBL   \n",
       "1  BS_SY6CCX4G    PT_3YW2V4JK      deletion      HET     KF-NBL   \n",
       "2  BS_8313ZZ6E    PT_3YW2V4JK      deletion       WT     KF-NBL   \n",
       "3  BS_8313ZZ6E    PT_3YW2V4JK      deletion       WT     KF-NBL   \n",
       "4  BS_Z40644EA    PT_W50HSNE2           SNV       WT     KF-NBL   \n",
       "\n",
       "  ensembl_transcript_id  ensembl_gene_id symbol feature_type  \\\n",
       "0       ENST00000428771  ENSG00000188290   HES4   Transcript   \n",
       "1        NM_001142467.2            57801   HES4   Transcript   \n",
       "2       ENST00000428771  ENSG00000188290   HES4   Transcript   \n",
       "3        NM_001142467.2            57801   HES4   Transcript   \n",
       "4        XM_017001945.1             5792  PTPRF   Transcript   \n",
       "\n",
       "                                 hgvsp         aa_change      consequence  \\\n",
       "0  ENSP00000393198.2:p.Arg50GlyfsTer60    p.Gly49/247Xaa   ['frameshift']   \n",
       "1     NP_001135939.1:p.Arg50GlyfsTer60    p.Gly49/247Xaa   ['frameshift']   \n",
       "2  ENSP00000393198.2:p.Arg50GlyfsTer60    p.Gly49/247Xaa   ['frameshift']   \n",
       "3     NP_001135939.1:p.Arg50GlyfsTer60    p.Gly49/247Xaa   ['frameshift']   \n",
       "4           XP_016857434.1:p.Ser531Ter  p.Ser531/1807Ter  ['stop gained']   \n",
       "\n",
       "  symbol.1 vep_impact rsnumber               hgvsg  \n",
       "0     HES4       HIGH      NaN    chr1:g.999832del  \n",
       "1     HES4       HIGH      NaN    chr1:g.999832del  \n",
       "2     HES4       HIGH      NaN    chr1:g.999832del  \n",
       "3     HES4       HIGH      NaN    chr1:g.999832del  \n",
       "4    PTPRF       HIGH      NaN  chr1:g.43591872C>A  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df=pd.read_csv('/home/stearb/U24/data/somatic_variants/NBL_somatic_variants_VEP_HIGH_June24.csv')\n",
    "df.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "b3ded8e8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "hgvsg    1609\n",
       "dtype: int64"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[['hgvsg']].nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "3201a76b",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1628083/722391764.py:5: DtypeWarning: Columns (32,34,38,40,50) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  hgnc_master = pd.read_csv('/home/stearb/U24/data/helper_files/hgnc_master.txt',\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>aa_change</th>\n",
       "      <th>consequence</th>\n",
       "      <th>ensembl_gene_id</th>\n",
       "      <th>ensembl_transcript_id</th>\n",
       "      <th>feature_type</th>\n",
       "      <th>hgvsg</th>\n",
       "      <th>hgvsp</th>\n",
       "      <th>participant_id</th>\n",
       "      <th>rsnumber</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>study_code</th>\n",
       "      <th>symbol</th>\n",
       "      <th>symbol.1</th>\n",
       "      <th>variant_class</th>\n",
       "      <th>vep_impact</th>\n",
       "      <th>zygosity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>p.Gly49/247Xaa</td>\n",
       "      <td>['frameshift']</td>\n",
       "      <td>ENSG00000188290</td>\n",
       "      <td>ENST00000428771</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>chr1:g.999832del</td>\n",
       "      <td>ENSP00000393198.2:p.Arg50GlyfsTer60</td>\n",
       "      <td>PT_3YW2V4JK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BS_SY6CCX4G</td>\n",
       "      <td>KF-NBL</td>\n",
       "      <td>HES4</td>\n",
       "      <td>HES4</td>\n",
       "      <td>deletion</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>HET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>p.Gly49/247Xaa</td>\n",
       "      <td>['frameshift']</td>\n",
       "      <td>ENSG00000188290</td>\n",
       "      <td>ENST00000428771</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>chr1:g.999832del</td>\n",
       "      <td>ENSP00000393198.2:p.Arg50GlyfsTer60</td>\n",
       "      <td>PT_3YW2V4JK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BS_8313ZZ6E</td>\n",
       "      <td>KF-NBL</td>\n",
       "      <td>HES4</td>\n",
       "      <td>HES4</td>\n",
       "      <td>deletion</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>WT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>p.Ser199/1291Ter</td>\n",
       "      <td>['stop gained']</td>\n",
       "      <td>ENSG00000142949</td>\n",
       "      <td>ENST00000412568</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>chr1:g.43591872C&gt;A</td>\n",
       "      <td>ENSP00000391764.2:p.Ser199Ter</td>\n",
       "      <td>PT_W50HSNE2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BS_Z40644EA</td>\n",
       "      <td>KF-NBL</td>\n",
       "      <td>PTPRF</td>\n",
       "      <td>PTPRF</td>\n",
       "      <td>SNV</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>WT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>p.Ser188/1553Ter</td>\n",
       "      <td>['stop gained']</td>\n",
       "      <td>ENSG00000142949</td>\n",
       "      <td>ENST00000429895</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>chr1:g.43591872C&gt;A</td>\n",
       "      <td>ENSP00000408952.1:p.Ser188Ter</td>\n",
       "      <td>PT_W50HSNE2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BS_Z40644EA</td>\n",
       "      <td>KF-NBL</td>\n",
       "      <td>PTPRF</td>\n",
       "      <td>PTPRF</td>\n",
       "      <td>SNV</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>WT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>p.Ser531/1907Ter</td>\n",
       "      <td>['stop gained']</td>\n",
       "      <td>ENSG00000142949</td>\n",
       "      <td>ENST00000359947</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>chr1:g.43591872C&gt;A</td>\n",
       "      <td>ENSP00000353030.4:p.Ser531Ter</td>\n",
       "      <td>PT_W50HSNE2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BS_Z40644EA</td>\n",
       "      <td>KF-NBL</td>\n",
       "      <td>PTPRF</td>\n",
       "      <td>PTPRF</td>\n",
       "      <td>SNV</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>WT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18223</th>\n",
       "      <td>p.Ser200/682Ter</td>\n",
       "      <td>['stop gained']</td>\n",
       "      <td>ENSG00000173681</td>\n",
       "      <td>XM_011545477.2</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>chrX:g.19966092G&gt;T</td>\n",
       "      <td>XP_011543779.1:p.Ser200Ter</td>\n",
       "      <td>PT_NBFP0BB7</td>\n",
       "      <td>rs1230094791</td>\n",
       "      <td>BS_54N90X5G</td>\n",
       "      <td>KF-NBL</td>\n",
       "      <td>BCLAF3</td>\n",
       "      <td>BCLAF3</td>\n",
       "      <td>SNV</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>WT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18224</th>\n",
       "      <td>p.Ser200/711Ter</td>\n",
       "      <td>['stop gained']</td>\n",
       "      <td>ENSG00000173681</td>\n",
       "      <td>NM_001367774.2</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>chrX:g.19966092G&gt;T</td>\n",
       "      <td>NP_001354703.1:p.Ser200Ter</td>\n",
       "      <td>PT_NBFP0BB7</td>\n",
       "      <td>rs1230094791</td>\n",
       "      <td>BS_54N90X5G</td>\n",
       "      <td>KF-NBL</td>\n",
       "      <td>BCLAF3</td>\n",
       "      <td>BCLAF3</td>\n",
       "      <td>SNV</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>WT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18225</th>\n",
       "      <td>p.Ser200/652Ter</td>\n",
       "      <td>['stop gained']</td>\n",
       "      <td>ENSG00000173681</td>\n",
       "      <td>XM_011545479.2</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>chrX:g.19966092G&gt;T</td>\n",
       "      <td>XP_011543781.1:p.Ser200Ter</td>\n",
       "      <td>PT_NBFP0BB7</td>\n",
       "      <td>rs1230094791</td>\n",
       "      <td>BS_54N90X5G</td>\n",
       "      <td>KF-NBL</td>\n",
       "      <td>BCLAF3</td>\n",
       "      <td>BCLAF3</td>\n",
       "      <td>SNV</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>WT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18226</th>\n",
       "      <td>p.Ser200/681Ter</td>\n",
       "      <td>['stop gained']</td>\n",
       "      <td>ENSG00000173681</td>\n",
       "      <td>XM_005274475.3</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>chrX:g.19966092G&gt;T</td>\n",
       "      <td>XP_005274532.1:p.Ser200Ter</td>\n",
       "      <td>PT_NBFP0BB7</td>\n",
       "      <td>rs1230094791</td>\n",
       "      <td>BS_54N90X5G</td>\n",
       "      <td>KF-NBL</td>\n",
       "      <td>BCLAF3</td>\n",
       "      <td>BCLAF3</td>\n",
       "      <td>SNV</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>WT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18227</th>\n",
       "      <td>p.Ser200/691Ter</td>\n",
       "      <td>['stop gained']</td>\n",
       "      <td>ENSG00000173681</td>\n",
       "      <td>XM_005274474.3</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>chrX:g.19966092G&gt;T</td>\n",
       "      <td>XP_005274531.1:p.Ser200Ter</td>\n",
       "      <td>PT_NBFP0BB7</td>\n",
       "      <td>rs1230094791</td>\n",
       "      <td>BS_54N90X5G</td>\n",
       "      <td>KF-NBL</td>\n",
       "      <td>BCLAF3</td>\n",
       "      <td>BCLAF3</td>\n",
       "      <td>SNV</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>WT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>28844 rows × 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              aa_change      consequence  ensembl_gene_id  \\\n",
       "0        p.Gly49/247Xaa   ['frameshift']  ENSG00000188290   \n",
       "2        p.Gly49/247Xaa   ['frameshift']  ENSG00000188290   \n",
       "9      p.Ser199/1291Ter  ['stop gained']  ENSG00000142949   \n",
       "15     p.Ser188/1553Ter  ['stop gained']  ENSG00000142949   \n",
       "25     p.Ser531/1907Ter  ['stop gained']  ENSG00000142949   \n",
       "...                 ...              ...              ...   \n",
       "18223   p.Ser200/682Ter  ['stop gained']  ENSG00000173681   \n",
       "18224   p.Ser200/711Ter  ['stop gained']  ENSG00000173681   \n",
       "18225   p.Ser200/652Ter  ['stop gained']  ENSG00000173681   \n",
       "18226   p.Ser200/681Ter  ['stop gained']  ENSG00000173681   \n",
       "18227   p.Ser200/691Ter  ['stop gained']  ENSG00000173681   \n",
       "\n",
       "      ensembl_transcript_id feature_type               hgvsg  \\\n",
       "0           ENST00000428771   Transcript    chr1:g.999832del   \n",
       "2           ENST00000428771   Transcript    chr1:g.999832del   \n",
       "9           ENST00000412568   Transcript  chr1:g.43591872C>A   \n",
       "15          ENST00000429895   Transcript  chr1:g.43591872C>A   \n",
       "25          ENST00000359947   Transcript  chr1:g.43591872C>A   \n",
       "...                     ...          ...                 ...   \n",
       "18223        XM_011545477.2   Transcript  chrX:g.19966092G>T   \n",
       "18224        NM_001367774.2   Transcript  chrX:g.19966092G>T   \n",
       "18225        XM_011545479.2   Transcript  chrX:g.19966092G>T   \n",
       "18226        XM_005274475.3   Transcript  chrX:g.19966092G>T   \n",
       "18227        XM_005274474.3   Transcript  chrX:g.19966092G>T   \n",
       "\n",
       "                                     hgvsp participant_id      rsnumber  \\\n",
       "0      ENSP00000393198.2:p.Arg50GlyfsTer60    PT_3YW2V4JK           NaN   \n",
       "2      ENSP00000393198.2:p.Arg50GlyfsTer60    PT_3YW2V4JK           NaN   \n",
       "9            ENSP00000391764.2:p.Ser199Ter    PT_W50HSNE2           NaN   \n",
       "15           ENSP00000408952.1:p.Ser188Ter    PT_W50HSNE2           NaN   \n",
       "25           ENSP00000353030.4:p.Ser531Ter    PT_W50HSNE2           NaN   \n",
       "...                                    ...            ...           ...   \n",
       "18223           XP_011543779.1:p.Ser200Ter    PT_NBFP0BB7  rs1230094791   \n",
       "18224           NP_001354703.1:p.Ser200Ter    PT_NBFP0BB7  rs1230094791   \n",
       "18225           XP_011543781.1:p.Ser200Ter    PT_NBFP0BB7  rs1230094791   \n",
       "18226           XP_005274532.1:p.Ser200Ter    PT_NBFP0BB7  rs1230094791   \n",
       "18227           XP_005274531.1:p.Ser200Ter    PT_NBFP0BB7  rs1230094791   \n",
       "\n",
       "         sample_id study_code  symbol symbol.1 variant_class vep_impact  \\\n",
       "0      BS_SY6CCX4G     KF-NBL    HES4     HES4      deletion       HIGH   \n",
       "2      BS_8313ZZ6E     KF-NBL    HES4     HES4      deletion       HIGH   \n",
       "9      BS_Z40644EA     KF-NBL   PTPRF    PTPRF           SNV       HIGH   \n",
       "15     BS_Z40644EA     KF-NBL   PTPRF    PTPRF           SNV       HIGH   \n",
       "25     BS_Z40644EA     KF-NBL   PTPRF    PTPRF           SNV       HIGH   \n",
       "...            ...        ...     ...      ...           ...        ...   \n",
       "18223  BS_54N90X5G     KF-NBL  BCLAF3   BCLAF3           SNV       HIGH   \n",
       "18224  BS_54N90X5G     KF-NBL  BCLAF3   BCLAF3           SNV       HIGH   \n",
       "18225  BS_54N90X5G     KF-NBL  BCLAF3   BCLAF3           SNV       HIGH   \n",
       "18226  BS_54N90X5G     KF-NBL  BCLAF3   BCLAF3           SNV       HIGH   \n",
       "18227  BS_54N90X5G     KF-NBL  BCLAF3   BCLAF3           SNV       HIGH   \n",
       "\n",
       "      zygosity  \n",
       "0          HET  \n",
       "2           WT  \n",
       "9           WT  \n",
       "15          WT  \n",
       "25          WT  \n",
       "...        ...  \n",
       "18223       WT  \n",
       "18224       WT  \n",
       "18225       WT  \n",
       "18226       WT  \n",
       "18227       WT  \n",
       "\n",
       "[28844 rows x 16 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# merge in ensembl gene ids for the rows where the ensembl_gene_id col actually has an entrez id\n",
    "df_ensembl = df[df['ensembl_gene_id'].str.startswith('ENSG')]\n",
    "df_entrez  = df[~df['ensembl_gene_id'].str.startswith('ENSG')]\n",
    "\n",
    "hgnc_master = pd.read_csv('/home/stearb/U24/data/helper_files/hgnc_master.txt',\n",
    "                          sep='\\t')\n",
    "hgnc_master.head()\n",
    "ensembl_entrez_map = hgnc_master[['ensembl_gene_id','entrez_id']].dropna()\n",
    "ensembl_entrez_map['entrez_id'] = ensembl_entrez_map['entrez_id'].astype(int).astype(str)\n",
    "\n",
    "df_entrez_mapped = pd.merge(df_entrez.rename({'ensembl_gene_id':'entrez_id'},axis=1)\n",
    "                           ,ensembl_entrez_map,on='entrez_id')\\\n",
    "                            .drop('entrez_id',axis=1)\n",
    "\n",
    "# add them back together, sort columns first though they got out of order during merge\n",
    "df = pd.concat([df_ensembl[df_ensembl.columns.sort_values()],\n",
    "                df_entrez_mapped[df_entrez_mapped.columns.sort_values()]])\n",
    "\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "77820baf",
   "metadata": {},
   "source": [
    "# Define codes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "1a665c34",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>aa_change</th>\n",
       "      <th>consequence</th>\n",
       "      <th>ensembl_gene_id</th>\n",
       "      <th>ensembl_transcript_id</th>\n",
       "      <th>feature_type</th>\n",
       "      <th>hgvsg</th>\n",
       "      <th>hgvsp</th>\n",
       "      <th>participant_id</th>\n",
       "      <th>rsnumber</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>...</th>\n",
       "      <th>symbol.1</th>\n",
       "      <th>variant_class</th>\n",
       "      <th>vep_impact</th>\n",
       "      <th>zygosity</th>\n",
       "      <th>hgvsg_variant_code</th>\n",
       "      <th>ens_gene_code</th>\n",
       "      <th>ens_transcript_code</th>\n",
       "      <th>ens_protein_code</th>\n",
       "      <th>disease_code</th>\n",
       "      <th>cohort_code</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>p.Gly49/247Xaa</td>\n",
       "      <td>['frameshift']</td>\n",
       "      <td>ENSG00000188290</td>\n",
       "      <td>ENST00000428771</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>chr1.g.999832del</td>\n",
       "      <td>ENSP00000393198.2:p.Arg50GlyfsTer60</td>\n",
       "      <td>PT_3YW2V4JK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BS_SY6CCX4G</td>\n",
       "      <td>...</td>\n",
       "      <td>HES4</td>\n",
       "      <td>deletion</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>HET</td>\n",
       "      <td>HGVSG:chr1.g.999832del</td>\n",
       "      <td>ENSEMBL:ENSG00000188290</td>\n",
       "      <td>ENSEMBL:ENST00000428771</td>\n",
       "      <td>ENSEMBL:ENST00000428771</td>\n",
       "      <td>MONDO:0005072</td>\n",
       "      <td>KFCOHORT:SD-DYPMEHHF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>p.Gly49/247Xaa</td>\n",
       "      <td>['frameshift']</td>\n",
       "      <td>ENSG00000188290</td>\n",
       "      <td>ENST00000428771</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>chr1.g.999832del</td>\n",
       "      <td>ENSP00000393198.2:p.Arg50GlyfsTer60</td>\n",
       "      <td>PT_3YW2V4JK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BS_8313ZZ6E</td>\n",
       "      <td>...</td>\n",
       "      <td>HES4</td>\n",
       "      <td>deletion</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>WT</td>\n",
       "      <td>HGVSG:chr1.g.999832del</td>\n",
       "      <td>ENSEMBL:ENSG00000188290</td>\n",
       "      <td>ENSEMBL:ENST00000428771</td>\n",
       "      <td>ENSEMBL:ENST00000428771</td>\n",
       "      <td>MONDO:0005072</td>\n",
       "      <td>KFCOHORT:SD-DYPMEHHF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>p.Ser199/1291Ter</td>\n",
       "      <td>['stop gained']</td>\n",
       "      <td>ENSG00000142949</td>\n",
       "      <td>ENST00000412568</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>chr1.g.43591872C&gt;A</td>\n",
       "      <td>ENSP00000391764.2:p.Ser199Ter</td>\n",
       "      <td>PT_W50HSNE2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BS_Z40644EA</td>\n",
       "      <td>...</td>\n",
       "      <td>PTPRF</td>\n",
       "      <td>SNV</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>WT</td>\n",
       "      <td>HGVSG:chr1.g.43591872C&gt;A</td>\n",
       "      <td>ENSEMBL:ENSG00000142949</td>\n",
       "      <td>ENSEMBL:ENST00000412568</td>\n",
       "      <td>ENSEMBL:ENST00000412568</td>\n",
       "      <td>MONDO:0005072</td>\n",
       "      <td>KFCOHORT:SD-DYPMEHHF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>p.Ser188/1553Ter</td>\n",
       "      <td>['stop gained']</td>\n",
       "      <td>ENSG00000142949</td>\n",
       "      <td>ENST00000429895</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>chr1.g.43591872C&gt;A</td>\n",
       "      <td>ENSP00000408952.1:p.Ser188Ter</td>\n",
       "      <td>PT_W50HSNE2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BS_Z40644EA</td>\n",
       "      <td>...</td>\n",
       "      <td>PTPRF</td>\n",
       "      <td>SNV</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>WT</td>\n",
       "      <td>HGVSG:chr1.g.43591872C&gt;A</td>\n",
       "      <td>ENSEMBL:ENSG00000142949</td>\n",
       "      <td>ENSEMBL:ENST00000429895</td>\n",
       "      <td>ENSEMBL:ENST00000429895</td>\n",
       "      <td>MONDO:0005072</td>\n",
       "      <td>KFCOHORT:SD-DYPMEHHF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>p.Ser531/1907Ter</td>\n",
       "      <td>['stop gained']</td>\n",
       "      <td>ENSG00000142949</td>\n",
       "      <td>ENST00000359947</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>chr1.g.43591872C&gt;A</td>\n",
       "      <td>ENSP00000353030.4:p.Ser531Ter</td>\n",
       "      <td>PT_W50HSNE2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BS_Z40644EA</td>\n",
       "      <td>...</td>\n",
       "      <td>PTPRF</td>\n",
       "      <td>SNV</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>WT</td>\n",
       "      <td>HGVSG:chr1.g.43591872C&gt;A</td>\n",
       "      <td>ENSEMBL:ENSG00000142949</td>\n",
       "      <td>ENSEMBL:ENST00000359947</td>\n",
       "      <td>ENSEMBL:ENST00000359947</td>\n",
       "      <td>MONDO:0005072</td>\n",
       "      <td>KFCOHORT:SD-DYPMEHHF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18223</th>\n",
       "      <td>p.Ser200/682Ter</td>\n",
       "      <td>['stop gained']</td>\n",
       "      <td>ENSG00000173681</td>\n",
       "      <td>XM_011545477.2</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>chrX.g.19966092G&gt;T</td>\n",
       "      <td>XP_011543779.1:p.Ser200Ter</td>\n",
       "      <td>PT_NBFP0BB7</td>\n",
       "      <td>rs1230094791</td>\n",
       "      <td>BS_54N90X5G</td>\n",
       "      <td>...</td>\n",
       "      <td>BCLAF3</td>\n",
       "      <td>SNV</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>WT</td>\n",
       "      <td>HGVSG:chrX.g.19966092G&gt;T</td>\n",
       "      <td>ENSEMBL:ENSG00000173681</td>\n",
       "      <td>ENSEMBL:XM_011545477.2</td>\n",
       "      <td>ENSEMBL:XM_011545477.2</td>\n",
       "      <td>MONDO:0005072</td>\n",
       "      <td>KFCOHORT:SD-DYPMEHHF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18224</th>\n",
       "      <td>p.Ser200/711Ter</td>\n",
       "      <td>['stop gained']</td>\n",
       "      <td>ENSG00000173681</td>\n",
       "      <td>NM_001367774.2</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>chrX.g.19966092G&gt;T</td>\n",
       "      <td>NP_001354703.1:p.Ser200Ter</td>\n",
       "      <td>PT_NBFP0BB7</td>\n",
       "      <td>rs1230094791</td>\n",
       "      <td>BS_54N90X5G</td>\n",
       "      <td>...</td>\n",
       "      <td>BCLAF3</td>\n",
       "      <td>SNV</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>WT</td>\n",
       "      <td>HGVSG:chrX.g.19966092G&gt;T</td>\n",
       "      <td>ENSEMBL:ENSG00000173681</td>\n",
       "      <td>ENSEMBL:NM_001367774.2</td>\n",
       "      <td>ENSEMBL:NM_001367774.2</td>\n",
       "      <td>MONDO:0005072</td>\n",
       "      <td>KFCOHORT:SD-DYPMEHHF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18225</th>\n",
       "      <td>p.Ser200/652Ter</td>\n",
       "      <td>['stop gained']</td>\n",
       "      <td>ENSG00000173681</td>\n",
       "      <td>XM_011545479.2</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>chrX.g.19966092G&gt;T</td>\n",
       "      <td>XP_011543781.1:p.Ser200Ter</td>\n",
       "      <td>PT_NBFP0BB7</td>\n",
       "      <td>rs1230094791</td>\n",
       "      <td>BS_54N90X5G</td>\n",
       "      <td>...</td>\n",
       "      <td>BCLAF3</td>\n",
       "      <td>SNV</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>WT</td>\n",
       "      <td>HGVSG:chrX.g.19966092G&gt;T</td>\n",
       "      <td>ENSEMBL:ENSG00000173681</td>\n",
       "      <td>ENSEMBL:XM_011545479.2</td>\n",
       "      <td>ENSEMBL:XM_011545479.2</td>\n",
       "      <td>MONDO:0005072</td>\n",
       "      <td>KFCOHORT:SD-DYPMEHHF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18226</th>\n",
       "      <td>p.Ser200/681Ter</td>\n",
       "      <td>['stop gained']</td>\n",
       "      <td>ENSG00000173681</td>\n",
       "      <td>XM_005274475.3</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>chrX.g.19966092G&gt;T</td>\n",
       "      <td>XP_005274532.1:p.Ser200Ter</td>\n",
       "      <td>PT_NBFP0BB7</td>\n",
       "      <td>rs1230094791</td>\n",
       "      <td>BS_54N90X5G</td>\n",
       "      <td>...</td>\n",
       "      <td>BCLAF3</td>\n",
       "      <td>SNV</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>WT</td>\n",
       "      <td>HGVSG:chrX.g.19966092G&gt;T</td>\n",
       "      <td>ENSEMBL:ENSG00000173681</td>\n",
       "      <td>ENSEMBL:XM_005274475.3</td>\n",
       "      <td>ENSEMBL:XM_005274475.3</td>\n",
       "      <td>MONDO:0005072</td>\n",
       "      <td>KFCOHORT:SD-DYPMEHHF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18227</th>\n",
       "      <td>p.Ser200/691Ter</td>\n",
       "      <td>['stop gained']</td>\n",
       "      <td>ENSG00000173681</td>\n",
       "      <td>XM_005274474.3</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>chrX.g.19966092G&gt;T</td>\n",
       "      <td>XP_005274531.1:p.Ser200Ter</td>\n",
       "      <td>PT_NBFP0BB7</td>\n",
       "      <td>rs1230094791</td>\n",
       "      <td>BS_54N90X5G</td>\n",
       "      <td>...</td>\n",
       "      <td>BCLAF3</td>\n",
       "      <td>SNV</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>WT</td>\n",
       "      <td>HGVSG:chrX.g.19966092G&gt;T</td>\n",
       "      <td>ENSEMBL:ENSG00000173681</td>\n",
       "      <td>ENSEMBL:XM_005274474.3</td>\n",
       "      <td>ENSEMBL:XM_005274474.3</td>\n",
       "      <td>MONDO:0005072</td>\n",
       "      <td>KFCOHORT:SD-DYPMEHHF</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>28844 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              aa_change      consequence  ensembl_gene_id  \\\n",
       "0        p.Gly49/247Xaa   ['frameshift']  ENSG00000188290   \n",
       "2        p.Gly49/247Xaa   ['frameshift']  ENSG00000188290   \n",
       "9      p.Ser199/1291Ter  ['stop gained']  ENSG00000142949   \n",
       "15     p.Ser188/1553Ter  ['stop gained']  ENSG00000142949   \n",
       "25     p.Ser531/1907Ter  ['stop gained']  ENSG00000142949   \n",
       "...                 ...              ...              ...   \n",
       "18223   p.Ser200/682Ter  ['stop gained']  ENSG00000173681   \n",
       "18224   p.Ser200/711Ter  ['stop gained']  ENSG00000173681   \n",
       "18225   p.Ser200/652Ter  ['stop gained']  ENSG00000173681   \n",
       "18226   p.Ser200/681Ter  ['stop gained']  ENSG00000173681   \n",
       "18227   p.Ser200/691Ter  ['stop gained']  ENSG00000173681   \n",
       "\n",
       "      ensembl_transcript_id feature_type               hgvsg  \\\n",
       "0           ENST00000428771   Transcript    chr1.g.999832del   \n",
       "2           ENST00000428771   Transcript    chr1.g.999832del   \n",
       "9           ENST00000412568   Transcript  chr1.g.43591872C>A   \n",
       "15          ENST00000429895   Transcript  chr1.g.43591872C>A   \n",
       "25          ENST00000359947   Transcript  chr1.g.43591872C>A   \n",
       "...                     ...          ...                 ...   \n",
       "18223        XM_011545477.2   Transcript  chrX.g.19966092G>T   \n",
       "18224        NM_001367774.2   Transcript  chrX.g.19966092G>T   \n",
       "18225        XM_011545479.2   Transcript  chrX.g.19966092G>T   \n",
       "18226        XM_005274475.3   Transcript  chrX.g.19966092G>T   \n",
       "18227        XM_005274474.3   Transcript  chrX.g.19966092G>T   \n",
       "\n",
       "                                     hgvsp participant_id      rsnumber  \\\n",
       "0      ENSP00000393198.2:p.Arg50GlyfsTer60    PT_3YW2V4JK           NaN   \n",
       "2      ENSP00000393198.2:p.Arg50GlyfsTer60    PT_3YW2V4JK           NaN   \n",
       "9            ENSP00000391764.2:p.Ser199Ter    PT_W50HSNE2           NaN   \n",
       "15           ENSP00000408952.1:p.Ser188Ter    PT_W50HSNE2           NaN   \n",
       "25           ENSP00000353030.4:p.Ser531Ter    PT_W50HSNE2           NaN   \n",
       "...                                    ...            ...           ...   \n",
       "18223           XP_011543779.1:p.Ser200Ter    PT_NBFP0BB7  rs1230094791   \n",
       "18224           NP_001354703.1:p.Ser200Ter    PT_NBFP0BB7  rs1230094791   \n",
       "18225           XP_011543781.1:p.Ser200Ter    PT_NBFP0BB7  rs1230094791   \n",
       "18226           XP_005274532.1:p.Ser200Ter    PT_NBFP0BB7  rs1230094791   \n",
       "18227           XP_005274531.1:p.Ser200Ter    PT_NBFP0BB7  rs1230094791   \n",
       "\n",
       "         sample_id  ... symbol.1 variant_class vep_impact zygosity  \\\n",
       "0      BS_SY6CCX4G  ...     HES4      deletion       HIGH      HET   \n",
       "2      BS_8313ZZ6E  ...     HES4      deletion       HIGH       WT   \n",
       "9      BS_Z40644EA  ...    PTPRF           SNV       HIGH       WT   \n",
       "15     BS_Z40644EA  ...    PTPRF           SNV       HIGH       WT   \n",
       "25     BS_Z40644EA  ...    PTPRF           SNV       HIGH       WT   \n",
       "...            ...  ...      ...           ...        ...      ...   \n",
       "18223  BS_54N90X5G  ...   BCLAF3           SNV       HIGH       WT   \n",
       "18224  BS_54N90X5G  ...   BCLAF3           SNV       HIGH       WT   \n",
       "18225  BS_54N90X5G  ...   BCLAF3           SNV       HIGH       WT   \n",
       "18226  BS_54N90X5G  ...   BCLAF3           SNV       HIGH       WT   \n",
       "18227  BS_54N90X5G  ...   BCLAF3           SNV       HIGH       WT   \n",
       "\n",
       "             hgvsg_variant_code            ens_gene_code  \\\n",
       "0        HGVSG:chr1.g.999832del  ENSEMBL:ENSG00000188290   \n",
       "2        HGVSG:chr1.g.999832del  ENSEMBL:ENSG00000188290   \n",
       "9      HGVSG:chr1.g.43591872C>A  ENSEMBL:ENSG00000142949   \n",
       "15     HGVSG:chr1.g.43591872C>A  ENSEMBL:ENSG00000142949   \n",
       "25     HGVSG:chr1.g.43591872C>A  ENSEMBL:ENSG00000142949   \n",
       "...                         ...                      ...   \n",
       "18223  HGVSG:chrX.g.19966092G>T  ENSEMBL:ENSG00000173681   \n",
       "18224  HGVSG:chrX.g.19966092G>T  ENSEMBL:ENSG00000173681   \n",
       "18225  HGVSG:chrX.g.19966092G>T  ENSEMBL:ENSG00000173681   \n",
       "18226  HGVSG:chrX.g.19966092G>T  ENSEMBL:ENSG00000173681   \n",
       "18227  HGVSG:chrX.g.19966092G>T  ENSEMBL:ENSG00000173681   \n",
       "\n",
       "           ens_transcript_code         ens_protein_code   disease_code  \\\n",
       "0      ENSEMBL:ENST00000428771  ENSEMBL:ENST00000428771  MONDO:0005072   \n",
       "2      ENSEMBL:ENST00000428771  ENSEMBL:ENST00000428771  MONDO:0005072   \n",
       "9      ENSEMBL:ENST00000412568  ENSEMBL:ENST00000412568  MONDO:0005072   \n",
       "15     ENSEMBL:ENST00000429895  ENSEMBL:ENST00000429895  MONDO:0005072   \n",
       "25     ENSEMBL:ENST00000359947  ENSEMBL:ENST00000359947  MONDO:0005072   \n",
       "...                        ...                      ...            ...   \n",
       "18223   ENSEMBL:XM_011545477.2   ENSEMBL:XM_011545477.2  MONDO:0005072   \n",
       "18224   ENSEMBL:NM_001367774.2   ENSEMBL:NM_001367774.2  MONDO:0005072   \n",
       "18225   ENSEMBL:XM_011545479.2   ENSEMBL:XM_011545479.2  MONDO:0005072   \n",
       "18226   ENSEMBL:XM_005274475.3   ENSEMBL:XM_005274475.3  MONDO:0005072   \n",
       "18227   ENSEMBL:XM_005274474.3   ENSEMBL:XM_005274474.3  MONDO:0005072   \n",
       "\n",
       "                cohort_code  \n",
       "0      KFCOHORT:SD-DYPMEHHF  \n",
       "2      KFCOHORT:SD-DYPMEHHF  \n",
       "9      KFCOHORT:SD-DYPMEHHF  \n",
       "15     KFCOHORT:SD-DYPMEHHF  \n",
       "25     KFCOHORT:SD-DYPMEHHF  \n",
       "...                     ...  \n",
       "18223  KFCOHORT:SD-DYPMEHHF  \n",
       "18224  KFCOHORT:SD-DYPMEHHF  \n",
       "18225  KFCOHORT:SD-DYPMEHHF  \n",
       "18226  KFCOHORT:SD-DYPMEHHF  \n",
       "18227  KFCOHORT:SD-DYPMEHHF  \n",
       "\n",
       "[28844 rows x 22 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# variant\n",
    "df['hgvsg'] = df['hgvsg'].replace({':': '.'},regex=True) \n",
    "df['hgvsg_variant_code'] = 'HGVSG:' + df['hgvsg'] \n",
    "\n",
    "# ensembl gene\n",
    "df['ens_gene_code'] = 'ENSEMBL:' + df['ensembl_gene_id']\n",
    "\n",
    "# ensembl transcript\n",
    "df['ens_transcript_code'] = 'ENSEMBL:' + df['ensembl_transcript_id']\n",
    "\n",
    "# disease/MONDO for Neuroblastoma\n",
    "df['disease_code'] = 'MONDO:0005072'\n",
    "\n",
    "# cohort\n",
    "df['cohort_code'] = 'KFCOHORT:SD-DYPMEHHF'\n",
    "\n",
    "# study\n",
    "df['study_code'] = 'KFSTUDY:KF-NBL-SOMATIC'\n",
    "\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9e04eaec",
   "metadata": {},
   "source": [
    "# Define edges"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "892565ae",
   "metadata": {},
   "source": [
    "###  Variant - Gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "43849d98",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subject</th>\n",
       "      <th>predicate</th>\n",
       "      <th>object</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1090</th>\n",
       "      <td>HGVSG:chr7.g.15686399C&gt;A</td>\n",
       "      <td>related_to_gene</td>\n",
       "      <td>ENSEMBL:ENSG00000106511</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1183</th>\n",
       "      <td>HGVSG:chr10.g.103999950T&gt;C</td>\n",
       "      <td>related_to_gene</td>\n",
       "      <td>ENSEMBL:ENSG00000065613</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1315</th>\n",
       "      <td>HGVSG:chr1.g.233258451C&gt;A</td>\n",
       "      <td>related_to_gene</td>\n",
       "      <td>ENSEMBL:ENSG00000135749</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>173</th>\n",
       "      <td>HGVSG:chrX.g.139804599_139804600delinsAA</td>\n",
       "      <td>related_to_gene</td>\n",
       "      <td>ENSEMBL:ENSG00000101974</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1293</th>\n",
       "      <td>HGVSG:chr14.g.99398980del</td>\n",
       "      <td>related_to_gene</td>\n",
       "      <td>ENSEMBL:ENSG00000183576</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       subject        predicate  \\\n",
       "1090                  HGVSG:chr7.g.15686399C>A  related_to_gene   \n",
       "1183                HGVSG:chr10.g.103999950T>C  related_to_gene   \n",
       "1315                 HGVSG:chr1.g.233258451C>A  related_to_gene   \n",
       "173   HGVSG:chrX.g.139804599_139804600delinsAA  related_to_gene   \n",
       "1293                 HGVSG:chr14.g.99398980del  related_to_gene   \n",
       "\n",
       "                       object  \n",
       "1090  ENSEMBL:ENSG00000106511  \n",
       "1183  ENSEMBL:ENSG00000065613  \n",
       "1315  ENSEMBL:ENSG00000135749  \n",
       "173   ENSEMBL:ENSG00000101974  \n",
       "1293  ENSEMBL:ENSG00000183576  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "e_var_gene = df[['hgvsg_variant_code','ens_gene_code']].dropna().drop_duplicates().reset_index(drop=True)\n",
    "e_var_gene.columns = ['subject','object']\n",
    "\n",
    "e_var_gene['predicate'] = 'related_to_gene'\n",
    "e_var_gene = e_var_gene[['subject','predicate','object']]\n",
    "\n",
    "e_var_gene.sample(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3aadd1e1",
   "metadata": {},
   "source": [
    "### Variant - Transcript"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "2f27c032",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\"\\ne_var_trans = df[['hgvsg_variant_code','ens_transcript_code']].dropna().drop_duplicates().reset_index(drop=True)\\ne_var_trans.columns = ['subject','object']\\ne_var_trans['predicate'] = 'has_transcript'\\ne_var_trans = e_var_trans[['subject','predicate','object']]\\ne_var_trans.sample(5)\\n\""
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'''\n",
    "e_var_trans = df[['hgvsg_variant_code','ens_transcript_code']].dropna().drop_duplicates().reset_index(drop=True)\n",
    "e_var_trans.columns = ['subject','object']\n",
    "e_var_trans['predicate'] = 'has_transcript'\n",
    "e_var_trans = e_var_trans[['subject','predicate','object']]\n",
    "e_var_trans.sample(5)\n",
    "'''"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dcfd337c",
   "metadata": {},
   "source": [
    "### Variant - Protein"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "d0d3c146",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1628083/2455318092.py:13: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  e_var_prot['predicate'] = 'has_protein'\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subject</th>\n",
       "      <th>predicate</th>\n",
       "      <th>object</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1260</th>\n",
       "      <td>HGVSG:chr17.g.7940210_7940211del</td>\n",
       "      <td>has_protein</td>\n",
       "      <td>ENSEMBL:ENSP00000484398</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1863</th>\n",
       "      <td>HGVSG:chr5.g.126488825C&gt;A</td>\n",
       "      <td>has_protein</td>\n",
       "      <td>ENSEMBL:ENSP00000426845</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>HGVSG:chr1.g.160823530dup</td>\n",
       "      <td>has_protein</td>\n",
       "      <td>ENSEMBL:ENSP00000289409</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1276</th>\n",
       "      <td>HGVSG:chr5.g.37179363C&gt;A</td>\n",
       "      <td>has_protein</td>\n",
       "      <td>ENSEMBL:ENSP00000474140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>888</th>\n",
       "      <td>HGVSG:chr1.g.181710978_181710979del</td>\n",
       "      <td>has_protein</td>\n",
       "      <td>ENSEMBL:ENSP00000441158</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  subject    predicate  \\\n",
       "1260     HGVSG:chr17.g.7940210_7940211del  has_protein   \n",
       "1863            HGVSG:chr5.g.126488825C>A  has_protein   \n",
       "97              HGVSG:chr1.g.160823530dup  has_protein   \n",
       "1276             HGVSG:chr5.g.37179363C>A  has_protein   \n",
       "888   HGVSG:chr1.g.181710978_181710979del  has_protein   \n",
       "\n",
       "                       object  \n",
       "1260  ENSEMBL:ENSP00000484398  \n",
       "1863  ENSEMBL:ENSP00000426845  \n",
       "97    ENSEMBL:ENSP00000289409  \n",
       "1276  ENSEMBL:ENSP00000474140  \n",
       "888   ENSEMBL:ENSP00000441158  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Some extra formatting needs to be done for the protein code...\n",
    "\n",
    "# Drop rows where its NaN\n",
    "df_prot = df.dropna(subset=['hgvsp'])\n",
    "\n",
    "# DROP rows where the ENSP is not defined.\n",
    "df_prot = df_prot[df_prot['hgvsp'].str.startswith('ENSP')]\n",
    "\n",
    "df_prot['ens_protein_code'] = 'ENSEMBL:' + pd.Series([i.split('.')[0] for i in df_prot['hgvsp']])\n",
    "\n",
    "e_var_prot = df_prot[['hgvsg_variant_code','ens_protein_code']]\n",
    "e_var_prot.columns = ['subject','object']\n",
    "e_var_prot['predicate'] = 'has_protein'\n",
    "e_var_prot = e_var_prot[['subject','predicate','object']].dropna().drop_duplicates().reset_index(drop=True)\n",
    "e_var_prot.sample(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "45f974a5",
   "metadata": {},
   "source": [
    "### Variant - Cohort"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "ea2128a7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subject</th>\n",
       "      <th>predicate</th>\n",
       "      <th>object</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>HGVSG:chr12.g.27961975del</td>\n",
       "      <td>belongs_to_cohort</td>\n",
       "      <td>KFCOHORT:SD-DYPMEHHF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1585</th>\n",
       "      <td>HGVSG:chr10.g.80079028G&gt;T</td>\n",
       "      <td>belongs_to_cohort</td>\n",
       "      <td>KFCOHORT:SD-DYPMEHHF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1391</th>\n",
       "      <td>HGVSG:chr10.g.122085864G&gt;T</td>\n",
       "      <td>belongs_to_cohort</td>\n",
       "      <td>KFCOHORT:SD-DYPMEHHF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>543</th>\n",
       "      <td>HGVSG:chr7.g.4861432G&gt;A</td>\n",
       "      <td>belongs_to_cohort</td>\n",
       "      <td>KFCOHORT:SD-DYPMEHHF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1378</th>\n",
       "      <td>HGVSG:chr1.g.40238794C&gt;A</td>\n",
       "      <td>belongs_to_cohort</td>\n",
       "      <td>KFCOHORT:SD-DYPMEHHF</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         subject          predicate                object\n",
       "42     HGVSG:chr12.g.27961975del  belongs_to_cohort  KFCOHORT:SD-DYPMEHHF\n",
       "1585   HGVSG:chr10.g.80079028G>T  belongs_to_cohort  KFCOHORT:SD-DYPMEHHF\n",
       "1391  HGVSG:chr10.g.122085864G>T  belongs_to_cohort  KFCOHORT:SD-DYPMEHHF\n",
       "543      HGVSG:chr7.g.4861432G>A  belongs_to_cohort  KFCOHORT:SD-DYPMEHHF\n",
       "1378    HGVSG:chr1.g.40238794C>A  belongs_to_cohort  KFCOHORT:SD-DYPMEHHF"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "e_var_cohort = df[['hgvsg_variant_code','cohort_code']].dropna().drop_duplicates().reset_index(drop=True)\n",
    "e_var_cohort.columns = ['subject','object']\n",
    "e_var_cohort['predicate'] = 'belongs_to_cohort'\n",
    "e_var_cohort = e_var_cohort[['subject','predicate','object']]\n",
    "e_var_cohort.sample(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c2e8057d",
   "metadata": {},
   "source": [
    "### Transcript - Protein"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "d45d7d8c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\"\\ne_trans_prot = df_prot[['ens_transcript_code','ens_protein_code']].dropna().drop_duplicates().reset_index(drop=True)\\ne_trans_prot.columns = ['subject','object']\\ne_trans_prot['predicate'] = 'has_protein'\\ne_trans_prot = e_trans_prot[['subject','predicate','object']]\\ne_trans_prot.sample(5)\\n\""
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'''\n",
    "e_trans_prot = df_prot[['ens_transcript_code','ens_protein_code']].dropna().drop_duplicates().reset_index(drop=True)\n",
    "e_trans_prot.columns = ['subject','object']\n",
    "e_trans_prot['predicate'] = 'has_protein'\n",
    "e_trans_prot = e_trans_prot[['subject','predicate','object']]\n",
    "e_trans_prot.sample(5)\n",
    "'''"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0a69c41a",
   "metadata": {},
   "source": [
    "### Study - Cohort"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "ce147625",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subject</th>\n",
       "      <th>predicate</th>\n",
       "      <th>object</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KFSTUDY:KF-NBL-SOMATIC</td>\n",
       "      <td>study_has_cohort</td>\n",
       "      <td>KFCOHORT:SD-DYPMEHHF</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  subject         predicate                object\n",
       "0  KFSTUDY:KF-NBL-SOMATIC  study_has_cohort  KFCOHORT:SD-DYPMEHHF"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "e_study_cohort = df[['study_code','cohort_code']].dropna().drop_duplicates().reset_index(drop=True)\n",
    "e_study_cohort.columns = ['subject','object']\n",
    "\n",
    "e_study_cohort['predicate'] = 'study_has_cohort'\n",
    "e_study_cohort = e_study_cohort[['subject','predicate','object']]\n",
    "e_study_cohort"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "23df038e",
   "metadata": {},
   "source": [
    "### Variant - HSCLO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "b553cdc8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['chr1',\n",
       " 'chr10',\n",
       " 'chr11',\n",
       " 'chr12',\n",
       " 'chr13',\n",
       " 'chr14',\n",
       " 'chr15',\n",
       " 'chr16',\n",
       " 'chr17',\n",
       " 'chr18',\n",
       " 'chr19',\n",
       " 'chr2',\n",
       " 'chr20',\n",
       " 'chr21',\n",
       " 'chr22',\n",
       " 'chr3',\n",
       " 'chr4',\n",
       " 'chr5',\n",
       " 'chr6',\n",
       " 'chr7',\n",
       " 'chr8',\n",
       " 'chr9',\n",
       " 'chrX',\n",
       " 'chrY']"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "q = pl.scan_csv('/home/stearb/U24/data/HSCLO/OWLNETS_edgelist_HSCLO.tsv',separator='\\t')\n",
    "hsclo = q.select(['node_id','lowerbound','upperbound']).collect()\n",
    "hsclo = hsclo.to_pandas()\n",
    "\n",
    "chroms = np.unique([i.split(' ')[-1].split('.')[0] for i in hsclo.node_id.astype(str)])\n",
    "chroms = [i.tolist() for i in chroms]  \n",
    "chroms.remove('Human_Genome_hg38')\n",
    "chroms.remove('MtDNA')\n",
    "chroms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "9650ac6b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\n",
      "194 194\n",
      "chr10\n",
      "72 72\n",
      "chr11\n",
      "105 104\n",
      "chr12\n",
      "83 83\n",
      "chr13\n",
      "37 37\n",
      "chr14\n",
      "54 54\n",
      "chr15\n",
      "64 64\n",
      "chr16\n",
      "76 76\n",
      "chr17\n",
      "94 94\n",
      "chr18\n",
      "32 32\n",
      "chr19\n",
      "98 98\n",
      "chr2\n",
      "591 589\n",
      "chr20\n",
      "38 37\n",
      "chr21\n",
      "20 20\n",
      "chr22\n",
      "39 39\n",
      "chr3\n",
      "109 108\n",
      "chr4\n",
      "83 82\n",
      "chr5\n",
      "86 86\n",
      "chr6\n",
      "83 83\n",
      "chr7\n",
      "72 72\n",
      "chr8\n",
      "50 50\n",
      "chr9\n",
      "73 73\n",
      "chrX\n",
      "75 75\n",
      "chrY\n",
      "3 3\n"
     ]
    }
   ],
   "source": [
    "e_var_hsclo_MASTER = list()\n",
    "\n",
    "# Go by chromosome and merge each variant to the correct HSCLO term/code\n",
    "# most of this is extracting the variant position from the hgvsg_variant_code col\n",
    "# and saving it to the variant_position col. Then bin on this col.\n",
    "\n",
    "for CHROM in chroms:\n",
    "    \n",
    "    print(CHROM)\n",
    "    \n",
    "    # search for 'chr1.' or 'chr2.' if CHROM is either 1 or 2. This prevents matching on 11,12,13, when matching on 'chr1'\n",
    "    if CHROM == 'chr1': CHROM = 'chr1\\.'\n",
    "    elif CHROM == 'chr2': CHROM == 'chr2\\.'\n",
    "    \n",
    "    # Select variant col which has the variant chromosomal position for just current CHROM\n",
    "    dfChrom = df[df['hgvsg_variant_code'].str.contains(CHROM)][['hgvsg_variant_code']].dropna()\n",
    "    \n",
    "    # format to extract just numerical position \n",
    "    dfChrom['variant_position'] = \\\n",
    "          [i.split('.')[-1].replace('del','').replace('ins','')\\\n",
    "            .replace('A','').replace('C','').replace('T','').replace('G','')\\\n",
    "            .replace('>','') if 'inv' not in i or 'dup' not in i else i for i\\\n",
    "            in dfChrom['hgvsg_variant_code']]\n",
    "\n",
    "    dfChrom['variant_position'] = dfChrom['variant_position'].replace('dup','',regex=True).replace('inv','',regex=True)\n",
    "    dfChrom['variant_position'] = [i.split('_')[0] for i in dfChrom['variant_position']]\n",
    "    dfChrom['variant_position'] = dfChrom['variant_position'].astype(int)\n",
    "    dfChrom = dfChrom.drop_duplicates().reset_index(drop=True)\n",
    "    \n",
    "    # get hsclo codes for current CHROM\n",
    "    hscloChrom = hsclo[hsclo['node_id'].str.contains(CHROM)].dropna()\n",
    "\n",
    "    ###### do the binning\n",
    "    dfChromBins = bin_column(pl.DataFrame(dfChrom),col2bin='variant_position').to_pandas()\n",
    "\n",
    "    # make lowerbound col to merge in hsclo codes\n",
    "    dfChromBins['lowerbound'] = [float(i.split(',')[0][1:]) for i in dfChromBins['bins']]\n",
    "\n",
    "    # merge in hsclo codes \n",
    "    dfMerged = pd.merge(dfChromBins,hscloChrom,on='lowerbound',how='left')\n",
    "    \n",
    "    # this may introduce an error, if im matching on lowerbound only, as there are multiple resolution\n",
    "    # levels, so upperbound must also be specified!\n",
    "    # ...check that the variant_position is b/t lower and upperbound\n",
    "    print(len(dfMerged) ,len(dfMerged[(dfMerged['variant_position'] > dfMerged['lowerbound']) &\\\n",
    "        (dfMerged['variant_position'] < dfMerged['upperbound'])]))\n",
    "    \n",
    "    #assert dfMerged.shape == dfMerged[(dfMerged['variant_position'] > dfMerged['lowerbound']) &\\\n",
    "    #    (dfMerged['variant_position'] < dfMerged['upperbound'])].shape\n",
    "    \n",
    "    e_var_hsclo = dfMerged[['hgvsg_variant_code','node_id']].drop_duplicates().dropna().reset_index(drop=True)\n",
    "    e_var_hsclo.columns = ['subject','object']\n",
    "    e_var_hsclo['predicate'] = 'has_location'\n",
    "    e_var_hsclo = e_var_hsclo[['subject','predicate','object']]\n",
    "    \n",
    "    # e_var_hsclo is longer than dfChrom bc each variant is mapped to multiple HSCLO resolution levels\n",
    "    #print(len(e_var_hsclo),len(dfChrom)) \n",
    "    \n",
    "    if len(e_var_hsclo_MASTER) == 0:\n",
    "        e_var_hsclo_MASTER = e_var_hsclo\n",
    "    else:\n",
    "        e_var_hsclo_MASTER = pd.concat([e_var_hsclo_MASTER,e_var_hsclo])\n",
    "       \n",
    "e_var_hsclo_MASTER = e_var_hsclo_MASTER.drop_duplicates().dropna().reset_index(drop=True)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0caf5a4d",
   "metadata": {},
   "source": [
    "# Save edges"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "3febdab4",
   "metadata": {},
   "outputs": [],
   "source": [
    "edges_all = pd.concat([e_var_gene,e_var_prot,#e_var_trans,e_trans_prot,\n",
    "                       e_var_cohort,e_study_cohort,\n",
    "                       e_var_hsclo_MASTER])\n",
    "\n",
    "edges_all = edges_all.drop_duplicates().dropna().reset_index(drop=True)\n",
    "\n",
    "edges_all.to_csv('/mnt/isilon/opentargets/U24KG/data/owlnets_nodes_and_edges/NBL_somatic/OWLNETS_edgelist.txt',\n",
    "             sep= \"\\t\",index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "14138ea8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Counter({2: 5101, 1: 2134})\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Counter({2: 7235})"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from collections import Counter\n",
    "\n",
    "# check that there is only 1 colon, or 0 colons (for HSCLO)\n",
    "print(Counter([len(i.split(':')) for i in edges_all['object']]))\n",
    "Counter([len(i.split(':')) for i in edges_all['subject']])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "fa5ee77d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subject</th>\n",
       "      <th>predicate</th>\n",
       "      <th>object</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HGVSG:chr1.g.999832del</td>\n",
       "      <td>related_to_gene</td>\n",
       "      <td>ENSEMBL:ENSG00000188290</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HGVSG:chr1.g.43591872C&gt;A</td>\n",
       "      <td>related_to_gene</td>\n",
       "      <td>ENSEMBL:ENSG00000142949</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HGVSG:chr11.g.63952438C&gt;T</td>\n",
       "      <td>related_to_gene</td>\n",
       "      <td>ENSEMBL:ENSG00000110583</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HGVSG:chr11.g.125102239C&gt;A</td>\n",
       "      <td>related_to_gene</td>\n",
       "      <td>ENSEMBL:ENSG00000150433</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HGVSG:chr13.g.51467537G&gt;T</td>\n",
       "      <td>related_to_gene</td>\n",
       "      <td>ENSEMBL:ENSG00000236778</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22901</th>\n",
       "      <td>HGVSG:chrX.g.53084722del</td>\n",
       "      <td>has_location</td>\n",
       "      <td>HSCLO chrX.53084001-53085000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22902</th>\n",
       "      <td>HGVSG:chrX.g.20151338C&gt;A</td>\n",
       "      <td>has_location</td>\n",
       "      <td>HSCLO chrX.20151001-20152000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22903</th>\n",
       "      <td>HGVSG:chrY.g.13160138_13160139del</td>\n",
       "      <td>has_location</td>\n",
       "      <td>HSCLO chrY.13160001-13170000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22904</th>\n",
       "      <td>HGVSG:chrY.g.13160138_13160139del</td>\n",
       "      <td>has_location</td>\n",
       "      <td>HSCLO chrY.13160001-13161000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22905</th>\n",
       "      <td>HGVSG:chrY.g.8129232A&gt;G</td>\n",
       "      <td>has_location</td>\n",
       "      <td>HSCLO chrY.8129001-8130000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>22906 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 subject        predicate  \\\n",
       "0                 HGVSG:chr1.g.999832del  related_to_gene   \n",
       "1               HGVSG:chr1.g.43591872C>A  related_to_gene   \n",
       "2              HGVSG:chr11.g.63952438C>T  related_to_gene   \n",
       "3             HGVSG:chr11.g.125102239C>A  related_to_gene   \n",
       "4              HGVSG:chr13.g.51467537G>T  related_to_gene   \n",
       "...                                  ...              ...   \n",
       "22901           HGVSG:chrX.g.53084722del     has_location   \n",
       "22902           HGVSG:chrX.g.20151338C>A     has_location   \n",
       "22903  HGVSG:chrY.g.13160138_13160139del     has_location   \n",
       "22904  HGVSG:chrY.g.13160138_13160139del     has_location   \n",
       "22905            HGVSG:chrY.g.8129232A>G     has_location   \n",
       "\n",
       "                             object  \n",
       "0           ENSEMBL:ENSG00000188290  \n",
       "1           ENSEMBL:ENSG00000142949  \n",
       "2           ENSEMBL:ENSG00000110583  \n",
       "3           ENSEMBL:ENSG00000150433  \n",
       "4           ENSEMBL:ENSG00000236778  \n",
       "...                             ...  \n",
       "22901  HSCLO chrX.53084001-53085000  \n",
       "22902  HSCLO chrX.20151001-20152000  \n",
       "22903  HSCLO chrY.13160001-13170000  \n",
       "22904  HSCLO chrY.13160001-13161000  \n",
       "22905    HSCLO chrY.8129001-8130000  \n",
       "\n",
       "[22906 rows x 3 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# check edges look good\n",
    "e = pd.read_csv('/mnt/isilon/opentargets/U24KG/data/owlnets_nodes_and_edges/NBL_somatic/OWLNETS_edgelist.txt',\n",
    "           sep='\\t')\n",
    "e"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dddfe0d0",
   "metadata": {},
   "source": [
    "# Save nodes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "572db994",
   "metadata": {},
   "outputs": [],
   "source": [
    "nodes_all = pd.concat([edges_all['subject'],edges_all['object']])\\\n",
    "                          .drop_duplicates().dropna().reset_index(drop=True)\n",
    "\n",
    "nodes_all = pd.DataFrame(nodes_all,columns=['node_id'])\n",
    "nodes_all = fill_missing_cols(nodes_all)\n",
    "\n",
    "# NO Need to save HSCLO node ids\n",
    "nodes_all = nodes_all[~nodes_all['node_id'].str.startswith('HSCLO')]\\\n",
    "    .drop_duplicates().dropna(subset=['node_id']).reset_index(drop=True)\n",
    "\n",
    "nodes_all.to_csv(\n",
    "    '/mnt/isilon/opentargets/U24KG/data/owlnets_nodes_and_edges/NBL_somatic/OWLNETS_node_metadata.txt',\n",
    "             sep= \"\\t\",index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "9648cdda",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>node_id</th>\n",
       "      <th>node_dbxrefs</th>\n",
       "      <th>node_synonyms</th>\n",
       "      <th>value</th>\n",
       "      <th>node_namespace</th>\n",
       "      <th>unit</th>\n",
       "      <th>node_label</th>\n",
       "      <th>node_definition</th>\n",
       "      <th>lowerbound</th>\n",
       "      <th>upperbound</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HGVSG:chr1.g.999832del</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HGVSG:chr1.g.43591872C&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HGVSG:chr11.g.63952438C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HGVSG:chr11.g.125102239C&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HGVSG:chr13.g.51467537G&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17413</th>\n",
       "      <td>ENSEMBL:NM_001367774.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17414</th>\n",
       "      <td>ENSEMBL:XM_011545479.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17415</th>\n",
       "      <td>ENSEMBL:XM_005274475.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17416</th>\n",
       "      <td>ENSEMBL:XM_005274474.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17417</th>\n",
       "      <td>KFSTUDY:KF-NBL-SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17418 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                          node_id  node_dbxrefs  node_synonyms  value  \\\n",
       "0          HGVSG:chr1.g.999832del           NaN            NaN    NaN   \n",
       "1        HGVSG:chr1.g.43591872C>A           NaN            NaN    NaN   \n",
       "2       HGVSG:chr11.g.63952438C>T           NaN            NaN    NaN   \n",
       "3      HGVSG:chr11.g.125102239C>A           NaN            NaN    NaN   \n",
       "4       HGVSG:chr13.g.51467537G>T           NaN            NaN    NaN   \n",
       "...                           ...           ...            ...    ...   \n",
       "17413      ENSEMBL:NM_001367774.2           NaN            NaN    NaN   \n",
       "17414      ENSEMBL:XM_011545479.2           NaN            NaN    NaN   \n",
       "17415      ENSEMBL:XM_005274475.3           NaN            NaN    NaN   \n",
       "17416      ENSEMBL:XM_005274474.3           NaN            NaN    NaN   \n",
       "17417      KFSTUDY:KF-NBL-SOMATIC           NaN            NaN    NaN   \n",
       "\n",
       "       node_namespace  unit  node_label  node_definition  lowerbound  \\\n",
       "0                 NaN   NaN         NaN              NaN         NaN   \n",
       "1                 NaN   NaN         NaN              NaN         NaN   \n",
       "2                 NaN   NaN         NaN              NaN         NaN   \n",
       "3                 NaN   NaN         NaN              NaN         NaN   \n",
       "4                 NaN   NaN         NaN              NaN         NaN   \n",
       "...               ...   ...         ...              ...         ...   \n",
       "17413             NaN   NaN         NaN              NaN         NaN   \n",
       "17414             NaN   NaN         NaN              NaN         NaN   \n",
       "17415             NaN   NaN         NaN              NaN         NaN   \n",
       "17416             NaN   NaN         NaN              NaN         NaN   \n",
       "17417             NaN   NaN         NaN              NaN         NaN   \n",
       "\n",
       "       upperbound  \n",
       "0             NaN  \n",
       "1             NaN  \n",
       "2             NaN  \n",
       "3             NaN  \n",
       "4             NaN  \n",
       "...           ...  \n",
       "17413         NaN  \n",
       "17414         NaN  \n",
       "17415         NaN  \n",
       "17416         NaN  \n",
       "17417         NaN  \n",
       "\n",
       "[17418 rows x 10 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# check nodes look good\n",
    "n = pd.read_csv('/mnt/isilon/opentargets/U24KG/data/owlnets_nodes_and_edges/NBL_somatic/OWLNETS_node_metadata.txt',\n",
    "           sep='\\t')\n",
    "n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "57c6c4e2",
   "metadata": {},
   "outputs": [],
   "source": [
    "from collections import Counter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "f407934d",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Counter({'HSCLO chr1.999001-1000000': 1,\n",
       "         'HSCLO chr1.43591001-43592000': 1,\n",
       "         'HSCLO chr1.268001-269000': 1,\n",
       "         'HSCLO chr1.39385001-39386000': 1,\n",
       "         'HSCLO chr1.154251001-154252000': 1,\n",
       "         'HSCLO chr1.160823001-160824000': 1,\n",
       "         'HSCLO chr1.171270001-171280000': 1,\n",
       "         'HSCLO chr1.171270001-171271000': 1,\n",
       "         'HSCLO chr1.27349001-27350000': 1,\n",
       "         'HSCLO chr1.91501001-91502000': 1,\n",
       "         'HSCLO chr1.39357001-39358000': 1,\n",
       "         'HSCLO chr1.63322001-63323000': 1,\n",
       "         'HSCLO chr1.151070001-151080000': 1,\n",
       "         'HSCLO chr1.151070001-151071000': 1,\n",
       "         'HSCLO chr1.208045001-208046000': 1,\n",
       "         'HSCLO chr1.248453001-248454000': 1,\n",
       "         'HSCLO chr1.243165001-243166000': 1,\n",
       "         'HSCLO chr1.197199001-197200000': 1,\n",
       "         'HSCLO chr1.1321001-1322000': 1,\n",
       "         'HSCLO chr1.157097001-157098000': 1,\n",
       "         'HSCLO chr1.181034001-181035000': 1,\n",
       "         'HSCLO chr1.78013001-78014000': 1,\n",
       "         'HSCLO chr1.201468001-201469000': 1,\n",
       "         'HSCLO chr1.88983001-88984000': 1,\n",
       "         'HSCLO chr1.158742001-158743000': 1,\n",
       "         'HSCLO chr1.181710001-181720000': 1,\n",
       "         'HSCLO chr1.181710001-181711000': 1,\n",
       "         'HSCLO chr1.197122001-197123000': 1,\n",
       "         'HSCLO chr1.35561001-35562000': 1,\n",
       "         'HSCLO chr1.196689001-196690000': 1,\n",
       "         'HSCLO chr1.201783001-201784000': 1,\n",
       "         'HSCLO chr1.82478001-82479000': 1,\n",
       "         'HSCLO chr1.42536001-42537000': 1,\n",
       "         'HSCLO chr1.22893001-22894000': 1,\n",
       "         'HSCLO chr1.117870001-117880000': 1,\n",
       "         'HSCLO chr1.117870001-117871000': 1,\n",
       "         'HSCLO chr1.241502001-241503000': 1,\n",
       "         'HSCLO chr1.34981001-34982000': 1,\n",
       "         'HSCLO chr1.35381001-35382000': 1,\n",
       "         'HSCLO chr1.114994001-114995000': 1,\n",
       "         'HSCLO chr1.84631001-84632000': 1,\n",
       "         'HSCLO chr1.152352001-152353000': 1,\n",
       "         'HSCLO chr1.210919001-210920000': 1,\n",
       "         'HSCLO chr1.31584001-31585000': 1,\n",
       "         'HSCLO chr1.102987001-102988000': 1,\n",
       "         'HSCLO chr1.153717001-153718000': 1,\n",
       "         'HSCLO chr1.186337001-186338000': 1,\n",
       "         'HSCLO chr1.116031001-116032000': 1,\n",
       "         'HSCLO chr1.206600001-206700000': 1,\n",
       "         'HSCLO chr1.206600001-206610000': 1,\n",
       "         'HSCLO chr1.206600001-206601000': 1,\n",
       "         'HSCLO chr1.155338001-155339000': 1,\n",
       "         'HSCLO chr1.180836001-180837000': 1,\n",
       "         'HSCLO chr1.15219001-15220000': 1,\n",
       "         'HSCLO chr1.70432001-70433000': 1,\n",
       "         'HSCLO chr1.43975001-43976000': 1,\n",
       "         'HSCLO chr1.64812001-64813000': 1,\n",
       "         'HSCLO chr1.197772001-197773000': 1,\n",
       "         'HSCLO chr1.154234001-154235000': 1,\n",
       "         'HSCLO chr1.212836001-212837000': 1,\n",
       "         'HSCLO chr1.158845001-158846000': 1,\n",
       "         'HSCLO chr1.86424001-86425000': 1,\n",
       "         'HSCLO chr1.108696001-108697000': 1,\n",
       "         'HSCLO chr1.6667001-6668000': 1,\n",
       "         'HSCLO chr1.63371001-63372000': 1,\n",
       "         'HSCLO chr1.219978001-219979000': 1,\n",
       "         'HSCLO chr1.10303001-10304000': 1,\n",
       "         'HSCLO chr1.207470001-207480000': 1,\n",
       "         'HSCLO chr1.207470001-207471000': 1,\n",
       "         'HSCLO chr1.42683001-42684000': 1,\n",
       "         'HSCLO chr1.193122001-193123000': 1,\n",
       "         'HSCLO chr1.215674001-215675000': 1,\n",
       "         'HSCLO chr1.110605001-110606000': 1,\n",
       "         'HSCLO chr1.145334001-145335000': 1,\n",
       "         'HSCLO chr1.248038001-248039000': 1,\n",
       "         'HSCLO chr1.228371001-228372000': 1,\n",
       "         'HSCLO chr1.92007001-92008000': 1,\n",
       "         'HSCLO chr1.196916001-196917000': 1,\n",
       "         'HSCLO chr1.1349001-1350000': 1,\n",
       "         'HSCLO chr1.42154001-42155000': 1,\n",
       "         'HSCLO chr1.172570001-172580000': 1,\n",
       "         'HSCLO chr1.172570001-172571000': 1,\n",
       "         'HSCLO chr1.8656001-8657000': 1,\n",
       "         'HSCLO chr1.83890001-83900000': 1,\n",
       "         'HSCLO chr1.83890001-83891000': 1,\n",
       "         'HSCLO chr1.109258001-109259000': 1,\n",
       "         'HSCLO chr1.62273001-62274000': 1,\n",
       "         'HSCLO chr1.243164001-243165000': 1,\n",
       "         'HSCLO chr1.81514001-81515000': 1,\n",
       "         'HSCLO chr1.16760001-16770000': 1,\n",
       "         'HSCLO chr1.16760001-16761000': 1,\n",
       "         'HSCLO chr1.171535001-171536000': 1,\n",
       "         'HSCLO chr1.100720001-100730000': 1,\n",
       "         'HSCLO chr1.100720001-100721000': 1,\n",
       "         'HSCLO chr1.209618001-209619000': 1,\n",
       "         'HSCLO chr1.235993001-235994000': 1,\n",
       "         'HSCLO chr1.12568001-12569000': 1,\n",
       "         'HSCLO chr1.28459001-28460000': 1,\n",
       "         'HSCLO chr1.28993001-28994000': 1,\n",
       "         'HSCLO chr1.42850001-42860000': 1,\n",
       "         'HSCLO chr1.42850001-42851000': 1,\n",
       "         'HSCLO chr1.155015001-155016000': 1,\n",
       "         'HSCLO chr1.89007001-89008000': 1,\n",
       "         'HSCLO chr1.228743001-228744000': 1,\n",
       "         'HSCLO chr1.74084001-74085000': 1,\n",
       "         'HSCLO chr1.48101001-48102000': 1,\n",
       "         'HSCLO chr1.185980001-185990000': 1,\n",
       "         'HSCLO chr1.185980001-185981000': 1,\n",
       "         'HSCLO chr1.145847001-145848000': 1,\n",
       "         'HSCLO chr1.92175001-92176000': 1,\n",
       "         'HSCLO chr1.112690001-112700000': 1,\n",
       "         'HSCLO chr1.112690001-112691000': 1,\n",
       "         'HSCLO chr1.112726001-112727000': 1,\n",
       "         'HSCLO chr1.153978001-153979000': 1,\n",
       "         'HSCLO chr1.23629001-23630000': 1,\n",
       "         'HSCLO chr1.166921001-166922000': 1,\n",
       "         'HSCLO chr1.196701001-196702000': 1,\n",
       "         'HSCLO chr1.12001001-12002000': 1,\n",
       "         'HSCLO chr1.15874001-15875000': 1,\n",
       "         'HSCLO chr1.54009001-54010000': 1,\n",
       "         'HSCLO chr1.143419001-143420000': 1,\n",
       "         'HSCLO chr1.148967001-148968000': 1,\n",
       "         'HSCLO chr1.222627001-222628000': 1,\n",
       "         'HSCLO chr1.182880001-182890000': 1,\n",
       "         'HSCLO chr1.182880001-182881000': 1,\n",
       "         'HSCLO chr1.22526001-22527000': 1,\n",
       "         'HSCLO chr1.155264001-155265000': 1,\n",
       "         'HSCLO chr1.156581001-156582000': 1,\n",
       "         'HSCLO chr1.181719001-181720000': 1,\n",
       "         'HSCLO chr1.244024001-244025000': 1,\n",
       "         'HSCLO chr1.39455001-39456000': 1,\n",
       "         'HSCLO chr1.218307001-218308000': 1,\n",
       "         'HSCLO chr1.67395001-67396000': 1,\n",
       "         'HSCLO chr1.111704001-111705000': 1,\n",
       "         'HSCLO chr1.183546001-183547000': 1,\n",
       "         'HSCLO chr1.207023001-207024000': 1,\n",
       "         'HSCLO chr1.225887001-225888000': 1,\n",
       "         'HSCLO chr1.235182001-235183000': 1,\n",
       "         'HSCLO chr1.16620001-16630000': 1,\n",
       "         'HSCLO chr1.16620001-16621000': 1,\n",
       "         'HSCLO chr1.98691001-98692000': 1,\n",
       "         'HSCLO chr1.204465001-204466000': 1,\n",
       "         'HSCLO chr1.231337001-231338000': 1,\n",
       "         'HSCLO chr1.247934001-247935000': 1,\n",
       "         'HSCLO chr1.156923001-156924000': 1,\n",
       "         'HSCLO chr1.118086001-118087000': 1,\n",
       "         'HSCLO chr1.71069001-71070000': 1,\n",
       "         'HSCLO chr1.152797001-152798000': 1,\n",
       "         'HSCLO chr1.235809001-235810000': 1,\n",
       "         'HSCLO chr1.40457001-40458000': 1,\n",
       "         'HSCLO chr1.229627001-229628000': 1,\n",
       "         'HSCLO chr1.248847001-248848000': 1,\n",
       "         'HSCLO chr1.43424001-43425000': 1,\n",
       "         'HSCLO chr1.56720001-56730000': 1,\n",
       "         'HSCLO chr1.56720001-56721000': 1,\n",
       "         'HSCLO chr1.158677001-158678000': 1,\n",
       "         'HSCLO chr1.160425001-160426000': 1,\n",
       "         'HSCLO chr1.233258001-233259000': 1,\n",
       "         'HSCLO chr1.20480001-20490000': 1,\n",
       "         'HSCLO chr1.20480001-20481000': 1,\n",
       "         'HSCLO chr1.65636001-65637000': 1,\n",
       "         'HSCLO chr1.155479001-155480000': 1,\n",
       "         'HSCLO chr1.156931001-156932000': 1,\n",
       "         'HSCLO chr1.157696001-157697000': 1,\n",
       "         'HSCLO chr1.228401001-228402000': 1,\n",
       "         'HSCLO chr1.225346001-225347000': 1,\n",
       "         'HSCLO chr1.225652001-225653000': 1,\n",
       "         'HSCLO chr1.152353001-152354000': 1,\n",
       "         'HSCLO chr1.228384001-228385000': 1,\n",
       "         'HSCLO chr1.40238001-40239000': 1,\n",
       "         'HSCLO chr1.201365001-201366000': 1,\n",
       "         'HSCLO chr1.236808001-236809000': 1,\n",
       "         'HSCLO chr1.152109001-152110000': 1,\n",
       "         'HSCLO chr1.197510001-197520000': 1,\n",
       "         'HSCLO chr1.197510001-197511000': 1,\n",
       "         'HSCLO chr1.240870001-240880000': 1,\n",
       "         'HSCLO chr1.240870001-240871000': 1,\n",
       "         'HSCLO chr1.37960001-37970000': 1,\n",
       "         'HSCLO chr1.37960001-37961000': 1,\n",
       "         'HSCLO chr1.160681001-160682000': 1,\n",
       "         'HSCLO chr1.214393001-214394000': 1,\n",
       "         'HSCLO chr1.115286001-115287000': 1,\n",
       "         'HSCLO chr1.146020001-146030000': 1,\n",
       "         'HSCLO chr1.146020001-146021000': 1,\n",
       "         'HSCLO chr1.198292001-198293000': 1,\n",
       "         'HSCLO chr1.78045001-78046000': 1,\n",
       "         'HSCLO chr1.2024001-2025000': 1,\n",
       "         'HSCLO chr1.151247001-151248000': 1,\n",
       "         'HSCLO chr1.41557001-41558000': 1,\n",
       "         'HSCLO chr1.196914001-196915000': 1,\n",
       "         'HSCLO chr1.2203001-2204000': 1,\n",
       "         'HSCLO chr1.107785001-107786000': 1,\n",
       "         'HSCLO chr10.209001-210000': 1,\n",
       "         'HSCLO chr10.21516001-21517000': 1,\n",
       "         'HSCLO chr10.68982001-68983000': 1,\n",
       "         'HSCLO chr10.38651001-38652000': 1,\n",
       "         'HSCLO chr10.100507001-100508000': 1,\n",
       "         'HSCLO chr10.22209001-22210000': 1,\n",
       "         'HSCLO chr10.116954001-116955000': 1,\n",
       "         'HSCLO chr10.95845001-95846000': 1,\n",
       "         'HSCLO chr10.28614001-28615000': 1,\n",
       "         'HSCLO chr10.60300001-60400000': 1,\n",
       "         'HSCLO chr10.60300001-60310000': 1,\n",
       "         'HSCLO chr10.60300001-60301000': 1,\n",
       "         'HSCLO chr10.97163001-97164000': 1,\n",
       "         'HSCLO chr10.68749001-68750000': 1,\n",
       "         'HSCLO chr10.125775001-125776000': 1,\n",
       "         'HSCLO chr10.60139001-60140000': 1,\n",
       "         'HSCLO chr10.79558001-79559000': 1,\n",
       "         'HSCLO chr10.114221001-114222000': 1,\n",
       "         'HSCLO chr10.5709001-5710000': 1,\n",
       "         'HSCLO chr10.7735001-7736000': 1,\n",
       "         'HSCLO chr10.68754001-68755000': 1,\n",
       "         'HSCLO chr10.97871001-97872000': 1,\n",
       "         'HSCLO chr10.100990001-101000000': 1,\n",
       "         'HSCLO chr10.100990001-100991000': 1,\n",
       "         'HSCLO chr10.48877001-48878000': 1,\n",
       "         'HSCLO chr10.112163001-112164000': 1,\n",
       "         'HSCLO chr10.46550001-46560000': 1,\n",
       "         'HSCLO chr10.46550001-46551000': 1,\n",
       "         'HSCLO chr10.77012001-77013000': 1,\n",
       "         'HSCLO chr10.62198001-62199000': 1,\n",
       "         'HSCLO chr10.91842001-91843000': 1,\n",
       "         'HSCLO chr10.119588001-119589000': 1,\n",
       "         'HSCLO chr10.127384001-127385000': 1,\n",
       "         'HSCLO chr10.124942001-124943000': 1,\n",
       "         'HSCLO chr10.103603001-103604000': 1,\n",
       "         'HSCLO chr10.3163001-3164000': 1,\n",
       "         'HSCLO chr10.87894001-87895000': 1,\n",
       "         'HSCLO chr10.26096001-26097000': 1,\n",
       "         'HSCLO chr10.89723001-89724000': 1,\n",
       "         'HSCLO chr10.102158001-102159000': 1,\n",
       "         'HSCLO chr10.5159001-5160000': 1,\n",
       "         'HSCLO chr10.128106001-128107000': 1,\n",
       "         'HSCLO chr10.103999001-104000000': 1,\n",
       "         'HSCLO chr10.5107001-5108000': 1,\n",
       "         'HSCLO chr10.27745001-27746000': 1,\n",
       "         'HSCLO chr10.73479001-73480000': 1,\n",
       "         'HSCLO chr10.73648001-73649000': 1,\n",
       "         'HSCLO chr10.98427001-98428000': 1,\n",
       "         'HSCLO chr10.97751001-97752000': 1,\n",
       "         'HSCLO chr10.68396001-68397000': 1,\n",
       "         'HSCLO chr10.72760001-72770000': 1,\n",
       "         'HSCLO chr10.72760001-72761000': 1,\n",
       "         'HSCLO chr10.13283001-13284000': 1,\n",
       "         'HSCLO chr10.86446001-86447000': 1,\n",
       "         'HSCLO chr10.133537001-133538000': 1,\n",
       "         'HSCLO chr10.32284001-32285000': 1,\n",
       "         'HSCLO chr10.126996001-126997000': 1,\n",
       "         'HSCLO chr10.5766001-5767000': 1,\n",
       "         'HSCLO chr10.122085001-122086000': 1,\n",
       "         'HSCLO chr10.73790001-73800000': 1,\n",
       "         'HSCLO chr10.73790001-73791000': 1,\n",
       "         'HSCLO chr10.104141001-104142000': 1,\n",
       "         'HSCLO chr10.88732001-88733000': 1,\n",
       "         'HSCLO chr10.122456001-122457000': 1,\n",
       "         'HSCLO chr10.58648001-58649000': 1,\n",
       "         'HSCLO chr10.89400001-89500000': 1,\n",
       "         'HSCLO chr10.89400001-89410000': 1,\n",
       "         'HSCLO chr10.89400001-89401000': 1,\n",
       "         'HSCLO chr10.88309001-88310000': 1,\n",
       "         'HSCLO chr10.93702001-93703000': 1,\n",
       "         'HSCLO chr10.45847001-45848000': 1,\n",
       "         'HSCLO chr10.80079001-80080000': 1,\n",
       "         'HSCLO chr11.63952001-63953000': 1,\n",
       "         'HSCLO chr11.125102001-125103000': 1,\n",
       "         'HSCLO chr11.7927001-7928000': 1,\n",
       "         'HSCLO chr11.45979001-45980000': 1,\n",
       "         'HSCLO chr11.74969001-74970000': 1,\n",
       "         'HSCLO chr11.107532001-107533000': 1,\n",
       "         'HSCLO chr11.57741001-57742000': 1,\n",
       "         'HSCLO chr11.104944001-104945000': 1,\n",
       "         'HSCLO chr11.118312001-118313000': 1,\n",
       "         'HSCLO chr11.92354001-92355000': 1,\n",
       "         'HSCLO chr11.130473001-130474000': 1,\n",
       "         'HSCLO chr11.65585001-65586000': 1,\n",
       "         'HSCLO chr11.64341001-64342000': 1,\n",
       "         'HSCLO chr11.113823001-113824000': 1,\n",
       "         'HSCLO chr11.62600001-62700000': 1,\n",
       "         'HSCLO chr11.62600001-62610000': 1,\n",
       "         'HSCLO chr11.62600001-62601000': 1,\n",
       "         'HSCLO chr11.66555001-66556000': 1,\n",
       "         'HSCLO chr11.61415001-61416000': 1,\n",
       "         'HSCLO chr11.94496001-94497000': 1,\n",
       "         'HSCLO chr11.124887001-124888000': 1,\n",
       "         'HSCLO chr11.55796001-55797000': 1,\n",
       "         'HSCLO chr11.73363001-73364000': 1,\n",
       "         'HSCLO chr11.105008001-105009000': 1,\n",
       "         'HSCLO chr11.66843001-66844000': 1,\n",
       "         'HSCLO chr11.68043001-68044000': 1,\n",
       "         'HSCLO chr11.88175001-88176000': 1,\n",
       "         'HSCLO chr11.1889001-1890000': 1,\n",
       "         'HSCLO chr11.31065001-31066000': 1,\n",
       "         'HSCLO chr11.74710001-74720000': 1,\n",
       "         'HSCLO chr11.74710001-74711000': 1,\n",
       "         'HSCLO chr11.118170001-118180000': 1,\n",
       "         'HSCLO chr11.118170001-118171000': 1,\n",
       "         'HSCLO chr11.62707001-62708000': 1,\n",
       "         'HSCLO chr11.119055001-119056000': 1,\n",
       "         'HSCLO chr11.57546001-57547000': 1,\n",
       "         'HSCLO chr11.119356001-119357000': 1,\n",
       "         'HSCLO chr11.49183001-49184000': 1,\n",
       "         'HSCLO chr11.126269001-126270000': 1,\n",
       "         'HSCLO chr11.1240001-1250000': 1,\n",
       "         'HSCLO chr11.1240001-1241000': 1,\n",
       "         'HSCLO chr11.118117001-118118000': 1,\n",
       "         'HSCLO chr11.46680001-46690000': 1,\n",
       "         'HSCLO chr11.46680001-46681000': 1,\n",
       "         'HSCLO chr11.134157001-134158000': 1,\n",
       "         'HSCLO chr11.74129001-74130000': 1,\n",
       "         'HSCLO chr11.64804001-64805000': 1,\n",
       "         'HSCLO chr11.67304001-67305000': 1,\n",
       "         'HSCLO chr11.120956001-120957000': 1,\n",
       "         'HSCLO chr11.3667001-3668000': 1,\n",
       "         'HSCLO chr11.75004001-75005000': 1,\n",
       "         'HSCLO chr11.100271001-100272000': 1,\n",
       "         'HSCLO chr11.102945001-102946000': 1,\n",
       "         'HSCLO chr11.8696001-8697000': 1,\n",
       "         'HSCLO chr11.9783001-9784000': 1,\n",
       "         'HSCLO chr11.130199001-130200000': 1,\n",
       "         'HSCLO chr11.127085001-127086000': 1,\n",
       "         'HSCLO chr11.66605001-66606000': 1,\n",
       "         'HSCLO chr11.130893001-130894000': 1,\n",
       "         'HSCLO chr11.114581001-114582000': 1,\n",
       "         'HSCLO chr11.6108001-6109000': 1,\n",
       "         'HSCLO chr11.47744001-47745000': 1,\n",
       "         'HSCLO chr11.66241001-66242000': 1,\n",
       "         'HSCLO chr11.32934001-32935000': 1,\n",
       "         'HSCLO chr11.9749001-9750000': 1,\n",
       "         'HSCLO chr11.46367001-46368000': 1,\n",
       "         'HSCLO chr11.117351001-117352000': 1,\n",
       "         'HSCLO chr11.67310001-67320000': 1,\n",
       "         'HSCLO chr11.67310001-67311000': 1,\n",
       "         'HSCLO chr11.77960001-77970000': 1,\n",
       "         'HSCLO chr11.77960001-77961000': 1,\n",
       "         'HSCLO chr11.615001-616000': 1,\n",
       "         'HSCLO chr11.66857001-66858000': 1,\n",
       "         'HSCLO chr11.16269001-16270000': 1,\n",
       "         'HSCLO chr11.55343001-55344000': 1,\n",
       "         'HSCLO chr11.94178001-94179000': 1,\n",
       "         'HSCLO chr11.75983001-75984000': 1,\n",
       "         'HSCLO chr11.125635001-125636000': 1,\n",
       "         'HSCLO chr11.62146001-62147000': 1,\n",
       "         'HSCLO chr11.124038001-124039000': 1,\n",
       "         'HSCLO chr11.67302001-67303000': 1,\n",
       "         'HSCLO chr11.61300001-61400000': 1,\n",
       "         'HSCLO chr11.61300001-61310000': 1,\n",
       "         'HSCLO chr11.61300001-61301000': 1,\n",
       "         'HSCLO chr11.119079001-119080000': 1,\n",
       "         'HSCLO chr11.119013001-119014000': 1,\n",
       "         'HSCLO chr11.108073001-108074000': 1,\n",
       "         'HSCLO chr11.125637001-125638000': 1,\n",
       "         'HSCLO chr11.32642001-32643000': 1,\n",
       "         'HSCLO chr11.56413001-56414000': 1,\n",
       "         'HSCLO chr11.118690001-118700000': 1,\n",
       "         'HSCLO chr11.118690001-118691000': 1,\n",
       "         'HSCLO chr11.134482001-134483000': 1,\n",
       "         'HSCLO chr11.27057001-27058000': 1,\n",
       "         'HSCLO chr11.5002001-5003000': 1,\n",
       "         'HSCLO chr11.89875001-89876000': 1,\n",
       "         'HSCLO chr11.106810001-106820000': 1,\n",
       "         'HSCLO chr11.106810001-106811000': 1,\n",
       "         'HSCLO chr11.128974001-128975000': 1,\n",
       "         'HSCLO chr11.66275001-66276000': 1,\n",
       "         'HSCLO chr11.75006001-75007000': 1,\n",
       "         'HSCLO chr11.122774001-122775000': 1,\n",
       "         'HSCLO chr11.123559001-123560000': 1,\n",
       "         'HSCLO chr12.27961001-27962000': 1,\n",
       "         'HSCLO chr12.50077001-50078000': 1,\n",
       "         'HSCLO chr12.49549001-49550000': 1,\n",
       "         'HSCLO chr12.123785001-123786000': 1,\n",
       "         'HSCLO chr12.45554001-45555000': 1,\n",
       "         'HSCLO chr12.10126001-10127000': 1,\n",
       "         'HSCLO chr12.62592001-62593000': 1,\n",
       "         'HSCLO chr12.7690001-7700000': 1,\n",
       "         'HSCLO chr12.7690001-7691000': 1,\n",
       "         'HSCLO chr12.51346001-51347000': 1,\n",
       "         'HSCLO chr12.66138001-66139000': 1,\n",
       "         'HSCLO chr12.52822001-52823000': 1,\n",
       "         'HSCLO chr12.108545001-108546000': 1,\n",
       "         'HSCLO chr12.39613001-39614000': 1,\n",
       "         'HSCLO chr12.9166001-9167000': 1,\n",
       "         'HSCLO chr12.122218001-122219000': 1,\n",
       "         'HSCLO chr12.202001-203000': 1,\n",
       "         'HSCLO chr12.133058001-133059000': 1,\n",
       "         'HSCLO chr12.81277001-81278000': 1,\n",
       "         'HSCLO chr12.38724001-38725000': 1,\n",
       "         'HSCLO chr12.80588001-80589000': 1,\n",
       "         'HSCLO chr12.6905001-6906000': 1,\n",
       "         'HSCLO chr12.14938001-14939000': 1,\n",
       "         'HSCLO chr12.56639001-56640000': 1,\n",
       "         'HSCLO chr12.102958001-102959000': 1,\n",
       "         'HSCLO chr12.94262001-94263000': 1,\n",
       "         'HSCLO chr12.55470001-55480000': 1,\n",
       "         'HSCLO chr12.55470001-55471000': 1,\n",
       "         'HSCLO chr12.9595001-9596000': 1,\n",
       "         'HSCLO chr12.66353001-66354000': 1,\n",
       "         'HSCLO chr12.113918001-113919000': 1,\n",
       "         'HSCLO chr12.8848001-8849000': 1,\n",
       "         'HSCLO chr12.50130001-50140000': 1,\n",
       "         'HSCLO chr12.50130001-50131000': 1,\n",
       "         'HSCLO chr12.92777001-92778000': 1,\n",
       "         'HSCLO chr12.103557001-103558000': 1,\n",
       "         'HSCLO chr12.51341001-51342000': 1,\n",
       "         'HSCLO chr12.48141001-48142000': 1,\n",
       "         'HSCLO chr12.96500001-96600000': 1,\n",
       "         'HSCLO chr12.96500001-96510000': 1,\n",
       "         'HSCLO chr12.96500001-96501000': 1,\n",
       "         'HSCLO chr12.113186001-113187000': 1,\n",
       "         'HSCLO chr12.39341001-39342000': 1,\n",
       "         'HSCLO chr12.40488001-40489000': 1,\n",
       "         'HSCLO chr12.55827001-55828000': 1,\n",
       "         'HSCLO chr12.44876001-44877000': 1,\n",
       "         'HSCLO chr12.1490001-1500000': 1,\n",
       "         'HSCLO chr12.1490001-1491000': 1,\n",
       "         'HSCLO chr12.62508001-62509000': 1,\n",
       "         'HSCLO chr12.2688001-2689000': 1,\n",
       "         'HSCLO chr12.10825001-10826000': 1,\n",
       "         'HSCLO chr12.47757001-47758000': 1,\n",
       "         'HSCLO chr12.110647001-110648000': 1,\n",
       "         'HSCLO chr12.31697001-31698000': 1,\n",
       "         'HSCLO chr12.109559001-109560000': 1,\n",
       "         'HSCLO chr12.2865001-2866000': 1,\n",
       "         'HSCLO chr12.31985001-31986000': 1,\n",
       "         'HSCLO chr12.52899001-52900000': 1,\n",
       "         'HSCLO chr12.100612001-100613000': 1,\n",
       "         'HSCLO chr12.103648001-103649000': 1,\n",
       "         'HSCLO chr12.7441001-7442000': 1,\n",
       "         'HSCLO chr12.40484001-40485000': 1,\n",
       "         'HSCLO chr12.64838001-64839000': 1,\n",
       "         'HSCLO chr12.106437001-106438000': 1,\n",
       "         'HSCLO chr12.51056001-51057000': 1,\n",
       "         'HSCLO chr12.21491001-21492000': 1,\n",
       "         'HSCLO chr12.26657001-26658000': 1,\n",
       "         'HSCLO chr12.56018001-56019000': 1,\n",
       "         'HSCLO chr12.64194001-64195000': 1,\n",
       "         'HSCLO chr12.29477001-29478000': 1,\n",
       "         'HSCLO chr12.57763001-57764000': 1,\n",
       "         'HSCLO chr12.112918001-112919000': 1,\n",
       "         'HSCLO chr12.48049001-48050000': 1,\n",
       "         'HSCLO chr12.56165001-56166000': 1,\n",
       "         'HSCLO chr12.43425001-43426000': 1,\n",
       "         'HSCLO chr12.49037001-49038000': 1,\n",
       "         'HSCLO chr12.111519001-111520000': 1,\n",
       "         'HSCLO chr12.2959001-2960000': 1,\n",
       "         'HSCLO chr12.49625001-49626000': 1,\n",
       "         'HSCLO chr12.19283001-19284000': 1,\n",
       "         'HSCLO chr12.6548001-6549000': 1,\n",
       "         'HSCLO chr13.51467001-51468000': 1,\n",
       "         'HSCLO chr13.33823001-33824000': 1,\n",
       "         'HSCLO chr13.24961001-24962000': 1,\n",
       "         'HSCLO chr13.23329001-23330000': 1,\n",
       "         'HSCLO chr13.49661001-49662000': 1,\n",
       "         'HSCLO chr13.21140001-21150000': 1,\n",
       "         'HSCLO chr13.21140001-21141000': 1,\n",
       "         'HSCLO chr13.51245001-51246000': 1,\n",
       "         'HSCLO chr13.98845001-98846000': 1,\n",
       "         'HSCLO chr13.52378001-52379000': 1,\n",
       "         'HSCLO chr13.27823001-27824000': 1,\n",
       "         'HSCLO chr13.20729001-20730000': 1,\n",
       "         'HSCLO chr13.29428001-29429000': 1,\n",
       "         'HSCLO chr13.83880001-83890000': 1,\n",
       "         'HSCLO chr13.83880001-83881000': 1,\n",
       "         'HSCLO chr13.110719001-110720000': 1,\n",
       "         'HSCLO chr13.45975001-45976000': 1,\n",
       "         'HSCLO chr13.53041001-53042000': 1,\n",
       "         'HSCLO chr13.77077001-77078000': 1,\n",
       "         'HSCLO chr13.21172001-21173000': 1,\n",
       "         'HSCLO chr13.33060001-33070000': 1,\n",
       "         'HSCLO chr13.33060001-33061000': 1,\n",
       "         'HSCLO chr13.30469001-30470000': 1,\n",
       "         'HSCLO chr13.71572001-71573000': 1,\n",
       "         'HSCLO chr13.77098001-77099000': 1,\n",
       "         'HSCLO chr13.102729001-102730000': 1,\n",
       "         'HSCLO chr13.112068001-112069000': 1,\n",
       "         'HSCLO chr13.77746001-77747000': 1,\n",
       "         'HSCLO chr13.36112001-36113000': 1,\n",
       "         'HSCLO chr13.110893001-110894000': 1,\n",
       "         'HSCLO chr13.113344001-113345000': 1,\n",
       "         'HSCLO chr13.63839001-63840000': 1,\n",
       "         'HSCLO chr13.33053001-33054000': 1,\n",
       "         'HSCLO chr13.72781001-72782000': 1,\n",
       "         'HSCLO chr13.26539001-26540000': 1,\n",
       "         'HSCLO chr13.28383001-28384000': 1,\n",
       "         'HSCLO chr14.73611001-73612000': 1,\n",
       "         'HSCLO chr14.45246001-45247000': 1,\n",
       "         'HSCLO chr14.22007001-22008000': 1,\n",
       "         'HSCLO chr14.45176001-45177000': 1,\n",
       "         'HSCLO chr14.73595001-73596000': 1,\n",
       "         'HSCLO chr14.97683001-97684000': 1,\n",
       "         'HSCLO chr14.45173001-45174000': 1,\n",
       "         'HSCLO chr14.22904001-22905000': 1,\n",
       "         'HSCLO chr14.92161001-92162000': 1,\n",
       "         'HSCLO chr14.57481001-57482000': 1,\n",
       "         'HSCLO chr14.23059001-23060000': 1,\n",
       "         'HSCLO chr14.75071001-75072000': 1,\n",
       "         'HSCLO chr14.50398001-50399000': 1,\n",
       "         'HSCLO chr14.88658001-88659000': 1,\n",
       "         'HSCLO chr14.77320001-77330000': 1,\n",
       "         'HSCLO chr14.77320001-77321000': 1,\n",
       "         'HSCLO chr14.102251001-102252000': 1,\n",
       "         'HSCLO chr14.57491001-57492000': 1,\n",
       "         'HSCLO chr14.103998001-103999000': 1,\n",
       "         'HSCLO chr14.90654001-90655000': 1,\n",
       "         'HSCLO chr14.88185001-88186000': 1,\n",
       "         'HSCLO chr14.104989001-104990000': 1,\n",
       "         'HSCLO chr14.102222001-102223000': 1,\n",
       "         'HSCLO chr14.64188001-64189000': 1,\n",
       "         'HSCLO chr14.34768001-34769000': 1,\n",
       "         'HSCLO chr14.30605001-30606000': 1,\n",
       "         'HSCLO chr14.44945001-44946000': 1,\n",
       "         'HSCLO chr14.100954001-100955000': 1,\n",
       "         'HSCLO chr14.67592001-67593000': 1,\n",
       "         'HSCLO chr14.95662001-95663000': 1,\n",
       "         'HSCLO chr14.104746001-104747000': 1,\n",
       "         'HSCLO chr14.92704001-92705000': 1,\n",
       "         'HSCLO chr14.20891001-20892000': 1,\n",
       "         'HSCLO chr14.24179001-24180000': 1,\n",
       "         'HSCLO chr14.35700001-35800000': 1,\n",
       "         'HSCLO chr14.35700001-35710000': 1,\n",
       "         'HSCLO chr14.35700001-35701000': 1,\n",
       "         'HSCLO chr14.55351001-55352000': 1,\n",
       "         'HSCLO chr14.59320001-59330000': 1,\n",
       "         'HSCLO chr14.59320001-59321000': 1,\n",
       "         'HSCLO chr14.35123001-35124000': 1,\n",
       "         'HSCLO chr14.106810001-106820000': 1,\n",
       "         'HSCLO chr14.106810001-106811000': 1,\n",
       "         'HSCLO chr14.22841001-22842000': 1,\n",
       "         'HSCLO chr14.23571001-23572000': 1,\n",
       "         'HSCLO chr14.99398001-99399000': 1,\n",
       "         'HSCLO chr14.73105001-73106000': 1,\n",
       "         'HSCLO chr14.105907001-105908000': 1,\n",
       "         'HSCLO chr14.21321001-21322000': 1,\n",
       "         'HSCLO chr14.77027001-77028000': 1,\n",
       "         'HSCLO chr14.63716001-63717000': 1,\n",
       "         'HSCLO chr14.100397001-100398000': 1,\n",
       "         'HSCLO chr15.63651001-63652000': 1,\n",
       "         'HSCLO chr15.56444001-56445000': 1,\n",
       "         'HSCLO chr15.55238001-55239000': 1,\n",
       "         'HSCLO chr15.34250001-34260000': 1,\n",
       "         'HSCLO chr15.34250001-34251000': 1,\n",
       "         'HSCLO chr15.60390001-60400000': 1,\n",
       "         'HSCLO chr15.60390001-60391000': 1,\n",
       "         'HSCLO chr15.20349001-20350000': 1,\n",
       "         'HSCLO chr15.41573001-41574000': 1,\n",
       "         'HSCLO chr15.43631001-43632000': 1,\n",
       "         'HSCLO chr15.42692001-42693000': 1,\n",
       "         'HSCLO chr15.62722001-62723000': 1,\n",
       "         'HSCLO chr15.84798001-84799000': 1,\n",
       "         'HSCLO chr15.41990001-42000000': 1,\n",
       "         'HSCLO chr15.41990001-41991000': 1,\n",
       "         'HSCLO chr15.93002001-93003000': 1,\n",
       "         'HSCLO chr15.72344001-72345000': 1,\n",
       "         'HSCLO chr15.34856001-34857000': 1,\n",
       "         'HSCLO chr15.43814001-43815000': 1,\n",
       "         'HSCLO chr15.54015001-54016000': 1,\n",
       "         'HSCLO chr15.68313001-68314000': 1,\n",
       "         'HSCLO chr15.58013001-58014000': 1,\n",
       "         'HSCLO chr15.92924001-92925000': 1,\n",
       "         'HSCLO chr15.50569001-50570000': 1,\n",
       "         'HSCLO chr15.58541001-58542000': 1,\n",
       "         'HSCLO chr15.93257001-93258000': 1,\n",
       "         'HSCLO chr15.40737001-40738000': 1,\n",
       "         'HSCLO chr15.48756001-48757000': 1,\n",
       "         'HSCLO chr15.85067001-85068000': 1,\n",
       "         'HSCLO chr15.50439001-50440000': 1,\n",
       "         'HSCLO chr15.35538001-35539000': 1,\n",
       "         'HSCLO chr15.51768001-51769000': 1,\n",
       "         'HSCLO chr15.43206001-43207000': 1,\n",
       "         'HSCLO chr15.82434001-82435000': 1,\n",
       "         'HSCLO chr15.86554001-86555000': 1,\n",
       "         'HSCLO chr15.84572001-84573000': 1,\n",
       "         'HSCLO chr15.44537001-44538000': 1,\n",
       "         'HSCLO chr15.89388001-89389000': 1,\n",
       "         'HSCLO chr15.42736001-42737000': 1,\n",
       "         'HSCLO chr15.80929001-80930000': 1,\n",
       "         'HSCLO chr15.75361001-75362000': 1,\n",
       "         'HSCLO chr15.82759001-82760000': 1,\n",
       "         'HSCLO chr15.39739001-39740000': 1,\n",
       "         'HSCLO chr15.58893001-58894000': 1,\n",
       "         'HSCLO chr15.74191001-74192000': 1,\n",
       "         'HSCLO chr15.75422001-75423000': 1,\n",
       "         'HSCLO chr15.88881001-88882000': 1,\n",
       "         'HSCLO chr15.92997001-92998000': 1,\n",
       "         'HSCLO chr15.55430001-55440000': 1,\n",
       "         'HSCLO chr15.55430001-55431000': 1,\n",
       "         'HSCLO chr15.52232001-52233000': 1,\n",
       "         'HSCLO chr15.85533001-85534000': 1,\n",
       "         'HSCLO chr15.89733001-89734000': 1,\n",
       "         'HSCLO chr15.80552001-80553000': 1,\n",
       "         'HSCLO chr15.44672001-44673000': 1,\n",
       "         'HSCLO chr15.23339001-23340000': 1,\n",
       "         'HSCLO chr15.101313001-101314000': 1,\n",
       "         'HSCLO chr15.45133001-45134000': 1,\n",
       "         'HSCLO chr15.90939001-90940000': 1,\n",
       "         'HSCLO chr15.27719001-27720000': 1,\n",
       "         'HSCLO chr15.28702001-28703000': 1,\n",
       "         'HSCLO chr15.48157001-48158000': 1,\n",
       "         'HSCLO chr15.88099001-88100000': 1,\n",
       "         'HSCLO chr16.50334001-50335000': 1,\n",
       "         'HSCLO chr16.31261001-31262000': 1,\n",
       "         'HSCLO chr16.21062001-21063000': 1,\n",
       "         'HSCLO chr16.30517001-30518000': 1,\n",
       "         'HSCLO chr16.4743001-4744000': 1,\n",
       "         'HSCLO chr16.55812001-55813000': 1,\n",
       "         'HSCLO chr16.1241001-1242000': 1,\n",
       "         'HSCLO chr16.3300001-3400000': 1,\n",
       "         'HSCLO chr16.3300001-3310000': 1,\n",
       "         'HSCLO chr16.3300001-3301000': 1,\n",
       "         'HSCLO chr16.9763001-9764000': 1,\n",
       "         'HSCLO chr16.79211001-79212000': 1,\n",
       "         'HSCLO chr16.4461001-4462000': 1,\n",
       "         'HSCLO chr16.15708001-15709000': 1,\n",
       "         'HSCLO chr16.56834001-56835000': 1,\n",
       "         'HSCLO chr16.67870001-67880000': 1,\n",
       "         'HSCLO chr16.67870001-67871000': 1,\n",
       "         'HSCLO chr16.29390001-29400000': 1,\n",
       "         'HSCLO chr16.29390001-29391000': 1,\n",
       "         'HSCLO chr16.84480001-84490000': 1,\n",
       "         'HSCLO chr16.84480001-84481000': 1,\n",
       "         'HSCLO chr16.68085001-68086000': 1,\n",
       "         'HSCLO chr16.16190001-16200000': 1,\n",
       "         'HSCLO chr16.16190001-16191000': 1,\n",
       "         'HSCLO chr16.1776001-1777000': 1,\n",
       "         'HSCLO chr16.67000001-68000000': 1,\n",
       "         'HSCLO chr16.67000001-67100000': 1,\n",
       "         'HSCLO chr16.67000001-67010000': 1,\n",
       "         'HSCLO chr16.67000001-67001000': 1,\n",
       "         'HSCLO chr16.53605001-53606000': 1,\n",
       "         'HSCLO chr16.3729001-3730000': 1,\n",
       "         'HSCLO chr16.11828001-11829000': 1,\n",
       "         'HSCLO chr16.55328001-55329000': 1,\n",
       "         'HSCLO chr16.1446001-1447000': 1,\n",
       "         'HSCLO chr16.58543001-58544000': 1,\n",
       "         'HSCLO chr16.5033001-5034000': 1,\n",
       "         'HSCLO chr16.23069001-23070000': 1,\n",
       "         'HSCLO chr16.56832001-56833000': 1,\n",
       "         'HSCLO chr16.452001-453000': 1,\n",
       "         'HSCLO chr16.5078001-5079000': 1,\n",
       "         'HSCLO chr16.57772001-57773000': 1,\n",
       "         'HSCLO chr16.89226001-89227000': 1,\n",
       "         'HSCLO chr16.68214001-68215000': 1,\n",
       "         'HSCLO chr16.10773001-10774000': 1,\n",
       "         'HSCLO chr16.19009001-19010000': 1,\n",
       "         'HSCLO chr16.795001-796000': 1,\n",
       "         'HSCLO chr16.3069001-3070000': 1,\n",
       "         'HSCLO chr16.19827001-19828000': 1,\n",
       "         'HSCLO chr16.81693001-81694000': 1,\n",
       "         'HSCLO chr16.2181001-2182000': 1,\n",
       "         'HSCLO chr16.77884001-77885000': 1,\n",
       "         'HSCLO chr16.5031001-5032000': 1,\n",
       "         'HSCLO chr16.27511001-27512000': 1,\n",
       "         'HSCLO chr16.30651001-30652000': 1,\n",
       "         'HSCLO chr16.81160001-81170000': 1,\n",
       "         'HSCLO chr16.81160001-81161000': 1,\n",
       "         'HSCLO chr16.88489001-88490000': 1,\n",
       "         'HSCLO chr16.739001-740000': 1,\n",
       "         'HSCLO chr16.1729001-1730000': 1,\n",
       "         'HSCLO chr16.50711001-50712000': 1,\n",
       "         'HSCLO chr16.3590001-3600000': 1,\n",
       "         'HSCLO chr16.3590001-3591000': 1,\n",
       "         'HSCLO chr16.8895001-8896000': 1,\n",
       "         'HSCLO chr16.16114001-16115000': 1,\n",
       "         'HSCLO chr16.88734001-88735000': 1,\n",
       "         'HSCLO chr16.51140001-51150000': 1,\n",
       "         'HSCLO chr16.51140001-51141000': 1,\n",
       "         'HSCLO chr16.8780001-8790000': 1,\n",
       "         'HSCLO chr16.8780001-8781000': 1,\n",
       "         'HSCLO chr16.71849001-71850000': 1,\n",
       "         'HSCLO chr16.72124001-72125000': 1,\n",
       "         'HSCLO chr16.23535001-23536000': 1,\n",
       "         'HSCLO chr16.72788001-72789000': 1,\n",
       "         'HSCLO chr16.67381001-67382000': 1,\n",
       "         'HSCLO chr16.72014001-72015000': 1,\n",
       "         'HSCLO chr17.31305001-31306000': 1,\n",
       "         'HSCLO chr17.3640001-3650000': 1,\n",
       "         'HSCLO chr17.3640001-3641000': 1,\n",
       "         'HSCLO chr17.41610001-41620000': 1,\n",
       "         'HSCLO chr17.41610001-41611000': 1,\n",
       "         'HSCLO chr17.10313001-10314000': 1,\n",
       "         'HSCLO chr17.16006001-16007000': 1,\n",
       "         'HSCLO chr17.47156001-47157000': 1,\n",
       "         'HSCLO chr17.61388001-61389000': 1,\n",
       "         'HSCLO chr17.41465001-41466000': 1,\n",
       "         'HSCLO chr17.16830001-16840000': 1,\n",
       "         'HSCLO chr17.16830001-16831000': 1,\n",
       "         'HSCLO chr17.20427001-20428000': 1,\n",
       "         'HSCLO chr17.76009001-76010000': 1,\n",
       "         'HSCLO chr17.74864001-74865000': 1,\n",
       "         'HSCLO chr17.76469001-76470000': 1,\n",
       "         'HSCLO chr17.79737001-79738000': 1,\n",
       "         'HSCLO chr17.7899001-7900000': 1,\n",
       "         'HSCLO chr17.76153001-76154000': 1,\n",
       "         'HSCLO chr17.10533001-10534000': 1,\n",
       "         'HSCLO chr17.7940001-7950000': 1,\n",
       "         'HSCLO chr17.7940001-7941000': 1,\n",
       "         'HSCLO chr17.7430001-7440000': 1,\n",
       "         'HSCLO chr17.7430001-7431000': 1,\n",
       "         'HSCLO chr17.35553001-35554000': 1,\n",
       "         'HSCLO chr17.39944001-39945000': 1,\n",
       "         'HSCLO chr17.62690001-62700000': 1,\n",
       "         'HSCLO chr17.62690001-62691000': 1,\n",
       "         'HSCLO chr17.72599001-72600000': 1,\n",
       "         'HSCLO chr17.78086001-78087000': 1,\n",
       "         'HSCLO chr17.8888001-8889000': 1,\n",
       "         'HSCLO chr17.40356001-40357000': 1,\n",
       "         'HSCLO chr17.43018001-43019000': 1,\n",
       "         'HSCLO chr17.3825001-3826000': 1,\n",
       "         'HSCLO chr17.59212001-59213000': 1,\n",
       "         'HSCLO chr17.78270001-78280000': 1,\n",
       "         'HSCLO chr17.78270001-78271000': 1,\n",
       "         'HSCLO chr17.50426001-50427000': 1,\n",
       "         'HSCLO chr17.40318001-40319000': 1,\n",
       "         'HSCLO chr17.80923001-80924000': 1,\n",
       "         'HSCLO chr17.10516001-10517000': 1,\n",
       "         'HSCLO chr17.56950001-56960000': 1,\n",
       "         'HSCLO chr17.56950001-56951000': 1,\n",
       "         'HSCLO chr17.74704001-74705000': 1,\n",
       "         'HSCLO chr17.43484001-43485000': 1,\n",
       "         'HSCLO chr17.20579001-20580000': 1,\n",
       "         'HSCLO chr17.28634001-28635000': 1,\n",
       "         'HSCLO chr17.47891001-47892000': 1,\n",
       "         'HSCLO chr17.44902001-44903000': 1,\n",
       "         'HSCLO chr17.78468001-78469000': 1,\n",
       "         'HSCLO chr17.8753001-8754000': 1,\n",
       "         'HSCLO chr17.4017001-4018000': 1,\n",
       "         'HSCLO chr17.44662001-44663000': 1,\n",
       "         'HSCLO chr17.17997001-17998000': 1,\n",
       "         'HSCLO chr17.8313001-8314000': 1,\n",
       "         'HSCLO chr17.49168001-49169000': 1,\n",
       "         'HSCLO chr17.82084001-82085000': 1,\n",
       "         'HSCLO chr17.7661001-7662000': 1,\n",
       "         'HSCLO chr17.10523001-10524000': 1,\n",
       "         'HSCLO chr17.42329001-42330000': 1,\n",
       "         'HSCLO chr17.81191001-81192000': 1,\n",
       "         'HSCLO chr17.57257001-57258000': 1,\n",
       "         'HSCLO chr17.35150001-35160000': 1,\n",
       "         'HSCLO chr17.35150001-35151000': 1,\n",
       "         'HSCLO chr17.41364001-41365000': 1,\n",
       "         'HSCLO chr17.28635001-28636000': 1,\n",
       "         'HSCLO chr17.73286001-73287000': 1,\n",
       "         'HSCLO chr17.80190001-80200000': 1,\n",
       "         'HSCLO chr17.80190001-80191000': 1,\n",
       "         'HSCLO chr17.50599001-50600000': 1,\n",
       "         'HSCLO chr17.4802001-4803000': 1,\n",
       "         'HSCLO chr17.7929001-7930000': 1,\n",
       "         'HSCLO chr17.41855001-41856000': 1,\n",
       "         'HSCLO chr17.39871001-39872000': 1,\n",
       "         'HSCLO chr17.58310001-58320000': 1,\n",
       "         'HSCLO chr17.58310001-58311000': 1,\n",
       "         'HSCLO chr17.78479001-78480000': 1,\n",
       "         'HSCLO chr17.82324001-82325000': 1,\n",
       "         'HSCLO chr17.28912001-28913000': 1,\n",
       "         'HSCLO chr17.36212001-36213000': 1,\n",
       "         'HSCLO chr17.44323001-44324000': 1,\n",
       "         'HSCLO chr17.39462001-39463000': 1,\n",
       "         'HSCLO chr17.40699001-40700000': 1,\n",
       "         'HSCLO chr17.75612001-75613000': 1,\n",
       "         'HSCLO chr17.75320001-75330000': 1,\n",
       "         'HSCLO chr17.75320001-75321000': 1,\n",
       "         'HSCLO chr17.63607001-63608000': 1,\n",
       "         'HSCLO chr17.28328001-28329000': 1,\n",
       "         'HSCLO chr17.41479001-41480000': 1,\n",
       "         'HSCLO chr17.56848001-56849000': 1,\n",
       "         'HSCLO chr17.82207001-82208000': 1,\n",
       "         'HSCLO chr17.7587001-7588000': 1,\n",
       "         'HSCLO chr17.18732001-18733000': 1,\n",
       "         'HSCLO chr17.7923001-7924000': 1,\n",
       "         'HSCLO chr18.58931001-58932000': 1,\n",
       "         'HSCLO chr18.50947001-50948000': 1,\n",
       "         'HSCLO chr18.2891001-2892000': 1,\n",
       "         'HSCLO chr18.46024001-46025000': 1,\n",
       "         'HSCLO chr18.23479001-23480000': 1,\n",
       "         'HSCLO chr18.44950001-44960000': 1,\n",
       "         'HSCLO chr18.44950001-44951000': 1,\n",
       "         'HSCLO chr18.36160001-36170000': 1,\n",
       "         'HSCLO chr18.36160001-36161000': 1,\n",
       "         'HSCLO chr18.7023001-7024000': 1,\n",
       "         'HSCLO chr18.47100001-47200000': 1,\n",
       "         'HSCLO chr18.47100001-47110000': 1,\n",
       "         'HSCLO chr18.47100001-47101000': 1,\n",
       "         'HSCLO chr18.35082001-35083000': 1,\n",
       "         'HSCLO chr18.33333001-33334000': 1,\n",
       "         'HSCLO chr18.22992001-22993000': 1,\n",
       "         'HSCLO chr18.61502001-61503000': 1,\n",
       "         'HSCLO chr18.46546001-46547000': 1,\n",
       "         'HSCLO chr18.3188001-3189000': 1,\n",
       "         'HSCLO chr18.63587001-63588000': 1,\n",
       "         'HSCLO chr18.36215001-36216000': 1,\n",
       "         'HSCLO chr18.3896001-3897000': 1,\n",
       "         'HSCLO chr18.10714001-10715000': 1,\n",
       "         'HSCLO chr18.47098001-47099000': 1,\n",
       "         'HSCLO chr18.36805001-36806000': 1,\n",
       "         'HSCLO chr18.62563001-62564000': 1,\n",
       "         'HSCLO chr18.46122001-46123000': 1,\n",
       "         'HSCLO chr18.12987001-12988000': 1,\n",
       "         'HSCLO chr18.79372001-79373000': 1,\n",
       "         'HSCLO chr18.2772001-2773000': 1,\n",
       "         'HSCLO chr18.33744001-33745000': 1,\n",
       "         'HSCLO chr18.58916001-58917000': 1,\n",
       "         'HSCLO chr19.11178001-11179000': 1,\n",
       "         'HSCLO chr19.19341001-19342000': 1,\n",
       "         'HSCLO chr19.1991001-1992000': 1,\n",
       "         'HSCLO chr19.34958001-34959000': 1,\n",
       "         'HSCLO chr19.37601001-37602000': 1,\n",
       "         'HSCLO chr19.57754001-57755000': 1,\n",
       "         'HSCLO chr19.35279001-35280000': 1,\n",
       "         'HSCLO chr19.18123001-18124000': 1,\n",
       "         'HSCLO chr19.23745001-23746000': 1,\n",
       "         'HSCLO chr19.47483001-47484000': 1,\n",
       "         'HSCLO chr19.15831001-15832000': 1,\n",
       "         'HSCLO chr19.47085001-47086000': 1,\n",
       "         'HSCLO chr19.48622001-48623000': 1,\n",
       "         'HSCLO chr19.9157001-9158000': 1,\n",
       "         'HSCLO chr19.15200001-15300000': 1,\n",
       "         'HSCLO chr19.15200001-15210000': 1,\n",
       "         'HSCLO chr19.15200001-15201000': 1,\n",
       "         'HSCLO chr19.8858001-8859000': 1,\n",
       "         'HSCLO chr19.13599001-13600000': 1,\n",
       "         'HSCLO chr19.42301001-42302000': 1,\n",
       "         'HSCLO chr19.41121001-41122000': 1,\n",
       "         'HSCLO chr19.2915001-2916000': 1,\n",
       "         'HSCLO chr19.34533001-34534000': 1,\n",
       "         'HSCLO chr19.4891001-4892000': 1,\n",
       "         'HSCLO chr19.10581001-10582000': 1,\n",
       "         'HSCLO chr19.11381001-11382000': 1,\n",
       "         'HSCLO chr19.58477001-58478000': 1,\n",
       "         'HSCLO chr19.49860001-49870000': 1,\n",
       "         'HSCLO chr19.49860001-49861000': 1,\n",
       "         'HSCLO chr19.50417001-50418000': 1,\n",
       "         'HSCLO chr19.54752001-54753000': 1,\n",
       "         'HSCLO chr19.9186001-9187000': 1,\n",
       "         'HSCLO chr19.7642001-7643000': 1,\n",
       "         'HSCLO chr19.40234001-40235000': 1,\n",
       "         'HSCLO chr19.8945001-8946000': 1,\n",
       "         'HSCLO chr19.7910001-7920000': 1,\n",
       "         'HSCLO chr19.7910001-7911000': 1,\n",
       "         'HSCLO chr19.8487001-8488000': 1,\n",
       "         'HSCLO chr19.7083001-7084000': 1,\n",
       "         'HSCLO chr19.7909001-7910000': 1,\n",
       "         'HSCLO chr19.7596001-7597000': 1,\n",
       "         'HSCLO chr19.7900001-8000000': 1,\n",
       "         'HSCLO chr19.7900001-7910000': 1,\n",
       "         'HSCLO chr19.7900001-7901000': 1,\n",
       "         'HSCLO chr19.16989001-16990000': 1,\n",
       "         'HSCLO chr19.50310001-50320000': 1,\n",
       "         'HSCLO chr19.50310001-50311000': 1,\n",
       "         'HSCLO chr19.49335001-49336000': 1,\n",
       "         'HSCLO chr19.49478001-49479000': 1,\n",
       "         'HSCLO chr19.57874001-57875000': 1,\n",
       "         'HSCLO chr19.6477001-6478000': 1,\n",
       "         'HSCLO chr19.44176001-44177000': 1,\n",
       "         'HSCLO chr19.16020001-16030000': 1,\n",
       "         'HSCLO chr19.16020001-16021000': 1,\n",
       "         'HSCLO chr19.49116001-49117000': 1,\n",
       "         'HSCLO chr19.44522001-44523000': 1,\n",
       "         'HSCLO chr19.8953001-8954000': 1,\n",
       "         'HSCLO chr19.33208001-33209000': 1,\n",
       "         'HSCLO chr19.44916001-44917000': 1,\n",
       "         'HSCLO chr19.10318001-10319000': 1,\n",
       "         'HSCLO chr19.54983001-54984000': 1,\n",
       "         'HSCLO chr19.35786001-35787000': 1,\n",
       "         'HSCLO chr19.10091001-10092000': 1,\n",
       "         'HSCLO chr19.33215001-33216000': 1,\n",
       "         'HSCLO chr19.58388001-58389000': 1,\n",
       "         'HSCLO chr19.55481001-55482000': 1,\n",
       "         'HSCLO chr19.45395001-45396000': 1,\n",
       "         'HSCLO chr19.37889001-37890000': 1,\n",
       "         'HSCLO chr19.54595001-54596000': 1,\n",
       "         'HSCLO chr19.17719001-17720000': 1,\n",
       "         'HSCLO chr19.4159001-4160000': 1,\n",
       "         'HSCLO chr19.43267001-43268000': 1,\n",
       "         'HSCLO chr19.51824001-51825000': 1,\n",
       "         'HSCLO chr19.17327001-17328000': 1,\n",
       "         'HSCLO chr19.49171001-49172000': 1,\n",
       "         'HSCLO chr19.49211001-49212000': 1,\n",
       "         'HSCLO chr19.58052001-58053000': 1,\n",
       "         'HSCLO chr19.18592001-18593000': 1,\n",
       "         'HSCLO chr19.3942001-3943000': 1,\n",
       "         'HSCLO chr19.19344001-19345000': 1,\n",
       "         'HSCLO chr19.41392001-41393000': 1,\n",
       "         'HSCLO chr19.19224001-19225000': 1,\n",
       "         'HSCLO chr19.48670001-48680000': 1,\n",
       "         'HSCLO chr19.48670001-48671000': 1,\n",
       "         'HSCLO chr19.14482001-14483000': 1,\n",
       "         'HSCLO chr19.39101001-39102000': 1,\n",
       "         'HSCLO chr19.40567001-40568000': 1,\n",
       "         'HSCLO chr19.41617001-41618000': 1,\n",
       "         'HSCLO chr19.11342001-11343000': 1,\n",
       "         'HSCLO chr19.50496001-50497000': 1,\n",
       "         'HSCLO chr19.54052001-54053000': 1,\n",
       "         'HSCLO chr19.55759001-55760000': 1,\n",
       "         'HSCLO chr19.35527001-35528000': 1,\n",
       "         'HSCLO chr19.54250001-54260000': 1,\n",
       "         'HSCLO chr19.54250001-54251000': 1,\n",
       "         'HSCLO chr19.45594001-45595000': 1,\n",
       "         'HSCLO chr19.15696001-15697000': 1,\n",
       "         'HSCLO chr19.14592001-14593000': 1,\n",
       "         'HSCLO chr2.55026001-55027000': 1,\n",
       "         'HSCLO chr20.55026001-55027000': 1,\n",
       "         'HSCLO chr2.222558001-222559000': 1,\n",
       "         'HSCLO chr2.764001-765000': 1,\n",
       "         'HSCLO chr20.764001-765000': 1,\n",
       "         'HSCLO chr21.764001-765000': 1,\n",
       "         'HSCLO chr22.764001-765000': 1,\n",
       "         'HSCLO chr2.45418001-45419000': 1,\n",
       "         'HSCLO chr20.45418001-45419000': 1,\n",
       "         'HSCLO chr21.45418001-45419000': 1,\n",
       "         'HSCLO chr22.45418001-45419000': 1,\n",
       "         'HSCLO chr2.36763001-36764000': 1,\n",
       "         'HSCLO chr20.36763001-36764000': 1,\n",
       "         'HSCLO chr21.36763001-36764000': 1,\n",
       "         'HSCLO chr22.36763001-36764000': 1,\n",
       "         'HSCLO chr2.26456001-26457000': 1,\n",
       "         'HSCLO chr20.26456001-26457000': 1,\n",
       "         'HSCLO chr21.26456001-26457000': 1,\n",
       "         'HSCLO chr22.26456001-26457000': 1,\n",
       "         'HSCLO chr2.144389001-144390000': 1,\n",
       "         'HSCLO chr2.55640001-55650000': 1,\n",
       "         'HSCLO chr2.55640001-55641000': 1,\n",
       "         'HSCLO chr20.55640001-55650000': 1,\n",
       "         'HSCLO chr20.55640001-55641000': 1,\n",
       "         'HSCLO chr2.74285001-74286000': 1,\n",
       "         'HSCLO chr2.96598001-96599000': 1,\n",
       "         'HSCLO chr2.159256001-159257000': 1,\n",
       "         'HSCLO chr2.46425001-46426000': 1,\n",
       "         'HSCLO chr20.46425001-46426000': 1,\n",
       "         'HSCLO chr21.46425001-46426000': 1,\n",
       "         'HSCLO chr22.46425001-46426000': 1,\n",
       "         'HSCLO chr2.32509001-32510000': 1,\n",
       "         'HSCLO chr20.32509001-32510000': 1,\n",
       "         'HSCLO chr21.32509001-32510000': 1,\n",
       "         'HSCLO chr22.32509001-32510000': 1,\n",
       "         'HSCLO chr2.209896001-209897000': 1,\n",
       "         'HSCLO chr2.195934001-195935000': 1,\n",
       "         'HSCLO chr2.46602001-46603000': 1,\n",
       "         'HSCLO chr20.46602001-46603000': 1,\n",
       "         'HSCLO chr21.46602001-46603000': 1,\n",
       "         'HSCLO chr22.46602001-46603000': 1,\n",
       "         'HSCLO chr2.87348001-87349000': 1,\n",
       "         'HSCLO chr2.201272001-201273000': 1,\n",
       "         'HSCLO chr2.69556001-69557000': 1,\n",
       "         'HSCLO chr2.27607001-27608000': 1,\n",
       "         'HSCLO chr20.27607001-27608000': 1,\n",
       "         'HSCLO chr21.27607001-27608000': 1,\n",
       "         'HSCLO chr22.27607001-27608000': 1,\n",
       "         'HSCLO chr2.30393001-30394000': 1,\n",
       "         'HSCLO chr20.30393001-30394000': 1,\n",
       "         'HSCLO chr21.30393001-30394000': 1,\n",
       "         'HSCLO chr22.30393001-30394000': 1,\n",
       "         'HSCLO chr2.120090001-120100000': 1,\n",
       "         'HSCLO chr2.120090001-120091000': 1,\n",
       "         'HSCLO chr2.28541001-28542000': 1,\n",
       "         'HSCLO chr20.28541001-28542000': 1,\n",
       "         'HSCLO chr21.28541001-28542000': 1,\n",
       "         'HSCLO chr22.28541001-28542000': 1,\n",
       "         'HSCLO chr2.74958001-74959000': 1,\n",
       "         'HSCLO chr2.189565001-189566000': 1,\n",
       "         'HSCLO chr2.44808001-44809000': 1,\n",
       "         'HSCLO chr20.44808001-44809000': 1,\n",
       "         'HSCLO chr21.44808001-44809000': 1,\n",
       "         'HSCLO chr22.44808001-44809000': 1,\n",
       "         'HSCLO chr2.33263001-33264000': 1,\n",
       "         'HSCLO chr20.33263001-33264000': 1,\n",
       "         'HSCLO chr21.33263001-33264000': 1,\n",
       "         'HSCLO chr22.33263001-33264000': 1,\n",
       "         'HSCLO chr2.169132001-169133000': 1,\n",
       "         'HSCLO chr2.178620001-178630000': 1,\n",
       "         'HSCLO chr2.178620001-178621000': 1,\n",
       "         'HSCLO chr2.197078001-197079000': 1,\n",
       "         'HSCLO chr2.20568001-20569000': 1,\n",
       "         'HSCLO chr20.20568001-20569000': 1,\n",
       "         'HSCLO chr21.20568001-20569000': 1,\n",
       "         'HSCLO chr22.20568001-20569000': 1,\n",
       "         'HSCLO chr2.2339001-2340000': 1,\n",
       "         'HSCLO chr20.2339001-2340000': 1,\n",
       "         'HSCLO chr21.2339001-2340000': 1,\n",
       "         'HSCLO chr22.2339001-2340000': 1,\n",
       "         'HSCLO chr2.33509001-33510000': 1,\n",
       "         'HSCLO chr20.33509001-33510000': 1,\n",
       "         'HSCLO chr21.33509001-33510000': 1,\n",
       "         'HSCLO chr22.33509001-33510000': 1,\n",
       "         'HSCLO chr2.37510001-37520000': 1,\n",
       "         'HSCLO chr2.37510001-37511000': 1,\n",
       "         'HSCLO chr20.37510001-37520000': 1,\n",
       "         'HSCLO chr20.37510001-37511000': 1,\n",
       "         'HSCLO chr21.37510001-37520000': 1,\n",
       "         'HSCLO chr21.37510001-37511000': 1,\n",
       "         'HSCLO chr22.37510001-37520000': 1,\n",
       "         'HSCLO chr22.37510001-37511000': 1,\n",
       "         'HSCLO chr2.1453001-1454000': 1,\n",
       "         'HSCLO chr20.1453001-1454000': 1,\n",
       "         'HSCLO chr21.1453001-1454000': 1,\n",
       "         'HSCLO chr22.1453001-1454000': 1,\n",
       "         'HSCLO chr2.46381001-46382000': 1,\n",
       "         'HSCLO chr20.46381001-46382000': 1,\n",
       "         'HSCLO chr21.46381001-46382000': 1,\n",
       "         'HSCLO chr22.46381001-46382000': 1,\n",
       "         'HSCLO chr2.238163001-238164000': 1,\n",
       "         'HSCLO chr2.29490001-29500000': 1,\n",
       "         ...})"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Counter([i.split(':')[0] for i in n['node_id']])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "503837ad",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
