{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "5654ad8e",
   "metadata": {},
   "source": [
    "# Somatic Variants from KF-TALL were pulled directly from consensus.maf files, (NOT the VWB somatic tables, like the NBL data was)\n",
    "\n",
    "### Like the NBL somatic variants, these variants have been filtered to include VEP IMPACT = HIGH.\n",
    "More info can be found at the somatic_files.ipynb in the U24 Data Studio session in the Taylor_Urbs project on CAVATICA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "c8ddf0a6",
   "metadata": {},
   "outputs": [],
   "source": [
    "#scp /Users/stearb/Downloads/TALL_somatic_variants_VEP_HIGH_July7th.csv \\\n",
    "#stearb-hpc9.research.chop.edu:/home/stearb/U24/data/somatic_variants/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "76019d91",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import polars as pl\n",
    "import numpy as np\n",
    "\n",
    "def fill_missing_cols(df):\n",
    "    if 'node_id' not in df.columns:\n",
    "        raise ValueError('Must have at least a \"node_id\" column.')\n",
    "    \n",
    "    all_cols = set([ 'node_label', 'node_synonyms', 'node_dbxrefs',\n",
    "            'node_definition','node_namespace','value','lowerbound','upperbound','unit'])\n",
    "    missing_cols = list(all_cols - set(df.columns))\n",
    "    nan_cols_df = pd.DataFrame(np.full([len(df), len(missing_cols)], np.nan),columns=missing_cols)\n",
    "\n",
    "    if isinstance(df, pd.DataFrame):\n",
    "        nan_cols_df.index = df.index\n",
    "        return pd.concat([df,nan_cols_df],axis=1)\n",
    "    elif isinstance(df, pl.DataFrame):\n",
    "        return pl.concat([df,pl.from_pandas(nan_cols_df)],how='horizontal') # no index for polars\n",
    "    else:\n",
    "        raise ValueError(f'Must Pass either a pandas DataFrame or a polars DataFrame but recieved \"{type(df)}\".')\n",
    "    \n",
    "\n",
    "def bin_column(df, col2bin=''):\n",
    "    '''works with polars df'''\n",
    "    \n",
    "    '''tpm_bins = list([0.0000000,7e-4,8e-4,9e-4]) + list(np.linspace(1e-3,9e-3,9)) + \\\n",
    "               list(np.round(np.linspace(1e-2,9e-2,9),2)) + list(np.round(np.linspace(.1,1,10),2)) + \\\n",
    "               list(np.linspace(2,100,99)) + list(np.arange(100,1100,100)[1:]) +  \\\n",
    "                list(np.arange(2000,11000,1000)) + list(np.arange(20_000,110_000,10_000)) + [300_000] '''\n",
    "    \n",
    "    bins = np.unique(hscloChrom['lowerbound'].values) #hscloChrom['lowerbound'].values\n",
    "    # check that none of the disease expression values are out of range, ie larger than the max bin value\n",
    "    # use for pandas\n",
    "    #if np.sum(df['Disease_MeanTpm'] > np.max(tpm_bins)) == 0:\n",
    "        \n",
    "    # use for polars\n",
    "    if len(df.filter(df[col2bin] > np.max(bins))) == 0:\n",
    "        df_binned = df.with_columns(  df[col2bin].cut(breaks=bins).alias('bins')  )\n",
    "        # check that order has been maintained #assert pl.all(df[col2bin] == df_binned[col2bin])\n",
    "    else:\n",
    "        print('OUT OF RANGE ERROR!')\n",
    "        assert False \n",
    "    return df_binned"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c4d52780",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "c564a9e3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HGVSg</th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>dbSNP_RS</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>SYMBOL_SOURCE</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>gnomad_3_1_1_AF</th>\n",
       "      <th>gnomad_3_1_1_AF_popmax</th>\n",
       "      <th>gnomad_3_1_1_AF_non_cancer_popmax</th>\n",
       "      <th>gnomad_3_1_1_AF_non_cancer_all_popmax</th>\n",
       "      <th>Matched_Norm_Sample_Barcode</th>\n",
       "      <th>Tumor_Sample_Barcode</th>\n",
       "      <th>path</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1:g.26697414G&gt;A</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>p.Trp337Ter</td>\n",
       "      <td>novel</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>BS_XD2PEVH1</td>\n",
       "      <td>BS_190CS8A4</td>\n",
       "      <td>/sbgenomics/project-files/000ef953-f37a-411b-8...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr6:g.41936013dup</td>\n",
       "      <td>CCND3</td>\n",
       "      <td>p.Arg271ProfsTer53</td>\n",
       "      <td>novel</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:1585</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>6.5e-06</td>\n",
       "      <td>1.45e-05</td>\n",
       "      <td>1.5e-05</td>\n",
       "      <td>1.5e-05</td>\n",
       "      <td>BS_XD2PEVH1</td>\n",
       "      <td>BS_190CS8A4</td>\n",
       "      <td>/sbgenomics/project-files/000ef953-f37a-411b-8...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr12:g.79594385G&gt;A</td>\n",
       "      <td>PAWR</td>\n",
       "      <td>p.Gln294Ter</td>\n",
       "      <td>novel</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:8614</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>BS_XD2PEVH1</td>\n",
       "      <td>BS_190CS8A4</td>\n",
       "      <td>/sbgenomics/project-files/000ef953-f37a-411b-8...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr15:g.45417349C&gt;T</td>\n",
       "      <td>SPATA5L1</td>\n",
       "      <td>p.Arg640Ter</td>\n",
       "      <td>rs199667349</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:28762</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>8.5e-05</td>\n",
       "      <td>0.000132</td>\n",
       "      <td>0.000123</td>\n",
       "      <td>0.000123</td>\n",
       "      <td>BS_XD2PEVH1</td>\n",
       "      <td>BS_190CS8A4</td>\n",
       "      <td>/sbgenomics/project-files/000ef953-f37a-411b-8...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr19:g.5923819del</td>\n",
       "      <td>RANBP3</td>\n",
       "      <td>p.Glu365ArgfsTer79</td>\n",
       "      <td>novel</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:9850</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>BS_XD2PEVH1</td>\n",
       "      <td>BS_190CS8A4</td>\n",
       "      <td>/sbgenomics/project-files/000ef953-f37a-411b-8...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 HGVSg Hugo_Symbol               HGVSp     dbSNP_RS  \\\n",
       "0   chr1:g.26697414G>A      ARID1A         p.Trp337Ter        novel   \n",
       "1   chr6:g.41936013dup       CCND3  p.Arg271ProfsTer53        novel   \n",
       "2  chr12:g.79594385G>A        PAWR         p.Gln294Ter        novel   \n",
       "3  chr15:g.45417349C>T    SPATA5L1         p.Arg640Ter  rs199667349   \n",
       "4   chr19:g.5923819del      RANBP3  p.Glu365ArgfsTer79        novel   \n",
       "\n",
       "          Consequence SYMBOL_SOURCE     HGNC_ID         BIOTYPE IMPACT  \\\n",
       "0         stop_gained          HGNC  HGNC:11110  protein_coding   HIGH   \n",
       "1  frameshift_variant          HGNC   HGNC:1585  protein_coding   HIGH   \n",
       "2         stop_gained          HGNC   HGNC:8614  protein_coding   HIGH   \n",
       "3         stop_gained          HGNC  HGNC:28762  protein_coding   HIGH   \n",
       "4  frameshift_variant          HGNC   HGNC:9850  protein_coding   HIGH   \n",
       "\n",
       "  gnomad_3_1_1_AF gnomad_3_1_1_AF_popmax gnomad_3_1_1_AF_non_cancer_popmax  \\\n",
       "0               .                      .                                 .   \n",
       "1         6.5e-06               1.45e-05                           1.5e-05   \n",
       "2               .                      .                                 .   \n",
       "3         8.5e-05               0.000132                          0.000123   \n",
       "4               .                      .                                 .   \n",
       "\n",
       "  gnomad_3_1_1_AF_non_cancer_all_popmax Matched_Norm_Sample_Barcode  \\\n",
       "0                                     .                 BS_XD2PEVH1   \n",
       "1                               1.5e-05                 BS_XD2PEVH1   \n",
       "2                                     .                 BS_XD2PEVH1   \n",
       "3                              0.000123                 BS_XD2PEVH1   \n",
       "4                                     .                 BS_XD2PEVH1   \n",
       "\n",
       "  Tumor_Sample_Barcode                                               path  \n",
       "0          BS_190CS8A4  /sbgenomics/project-files/000ef953-f37a-411b-8...  \n",
       "1          BS_190CS8A4  /sbgenomics/project-files/000ef953-f37a-411b-8...  \n",
       "2          BS_190CS8A4  /sbgenomics/project-files/000ef953-f37a-411b-8...  \n",
       "3          BS_190CS8A4  /sbgenomics/project-files/000ef953-f37a-411b-8...  \n",
       "4          BS_190CS8A4  /sbgenomics/project-files/000ef953-f37a-411b-8...  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv('/home/stearb/U24/data/somatic_variants/TALL_somatic_variants_VEP_HIGH_July7th.csv')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "0fa12a53",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HGVSg</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2506</th>\n",
       "      <td>chr10:g.73016878C&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4539</th>\n",
       "      <td>chr1:g.186367984delinsGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10596</th>\n",
       "      <td>chrM:m.15611G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1395</th>\n",
       "      <td>chr20:g.62310830C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7790</th>\n",
       "      <td>chr10:g.68292024C&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8320</th>\n",
       "      <td>chr5:g.33576677C&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3569</th>\n",
       "      <td>chr12:g.12718026_12718032del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7887</th>\n",
       "      <td>chr8:g.21985665_21985666insCGTTCACTCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13865</th>\n",
       "      <td>chr12:g.56716428delinsGTG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>993</th>\n",
       "      <td>chr1:g.92834790_92834794del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13647</th>\n",
       "      <td>chr20:g.4248231C&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6897</th>\n",
       "      <td>chr3:g.50218324_50218325insCT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       HGVSg\n",
       "2506                     chr10:g.73016878C>A\n",
       "4539                chr1:g.186367984delinsGG\n",
       "10596                        chrM:m.15611G>A\n",
       "1395                     chr20:g.62310830C>T\n",
       "7790                     chr10:g.68292024C>A\n",
       "8320                      chr5:g.33576677C>A\n",
       "3569            chr12:g.12718026_12718032del\n",
       "7887   chr8:g.21985665_21985666insCGTTCACTCC\n",
       "13865              chr12:g.56716428delinsGTG\n",
       "993              chr1:g.92834790_92834794del\n",
       "13647                     chr20:g.4248231C>A\n",
       "6897           chr3:g.50218324_50218325insCT"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[['HGVSg']].sample(12)\n",
    "\n",
    "#[i[-3:] for i in df['HGVSg']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "b9f15f25",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Counter({'del': 3237,\n",
       "         'G>T': 2507,\n",
       "         'C>A': 2384,\n",
       "         'dup': 1395,\n",
       "         'G>A': 1298,\n",
       "         'C>T': 1273,\n",
       "         'A>G': 162,\n",
       "         'T>C': 141,\n",
       "         'A>T': 139,\n",
       "         'T>G': 116,\n",
       "         'A>C': 95,\n",
       "         'T>A': 92,\n",
       "         'CCC': 71,\n",
       "         'G>C': 60,\n",
       "         'C>G': 58,\n",
       "         'nsA': 54,\n",
       "         'sGG': 52,\n",
       "         'GGG': 51,\n",
       "         'CCG': 47,\n",
       "         'nsT': 44,\n",
       "         'CTC': 39,\n",
       "         'nsG': 39,\n",
       "         'TCG': 39,\n",
       "         'sCC': 36,\n",
       "         'ACC': 33,\n",
       "         'CCT': 32,\n",
       "         'TCC': 32,\n",
       "         'sTC': 29,\n",
       "         'GGA': 24,\n",
       "         'nsC': 24,\n",
       "         'GGT': 24,\n",
       "         'AAG': 22,\n",
       "         'TCT': 22,\n",
       "         'GGC': 21,\n",
       "         'GAA': 21,\n",
       "         'AGG': 21,\n",
       "         'GTC': 21,\n",
       "         'GCT': 20,\n",
       "         'AAA': 20,\n",
       "         'TTC': 19,\n",
       "         'sGA': 19,\n",
       "         'GCC': 19,\n",
       "         'TTT': 18,\n",
       "         'GAG': 18,\n",
       "         'AGA': 18,\n",
       "         'TGG': 18,\n",
       "         'CTT': 17,\n",
       "         'GTG': 17,\n",
       "         'ATC': 16,\n",
       "         'ACG': 16,\n",
       "         'TGA': 16,\n",
       "         'CGG': 15,\n",
       "         'TTA': 15,\n",
       "         'sAA': 15,\n",
       "         'sAG': 15,\n",
       "         'CAA': 15,\n",
       "         'TCA': 15,\n",
       "         'TTG': 15,\n",
       "         'CGC': 15,\n",
       "         'sCT': 15,\n",
       "         'ATT': 14,\n",
       "         'GCG': 14,\n",
       "         'CCA': 14,\n",
       "         'CTG': 13,\n",
       "         'GCA': 13,\n",
       "         'AGC': 13,\n",
       "         'TAA': 13,\n",
       "         'CTA': 13,\n",
       "         'TAC': 13,\n",
       "         'sAC': 13,\n",
       "         'GTT': 12,\n",
       "         'sTG': 12,\n",
       "         'CGA': 12,\n",
       "         'CAC': 12,\n",
       "         'AAC': 12,\n",
       "         'TAG': 11,\n",
       "         'ACA': 11,\n",
       "         'sAT': 11,\n",
       "         'sTT': 11,\n",
       "         'ATG': 10,\n",
       "         'CAT': 10,\n",
       "         'sGT': 9,\n",
       "         'CAG': 9,\n",
       "         'GAT': 8,\n",
       "         'sGC': 8,\n",
       "         'GAC': 8,\n",
       "         'TAT': 8,\n",
       "         'sTA': 8,\n",
       "         'TGT': 8,\n",
       "         'GTA': 7,\n",
       "         'AAT': 7,\n",
       "         'ATA': 7,\n",
       "         'ACT': 6,\n",
       "         'CGT': 6,\n",
       "         'TGC': 5,\n",
       "         'AGT': 5,\n",
       "         'sCG': 5,\n",
       "         'sCA': 3,\n",
       "         'inv': 1})"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import collections\n",
    "collections.Counter([i[-3:] for i in df['HGVSg']]) 'dup', 'del', 'inv', "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "6dace509",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1628114/2102876145.py:1: DtypeWarning: Columns (32,34,38,40,50) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  hgnc_master = pd.read_csv('/home/stearb/U24/data/helper_files/hgnc_master.txt',sep='\\t')\n"
     ]
    }
   ],
   "source": [
    "hgnc_master = pd.read_csv('/home/stearb/U24/data/helper_files/hgnc_master.txt',sep='\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "70794f22",
   "metadata": {},
   "outputs": [],
   "source": [
    "# merge in ensembl gene ids\n",
    "df = pd.merge(df,hgnc_master[['hgnc_id','ensembl_gene_id']].rename({'hgnc_id':'HGNC_ID'},axis=1))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "44da1634",
   "metadata": {},
   "source": [
    "# Define codes\n",
    "### Need to get mappings to ensembl proteins and ensembl transcripts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "bea30d93",
   "metadata": {},
   "outputs": [],
   "source": [
    "# variant\n",
    "df['HGVSg'] = df['HGVSg'].replace({':': '.'},regex=True) \n",
    "df['hgvsg_variant_code'] = 'HGVSG:' + df['HGVSg'] \n",
    "\n",
    "# ensembl gene\n",
    "df['ens_gene_code'] = 'ENSEMBL:' + df['ensembl_gene_id']\n",
    "\n",
    "# disease/MONDO for 'T-cell childhood acute lymphocytic leukemia'\n",
    "# https://www.ebi.ac.uk/ols4/search?q=t-all&ontology=mondo\n",
    "df['disease_code'] = 'MONDO:0000871'\n",
    "\n",
    "# cohort\n",
    "df['cohort_code'] = 'KFCOHORT:SD-AQ9KVN5P'\n",
    "\n",
    "# study\n",
    "df['study_code'] = 'KFSTUDY:KF-TALL-SOMATIC'"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2d40ba25",
   "metadata": {},
   "source": [
    "# Define edges"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "410e8c4d",
   "metadata": {},
   "source": [
    "### Variant - Gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "00c3a5df",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subject</th>\n",
       "      <th>predicate</th>\n",
       "      <th>object</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5407</th>\n",
       "      <td>HGVSG:chr15.g.23645455G&gt;T</td>\n",
       "      <td>related_to_gene</td>\n",
       "      <td>ENSEMBL:ENSG00000254585</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12710</th>\n",
       "      <td>HGVSG:chr3.g.195864183A&gt;C</td>\n",
       "      <td>related_to_gene</td>\n",
       "      <td>ENSEMBL:ENSG00000061938</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2361</th>\n",
       "      <td>HGVSG:chr17.g.66223697C&gt;T</td>\n",
       "      <td>related_to_gene</td>\n",
       "      <td>ENSEMBL:ENSG00000091583</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5116</th>\n",
       "      <td>HGVSG:chr16.g.8894828G&gt;A</td>\n",
       "      <td>related_to_gene</td>\n",
       "      <td>ENSEMBL:ENSG00000187555</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4175</th>\n",
       "      <td>HGVSG:chr1.g.244705700G&gt;T</td>\n",
       "      <td>related_to_gene</td>\n",
       "      <td>ENSEMBL:ENSG00000121644</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         subject        predicate                   object\n",
       "5407   HGVSG:chr15.g.23645455G>T  related_to_gene  ENSEMBL:ENSG00000254585\n",
       "12710  HGVSG:chr3.g.195864183A>C  related_to_gene  ENSEMBL:ENSG00000061938\n",
       "2361   HGVSG:chr17.g.66223697C>T  related_to_gene  ENSEMBL:ENSG00000091583\n",
       "5116    HGVSG:chr16.g.8894828G>A  related_to_gene  ENSEMBL:ENSG00000187555\n",
       "4175   HGVSG:chr1.g.244705700G>T  related_to_gene  ENSEMBL:ENSG00000121644"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "e_var_gene = df[['hgvsg_variant_code','ens_gene_code']].dropna().drop_duplicates().reset_index(drop=True)\n",
    "e_var_gene.columns = ['subject','object']\n",
    "\n",
    "e_var_gene['predicate'] = 'related_to_gene'\n",
    "e_var_gene = e_var_gene[['subject','predicate','object']]\n",
    "\n",
    "e_var_gene.sample(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "8a73b39e-26d5-45f2-b6bd-dee91a833dd3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HGVSg</th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>dbSNP_RS</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>SYMBOL_SOURCE</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>gnomad_3_1_1_AF</th>\n",
       "      <th>gnomad_3_1_1_AF_popmax</th>\n",
       "      <th>gnomad_3_1_1_AF_non_cancer_popmax</th>\n",
       "      <th>gnomad_3_1_1_AF_non_cancer_all_popmax</th>\n",
       "      <th>Matched_Norm_Sample_Barcode</th>\n",
       "      <th>Tumor_Sample_Barcode</th>\n",
       "      <th>path</th>\n",
       "      <th>ensembl_gene_id</th>\n",
       "      <th>hgvsg_variant_code</th>\n",
       "      <th>ens_gene_code</th>\n",
       "      <th>disease_code</th>\n",
       "      <th>cohort_code</th>\n",
       "      <th>study_code</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1.g.26697414G&gt;A</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>p.Trp337Ter</td>\n",
       "      <td>novel</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>BS_XD2PEVH1</td>\n",
       "      <td>BS_190CS8A4</td>\n",
       "      <td>/sbgenomics/project-files/000ef953-f37a-411b-8...</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>HGVSG:chr1.g.26697414G&gt;A</td>\n",
       "      <td>ENSEMBL:ENSG00000117713</td>\n",
       "      <td>MONDO:0000871</td>\n",
       "      <td>KFCOHORT:SD-AQ9KVN5P</td>\n",
       "      <td>KFSTUDY:KF-TALL-SOMATIC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr6.g.41936013dup</td>\n",
       "      <td>CCND3</td>\n",
       "      <td>p.Arg271ProfsTer53</td>\n",
       "      <td>novel</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:1585</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>6.5e-06</td>\n",
       "      <td>1.45e-05</td>\n",
       "      <td>1.5e-05</td>\n",
       "      <td>1.5e-05</td>\n",
       "      <td>BS_XD2PEVH1</td>\n",
       "      <td>BS_190CS8A4</td>\n",
       "      <td>/sbgenomics/project-files/000ef953-f37a-411b-8...</td>\n",
       "      <td>ENSG00000112576</td>\n",
       "      <td>HGVSG:chr6.g.41936013dup</td>\n",
       "      <td>ENSEMBL:ENSG00000112576</td>\n",
       "      <td>MONDO:0000871</td>\n",
       "      <td>KFCOHORT:SD-AQ9KVN5P</td>\n",
       "      <td>KFSTUDY:KF-TALL-SOMATIC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                HGVSg Hugo_Symbol               HGVSp dbSNP_RS  \\\n",
       "0  chr1.g.26697414G>A      ARID1A         p.Trp337Ter    novel   \n",
       "1  chr6.g.41936013dup       CCND3  p.Arg271ProfsTer53    novel   \n",
       "\n",
       "          Consequence SYMBOL_SOURCE     HGNC_ID         BIOTYPE IMPACT  \\\n",
       "0         stop_gained          HGNC  HGNC:11110  protein_coding   HIGH   \n",
       "1  frameshift_variant          HGNC   HGNC:1585  protein_coding   HIGH   \n",
       "\n",
       "  gnomad_3_1_1_AF gnomad_3_1_1_AF_popmax gnomad_3_1_1_AF_non_cancer_popmax  \\\n",
       "0               .                      .                                 .   \n",
       "1         6.5e-06               1.45e-05                           1.5e-05   \n",
       "\n",
       "  gnomad_3_1_1_AF_non_cancer_all_popmax Matched_Norm_Sample_Barcode  \\\n",
       "0                                     .                 BS_XD2PEVH1   \n",
       "1                               1.5e-05                 BS_XD2PEVH1   \n",
       "\n",
       "  Tumor_Sample_Barcode                                               path  \\\n",
       "0          BS_190CS8A4  /sbgenomics/project-files/000ef953-f37a-411b-8...   \n",
       "1          BS_190CS8A4  /sbgenomics/project-files/000ef953-f37a-411b-8...   \n",
       "\n",
       "   ensembl_gene_id        hgvsg_variant_code            ens_gene_code  \\\n",
       "0  ENSG00000117713  HGVSG:chr1.g.26697414G>A  ENSEMBL:ENSG00000117713   \n",
       "1  ENSG00000112576  HGVSG:chr6.g.41936013dup  ENSEMBL:ENSG00000112576   \n",
       "\n",
       "    disease_code           cohort_code               study_code  \n",
       "0  MONDO:0000871  KFCOHORT:SD-AQ9KVN5P  KFSTUDY:KF-TALL-SOMATIC  \n",
       "1  MONDO:0000871  KFCOHORT:SD-AQ9KVN5P  KFSTUDY:KF-TALL-SOMATIC  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.set_option('display.max_columns', None)\n",
    "df.head(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "91eacaaa-b3c7-4611-aa9e-deed384f8346",
   "metadata": {},
   "source": [
    "### Variant - Protein   --- *****No ENSEMBL protein IDs for the TALL dataset****"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "cc0ed987-5c4c-42ad-bb62-77863204e26e",
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "['hgvsp']",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m/tmp/ipykernel_1628114/1178206435.py\u001b[0m in \u001b[0;36m?\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;31m# Some extra formatting needs to be done for the protein code...\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0;31m# Drop rows where its NaN\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 4\u001b[0;31m \u001b[0mdf_prot\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdropna\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msubset\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'hgvsp'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      5\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      6\u001b[0m \u001b[0;31m# DROP rows where the ENSP is not defined.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      7\u001b[0m \u001b[0mdf_prot\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdf_prot\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mdf_prot\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'hgvsp'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstartswith\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'ENSP'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.10/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m?\u001b[0;34m(self, axis, how, thresh, subset, inplace, ignore_index)\u001b[0m\n\u001b[1;32m   6666\u001b[0m             \u001b[0max\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_axis\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0magg_axis\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   6667\u001b[0m             \u001b[0mindices\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0max\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_indexer_for\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msubset\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   6668\u001b[0m             \u001b[0mcheck\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mindices\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m-\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   6669\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mcheck\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0many\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 6670\u001b[0;31m                 \u001b[0;32mraise\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0marray\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msubset\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mcheck\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtolist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   6671\u001b[0m             \u001b[0magg_obj\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtake\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mindices\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0magg_axis\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   6672\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   6673\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mthresh\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mlib\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mno_default\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: ['hgvsp']"
     ]
    }
   ],
   "source": [
    "\n",
    "# Some extra formatting needs to be done for the protein code...\n",
    "\n",
    "# Drop rows where its NaN\n",
    "df_prot = df.dropna(subset=['hgvsp'])\n",
    "\n",
    "# DROP rows where the ENSP is not defined.\n",
    "df_prot = df_prot[df_prot['hgvsp'].str.startswith('ENSP')]\n",
    "\n",
    "df_prot['ens_protein_code'] = 'ENSEMBL:' + pd.Series([i.split('.')[0] for i in df_prot['hgvsp']])\n",
    "\n",
    "e_var_prot = df_prot[['hgvsg_variant_code','ens_protein_code']]\n",
    "e_var_prot.columns = ['subject','object']\n",
    "e_var_prot['predicate'] = 'has_protein'\n",
    "e_var_prot = e_var_prot[['subject','predicate','object']].dropna().drop_duplicates().reset_index(drop=True)\n",
    "e_var_prot.sample(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "768f34c1-6e9f-455f-92a4-d5f6c7068131",
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "['hgvsp']",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m/tmp/ipykernel_1628114/2011493956.py\u001b[0m in \u001b[0;36m?\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;31m# Drop rows where its NaN\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mdf_prot\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdropna\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msubset\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'hgvsp'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0;31m# DROP rows where the ENSP is not defined.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0mdf_prot\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdf_prot\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mdf_prot\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'hgvsp'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstartswith\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'ENSP'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.10/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m?\u001b[0;34m(self, axis, how, thresh, subset, inplace, ignore_index)\u001b[0m\n\u001b[1;32m   6666\u001b[0m             \u001b[0max\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_axis\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0magg_axis\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   6667\u001b[0m             \u001b[0mindices\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0max\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_indexer_for\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msubset\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   6668\u001b[0m             \u001b[0mcheck\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mindices\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m-\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   6669\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mcheck\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0many\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 6670\u001b[0;31m                 \u001b[0;32mraise\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0marray\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msubset\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mcheck\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtolist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   6671\u001b[0m             \u001b[0magg_obj\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtake\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mindices\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0magg_axis\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   6672\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   6673\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mthresh\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mlib\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mno_default\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: ['hgvsp']"
     ]
    }
   ],
   "source": [
    "\n",
    "# Drop rows where its NaN\n",
    "df_prot = df.dropna(subset=['HGVSp'])\n",
    "\n",
    "# DROP rows where the ENSP is not defined.\n",
    "df_prot = df_prot[df_prot['hgvsp'].str.startswith('ENSP')]\n",
    "\n",
    "df_prot['ens_protein_code'] = 'ENSEMBL:' + pd.Series([i.split('.')[0] for i in df_prot['hgvsp']])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fa913c7b-340d-4372-a360-1e986f85e837",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c1feab3a-34e0-4ea9-95c9-3b4214597473",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Drop rows where its NaN\n",
    "df_prot = df.dropna(subset=['hgvsp'])\n",
    "\n",
    "# DROP rows where the ENSP is not defined.\n",
    "df_prot = df_prot[df_prot['hgvsp'].str.startswith('ENSP')]\n",
    "\n",
    "df_prot['ens_protein_code'] = 'ENSEMBL:' + pd.Series([i.split('.')[0] for i in df_prot['hgvsp']])\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0aba6da6",
   "metadata": {},
   "source": [
    "### Variant - Cohort"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "f6ab4285",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subject</th>\n",
       "      <th>predicate</th>\n",
       "      <th>object</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>248</th>\n",
       "      <td>HGVSG:chr20.g.3044514G&gt;T</td>\n",
       "      <td>belongs_to_cohort</td>\n",
       "      <td>KFCOHORT:SD-AQ9KVN5P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13176</th>\n",
       "      <td>HGVSG:chr20.g.50903900C&gt;A</td>\n",
       "      <td>belongs_to_cohort</td>\n",
       "      <td>KFCOHORT:SD-AQ9KVN5P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9464</th>\n",
       "      <td>HGVSG:chr14.g.26448240T&gt;A</td>\n",
       "      <td>belongs_to_cohort</td>\n",
       "      <td>KFCOHORT:SD-AQ9KVN5P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11522</th>\n",
       "      <td>HGVSG:chr11.g.60339865G&gt;T</td>\n",
       "      <td>belongs_to_cohort</td>\n",
       "      <td>KFCOHORT:SD-AQ9KVN5P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12570</th>\n",
       "      <td>HGVSG:chr12.g.112887834G&gt;T</td>\n",
       "      <td>belongs_to_cohort</td>\n",
       "      <td>KFCOHORT:SD-AQ9KVN5P</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                          subject          predicate                object\n",
       "248      HGVSG:chr20.g.3044514G>T  belongs_to_cohort  KFCOHORT:SD-AQ9KVN5P\n",
       "13176   HGVSG:chr20.g.50903900C>A  belongs_to_cohort  KFCOHORT:SD-AQ9KVN5P\n",
       "9464    HGVSG:chr14.g.26448240T>A  belongs_to_cohort  KFCOHORT:SD-AQ9KVN5P\n",
       "11522   HGVSG:chr11.g.60339865G>T  belongs_to_cohort  KFCOHORT:SD-AQ9KVN5P\n",
       "12570  HGVSG:chr12.g.112887834G>T  belongs_to_cohort  KFCOHORT:SD-AQ9KVN5P"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "e_var_cohort = df[['hgvsg_variant_code','cohort_code']].dropna().drop_duplicates().reset_index(drop=True)\n",
    "e_var_cohort.columns = ['subject','object']\n",
    "e_var_cohort['predicate'] = 'belongs_to_cohort'\n",
    "e_var_cohort = e_var_cohort[['subject','predicate','object']]\n",
    "e_var_cohort.sample(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "70152ca1",
   "metadata": {},
   "source": [
    "### Study - Cohort"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "470ab44d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subject</th>\n",
       "      <th>predicate</th>\n",
       "      <th>object</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KFSTUDY:KF-TALL-SOMATIC</td>\n",
       "      <td>study_has_cohort</td>\n",
       "      <td>KFCOHORT:SD-AQ9KVN5P</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   subject         predicate                object\n",
       "0  KFSTUDY:KF-TALL-SOMATIC  study_has_cohort  KFCOHORT:SD-AQ9KVN5P"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "e_study_cohort = df[['study_code','cohort_code']].dropna().drop_duplicates().reset_index(drop=True)\n",
    "e_study_cohort.columns = ['subject','object']\n",
    "\n",
    "e_study_cohort['predicate'] = 'study_has_cohort'\n",
    "e_study_cohort = e_study_cohort[['subject','predicate','object']]\n",
    "e_study_cohort"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "59ffcfb1",
   "metadata": {},
   "source": [
    "### Variant - HSCLO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "1aa8eea2",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['chr1',\n",
       " 'chr10',\n",
       " 'chr11',\n",
       " 'chr12',\n",
       " 'chr13',\n",
       " 'chr14',\n",
       " 'chr15',\n",
       " 'chr16',\n",
       " 'chr17',\n",
       " 'chr18',\n",
       " 'chr19',\n",
       " 'chr2',\n",
       " 'chr20',\n",
       " 'chr21',\n",
       " 'chr22',\n",
       " 'chr3',\n",
       " 'chr4',\n",
       " 'chr5',\n",
       " 'chr6',\n",
       " 'chr7',\n",
       " 'chr8',\n",
       " 'chr9',\n",
       " 'chrX',\n",
       " 'chrY']"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "q = pl.scan_csv('/home/stearb/U24/data/HSCLO/OWLNETS_edgelist_HSCLO.tsv',separator='\\t')\n",
    "hsclo = q.select(['node_id','lowerbound','upperbound']).collect()\n",
    "hsclo = hsclo.to_pandas()\n",
    "\n",
    "chroms = np.unique([i.split(' ')[-1].split('.')[0] for i in hsclo.node_id.astype(str)])\n",
    "chroms = [i.tolist() for i in chroms]  \n",
    "chroms.remove('Human_Genome_hg38')\n",
    "chroms.remove('MtDNA')\n",
    "chroms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "6b0cb9fd",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\n",
      "1282 1279\n",
      "chr10\n",
      "730 728\n",
      "chr11\n",
      "902 899\n",
      "chr12\n",
      "774 769\n",
      "chr13\n",
      "246 244\n",
      "chr14\n",
      "563 562\n",
      "chr15\n",
      "442 440\n",
      "chr16\n",
      "706 704\n",
      "chr17\n",
      "764 760\n",
      "chr18\n",
      "181 181\n",
      "chr19\n",
      "879 879\n",
      "chr2\n",
      "4459 4443\n",
      "chr20\n",
      "371 368\n",
      "chr21\n",
      "141 141\n",
      "chr22\n",
      "261 260\n",
      "chr3\n",
      "727 726\n",
      "chr4\n",
      "619 616\n",
      "chr5\n",
      "721 718\n",
      "chr6\n",
      "675 674\n",
      "chr7\n",
      "610 607\n",
      "chr8\n",
      "457 457\n",
      "chr9\n",
      "807 807\n",
      "chrX\n",
      "668 667\n",
      "chrY\n",
      "11 11\n"
     ]
    }
   ],
   "source": [
    "e_var_hsclo_MASTER = list()\n",
    "\n",
    "# Go by chromosome and merge each variant to the correct HSCLO term/code\n",
    "# most of this is extracting the variant position from the hgvsg_variant_code col\n",
    "# and saving it to the variant_position col. Then bin on this col.\n",
    "\n",
    "for CHROM in chroms:\n",
    "    \n",
    "    print(CHROM)\n",
    "    \n",
    "    # search for 'chr1.' or 'chr2.' if CHROM is either 1 or 2. This prevents matching on 11,12,13, when matching on 'chr1'\n",
    "    if CHROM == 'chr1': CHROM = 'chr1\\.'\n",
    "    elif CHROM == 'chr2': CHROM == 'chr2\\.'\n",
    "    \n",
    "    # Select variant col which has the variant chromosomal position for just current CHROM\n",
    "    dfChrom = df[df['hgvsg_variant_code'].str.contains(CHROM)][['hgvsg_variant_code']].dropna()\n",
    "    \n",
    "    # format to extract just numerical position \n",
    "    dfChrom['variant_position'] = \\\n",
    "          [i.split('.')[-1].replace('del','').replace('ins','')\\\n",
    "            .replace('A','').replace('C','').replace('T','').replace('G','')\\\n",
    "            .replace('>','') if 'inv' not in i or 'dup' not in i else i for i\\\n",
    "            in dfChrom['hgvsg_variant_code']]\n",
    "\n",
    "    dfChrom['variant_position'] = dfChrom['variant_position'].replace('dup','',regex=True).replace('inv','',regex=True)\n",
    "    dfChrom['variant_position'] = [i.split('_')[0] for i in dfChrom['variant_position']]\n",
    "    dfChrom['variant_position'] = dfChrom['variant_position'].astype(int)\n",
    "    dfChrom = dfChrom.drop_duplicates().reset_index(drop=True)\n",
    "    \n",
    "    # get hsclo codes for current CHROM\n",
    "    hscloChrom = hsclo[hsclo['node_id'].str.contains(CHROM)].dropna()\n",
    "\n",
    "    ###### do the binning\n",
    "    dfChromBins = bin_column(pl.DataFrame(dfChrom),col2bin='variant_position').to_pandas()\n",
    "\n",
    "    # make lowerbound col to merge in hsclo codes\n",
    "    dfChromBins['lowerbound'] = [float(i.split(',')[0][1:]) for i in dfChromBins['bins']]\n",
    "\n",
    "    # merge in hsclo codes \n",
    "    dfMerged = pd.merge(dfChromBins,hscloChrom,on='lowerbound',how='left')\n",
    "    \n",
    "    # this may introduce an error, if im matching on lowerbound only, as there are multiple resolution\n",
    "    # levels, so upperbound must also be specified!\n",
    "    # ...check that the variant_position is b/t lower and upperbound\n",
    "    print(len(dfMerged) ,len(dfMerged[(dfMerged['variant_position'] > dfMerged['lowerbound']) &\\\n",
    "        (dfMerged['variant_position'] < dfMerged['upperbound'])]))\n",
    "    \n",
    "    #assert dfMerged.shape == dfMerged[(dfMerged['variant_position'] > dfMerged['lowerbound']) &\\\n",
    "    #    (dfMerged['variant_position'] < dfMerged['upperbound'])].shape\n",
    "    \n",
    "    e_var_hsclo = dfMerged[['hgvsg_variant_code','node_id']].drop_duplicates().dropna().reset_index(drop=True)\n",
    "    e_var_hsclo.columns = ['subject','object']\n",
    "    e_var_hsclo['predicate'] = 'has_location'\n",
    "    e_var_hsclo = e_var_hsclo[['subject','predicate','object']]\n",
    "    \n",
    "    # e_var_hsclo is longer than dfChrom bc each variant is mapped to multiple HSCLO resolution levels\n",
    "    #print(len(e_var_hsclo),len(dfChrom)) \n",
    "    \n",
    "    if len(e_var_hsclo_MASTER) == 0:\n",
    "        e_var_hsclo_MASTER = e_var_hsclo\n",
    "    else:\n",
    "        e_var_hsclo_MASTER = pd.concat([e_var_hsclo_MASTER,e_var_hsclo])\n",
    "       \n",
    "e_var_hsclo_MASTER = e_var_hsclo_MASTER.drop_duplicates().dropna().reset_index(drop=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "b9568c70",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subject</th>\n",
       "      <th>predicate</th>\n",
       "      <th>object</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HGVSG:chr1.g.26697414G&gt;A</td>\n",
       "      <td>has_location</td>\n",
       "      <td>HSCLO chr1.26697001-26698000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HGVSG:chr1.g.33586525C&gt;A</td>\n",
       "      <td>has_location</td>\n",
       "      <td>HSCLO chr1.33586001-33587000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HGVSG:chr1.g.24652520del</td>\n",
       "      <td>has_location</td>\n",
       "      <td>HSCLO chr1.24652001-24653000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HGVSG:chr1.g.167128564C&gt;T</td>\n",
       "      <td>has_location</td>\n",
       "      <td>HSCLO chr1.167128001-167129000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HGVSG:chr1.g.228279243G&gt;T</td>\n",
       "      <td>has_location</td>\n",
       "      <td>HSCLO chr1.228279001-228280000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17218</th>\n",
       "      <td>HGVSG:chrY.g.12720654del</td>\n",
       "      <td>has_location</td>\n",
       "      <td>HSCLO chrY.12720001-12721000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17219</th>\n",
       "      <td>HGVSG:chrY.g.13251168del</td>\n",
       "      <td>has_location</td>\n",
       "      <td>HSCLO chrY.13251001-13252000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17220</th>\n",
       "      <td>HGVSG:chrY.g.19739546del</td>\n",
       "      <td>has_location</td>\n",
       "      <td>HSCLO chrY.19739001-19740000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17221</th>\n",
       "      <td>HGVSG:chrY.g.2787297C&gt;A</td>\n",
       "      <td>has_location</td>\n",
       "      <td>HSCLO chrY.2787001-2788000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17222</th>\n",
       "      <td>HGVSG:chrY.g.12758509C&gt;T</td>\n",
       "      <td>has_location</td>\n",
       "      <td>HSCLO chrY.12758001-12759000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17223 rows  3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                         subject     predicate                          object\n",
       "0       HGVSG:chr1.g.26697414G>A  has_location    HSCLO chr1.26697001-26698000\n",
       "1       HGVSG:chr1.g.33586525C>A  has_location    HSCLO chr1.33586001-33587000\n",
       "2       HGVSG:chr1.g.24652520del  has_location    HSCLO chr1.24652001-24653000\n",
       "3      HGVSG:chr1.g.167128564C>T  has_location  HSCLO chr1.167128001-167129000\n",
       "4      HGVSG:chr1.g.228279243G>T  has_location  HSCLO chr1.228279001-228280000\n",
       "...                          ...           ...                             ...\n",
       "17218   HGVSG:chrY.g.12720654del  has_location    HSCLO chrY.12720001-12721000\n",
       "17219   HGVSG:chrY.g.13251168del  has_location    HSCLO chrY.13251001-13252000\n",
       "17220   HGVSG:chrY.g.19739546del  has_location    HSCLO chrY.19739001-19740000\n",
       "17221    HGVSG:chrY.g.2787297C>A  has_location      HSCLO chrY.2787001-2788000\n",
       "17222   HGVSG:chrY.g.12758509C>T  has_location    HSCLO chrY.12758001-12759000\n",
       "\n",
       "[17223 rows x 3 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "e_var_hsclo_MASTER"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ef2d0c95",
   "metadata": {},
   "source": [
    "# Save edges"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "729e680b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "4f8952c6",
   "metadata": {},
   "outputs": [],
   "source": [
    "edges_all = pd.concat([e_var_gene,e_var_cohort,e_study_cohort,\n",
    "                       e_var_hsclo_MASTER])   # e_var_prot,e_var_trans, e_trans_prot\n",
    "\n",
    "edges_all = edges_all.drop_duplicates().dropna().reset_index(drop=True)\n",
    "\n",
    "edges_all.to_csv('/mnt/isilon/opentargets/U24KG/data/owlnets_nodes_and_edges/TALL_somatic/OWLNETS_edgelist.txt',\n",
    "             sep= \"\\t\",index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "be26f8e7",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>node_id</th>\n",
       "      <th>node_label</th>\n",
       "      <th>node_definition</th>\n",
       "      <th>node_synonyms</th>\n",
       "      <th>upperbound</th>\n",
       "      <th>node_namespace</th>\n",
       "      <th>unit</th>\n",
       "      <th>value</th>\n",
       "      <th>node_dbxrefs</th>\n",
       "      <th>lowerbound</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HGVSG:chr1.g.26697414G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HGVSG:chr6.g.41936013dup</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HGVSG:chr12.g.79594385G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HGVSG:chr15.g.45417349C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HGVSG:chr19.g.5923819del</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20545</th>\n",
       "      <td>ENSEMBL:ENSG00000105298</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20546</th>\n",
       "      <td>ENSEMBL:ENSG00000184635</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20547</th>\n",
       "      <td>ENSEMBL:ENSG00000163159</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20548</th>\n",
       "      <td>ENSEMBL:ENSG00000113068</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20549</th>\n",
       "      <td>KFSTUDY:KF-TALL-SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>20550 rows  10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                         node_id  node_label  node_definition  node_synonyms  \\\n",
       "0       HGVSG:chr1.g.26697414G>A         NaN              NaN            NaN   \n",
       "1       HGVSG:chr6.g.41936013dup         NaN              NaN            NaN   \n",
       "2      HGVSG:chr12.g.79594385G>A         NaN              NaN            NaN   \n",
       "3      HGVSG:chr15.g.45417349C>T         NaN              NaN            NaN   \n",
       "4       HGVSG:chr19.g.5923819del         NaN              NaN            NaN   \n",
       "...                          ...         ...              ...            ...   \n",
       "20545    ENSEMBL:ENSG00000105298         NaN              NaN            NaN   \n",
       "20546    ENSEMBL:ENSG00000184635         NaN              NaN            NaN   \n",
       "20547    ENSEMBL:ENSG00000163159         NaN              NaN            NaN   \n",
       "20548    ENSEMBL:ENSG00000113068         NaN              NaN            NaN   \n",
       "20549    KFSTUDY:KF-TALL-SOMATIC         NaN              NaN            NaN   \n",
       "\n",
       "       upperbound  node_namespace  unit  value  node_dbxrefs  lowerbound  \n",
       "0             NaN             NaN   NaN    NaN           NaN         NaN  \n",
       "1             NaN             NaN   NaN    NaN           NaN         NaN  \n",
       "2             NaN             NaN   NaN    NaN           NaN         NaN  \n",
       "3             NaN             NaN   NaN    NaN           NaN         NaN  \n",
       "4             NaN             NaN   NaN    NaN           NaN         NaN  \n",
       "...           ...             ...   ...    ...           ...         ...  \n",
       "20545         NaN             NaN   NaN    NaN           NaN         NaN  \n",
       "20546         NaN             NaN   NaN    NaN           NaN         NaN  \n",
       "20547         NaN             NaN   NaN    NaN           NaN         NaN  \n",
       "20548         NaN             NaN   NaN    NaN           NaN         NaN  \n",
       "20549         NaN             NaN   NaN    NaN           NaN         NaN  \n",
       "\n",
       "[20550 rows x 10 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nodes_all"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "15f4b2ed-b88a-46d9-92a4-0c8f6c0453d5",
   "metadata": {},
   "source": [
    "# Save nodes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "5a4c5d57",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>node_id</th>\n",
       "      <th>node_namespace</th>\n",
       "      <th>node_definition</th>\n",
       "      <th>unit</th>\n",
       "      <th>node_label</th>\n",
       "      <th>node_dbxrefs</th>\n",
       "      <th>lowerbound</th>\n",
       "      <th>value</th>\n",
       "      <th>node_synonyms</th>\n",
       "      <th>upperbound</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HGVSG:chr1.g.26697414G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HGVSG:chr6.g.41936013dup</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HGVSG:chr12.g.79594385G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HGVSG:chr15.g.45417349C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HGVSG:chr19.g.5923819del</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20545</th>\n",
       "      <td>ENSEMBL:ENSG00000105298</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20546</th>\n",
       "      <td>ENSEMBL:ENSG00000184635</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20547</th>\n",
       "      <td>ENSEMBL:ENSG00000163159</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20548</th>\n",
       "      <td>ENSEMBL:ENSG00000113068</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20549</th>\n",
       "      <td>KFCOHORT:SD-AQ9KVN5P</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>20550 rows  10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                         node_id  node_namespace  node_definition  unit  \\\n",
       "0       HGVSG:chr1.g.26697414G>A             NaN              NaN   NaN   \n",
       "1       HGVSG:chr6.g.41936013dup             NaN              NaN   NaN   \n",
       "2      HGVSG:chr12.g.79594385G>A             NaN              NaN   NaN   \n",
       "3      HGVSG:chr15.g.45417349C>T             NaN              NaN   NaN   \n",
       "4       HGVSG:chr19.g.5923819del             NaN              NaN   NaN   \n",
       "...                          ...             ...              ...   ...   \n",
       "20545    ENSEMBL:ENSG00000105298             NaN              NaN   NaN   \n",
       "20546    ENSEMBL:ENSG00000184635             NaN              NaN   NaN   \n",
       "20547    ENSEMBL:ENSG00000163159             NaN              NaN   NaN   \n",
       "20548    ENSEMBL:ENSG00000113068             NaN              NaN   NaN   \n",
       "20549       KFCOHORT:SD-AQ9KVN5P             NaN              NaN   NaN   \n",
       "\n",
       "       node_label  node_dbxrefs  lowerbound  value  node_synonyms  upperbound  \n",
       "0             NaN           NaN         NaN    NaN            NaN         NaN  \n",
       "1             NaN           NaN         NaN    NaN            NaN         NaN  \n",
       "2             NaN           NaN         NaN    NaN            NaN         NaN  \n",
       "3             NaN           NaN         NaN    NaN            NaN         NaN  \n",
       "4             NaN           NaN         NaN    NaN            NaN         NaN  \n",
       "...           ...           ...         ...    ...            ...         ...  \n",
       "20545         NaN           NaN         NaN    NaN            NaN         NaN  \n",
       "20546         NaN           NaN         NaN    NaN            NaN         NaN  \n",
       "20547         NaN           NaN         NaN    NaN            NaN         NaN  \n",
       "20548         NaN           NaN         NaN    NaN            NaN         NaN  \n",
       "20549         NaN           NaN         NaN    NaN            NaN         NaN  \n",
       "\n",
       "[20550 rows x 10 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nodes_all = pd.concat([edges_all['subject'],edges_all['object']])\\\n",
    "                          .drop_duplicates().dropna().reset_index(drop=True)\n",
    "\n",
    "nodes_all = pd.DataFrame(nodes_all,columns=['node_id'])\n",
    "nodes_all = fill_missing_cols(nodes_all)\n",
    "\n",
    "# NO Need to save HSCLO node ids\n",
    "nodes_all = nodes_all[~nodes_all['node_id'].str.startswith('HSCLO')]\\\n",
    "    .drop_duplicates().dropna(subset=['node_id']).reset_index(drop=True)\n",
    "\n",
    "nodes_all"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "8c2a144e-3be0-43f5-9aed-5ba6bf2c9249",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "nodes_all.to_csv(\n",
    "    '/mnt/isilon/opentargets/U24KG/data/owlnets_nodes_and_edges/TALL_somatic/OWLNETS_node_metadata.txt',\n",
    "             sep= \"\\t\",index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c22566f0-05cb-4acc-a516-9e05b9c75222",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
